// auth0/dist/cjs/management/__generated/models/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConnectionCreateStrategyEnum = exports.ConnectedAccountAccessTypeEnum = exports.ClientUpdateOidcLogoutBackchannelLogoutInitiatorsSelectedInitiatorsEnum = exports.ClientUpdateOidcLogoutBackchannelLogoutInitiatorsModeEnum = exports.ClientUpdateJwtConfigurationAlgEnum = exports.ClientUpdateComplianceLevelEnum = exports.ClientUpdateOrganizationRequireBehaviorEnum = exports.ClientUpdateOrganizationUsageEnum = exports.ClientUpdateAppTypeEnum = exports.ClientUpdateTokenEndpointAuthMethodEnum = exports.ClientSessionTransferEnforceDeviceBindingEnum = exports.ClientSessionTransferAllowedAuthenticationMethodsEnum = exports.ClientRefreshTokenExpirationTypeEnum = exports.ClientRefreshTokenRotationTypeEnum = exports.ClientOrganizationDiscoveryEnum = exports.ClientOidcLogoutBackchannelLogoutInitiatorsSelectedInitiatorsEnum = exports.ClientOidcLogoutBackchannelLogoutInitiatorsModeEnum = exports.ClientJwtConfigurationAlgEnum = exports.ClientGrantCreateSubjectTypeEnum = exports.ClientGrantCreateOrganizationUsageEnum = exports.ClientGrantSubjectTypeEnum = exports.ClientGrantOrganizationUsageEnum = exports.ClientDefaultOrganizationFlowsEnum = exports.ClientCreateJwtConfigurationAlgEnum = exports.ClientCreateDefaultOrganizationFlowsEnum = exports.ClientCreateClientAuthenticationMethodsTlsClientAuthCredentialsInnerCredentialTypeEnum = exports.ClientCreateClientAuthenticationMethodsSelfSignedTlsClientAuthCredentialsInnerCredentialTypeEnum = exports.ClientCreateClientAuthenticationMethodsPrivateKeyJwtCredentialsInnerAlgEnum = exports.ClientCreateClientAuthenticationMethodsPrivateKeyJwtCredentialsInnerCredentialTypeEnum = exports.ClientCreateComplianceLevelEnum = exports.ClientCreateOrganizationRequireBehaviorEnum = exports.ClientCreateOrganizationUsageEnum = exports.ClientCreateAppTypeEnum = exports.ClientCreateTokenEndpointAuthMethodEnum = exports.ClientComplianceLevelEnum = exports.ClientOrganizationRequireBehaviorEnum = exports.ClientOrganizationUsageEnum = exports.ClientTokenEndpointAuthMethodEnum = exports.ClientAppTypeEnum = exports.CertificateCertificateAuthorityEnum = exports.CertificateStatusEnum = exports.CaptchaProviderEnum = exports.CaptchaEnforcementPolicyEnum = exports.BotDetectionLevelEnum = exports.BotDetectionChallengePolicyPasswordlessFlowEnum = exports.BotDetectionChallengePolicyPasswordResetFlowEnum = exports.BotDetectionChallengePolicyPasswordFlowEnum = exports.AttackProtectionCaptchaProviderId = exports.AsyncApprovalNotificationsChannelsEnum = exports.AssessorsTypeEnum = void 0;
exports.GetLogStreams200ResponseInnerOneOfTypeEnum = exports.GetLogStreams200ResponseInnerOneOfStatusEnum = exports.GetExecution200ResponseStatusEnum = exports.GetEncryptionKeys200ResponseOneOfInnerStateEnum = exports.GetEncryptionKeys200ResponseOneOfInnerTypeEnum = exports.GetEmailTemplatesByTemplateName200ResponseTemplateEnum = exports.GetCredentials200ResponseInnerAlgEnum = exports.GetConnectionsKeysResponseContentInnerKeyUseEnum = exports.GetBruteForceProtection200ResponseModeEnum = exports.GetBruteForceProtection200ResponseShieldsEnum = exports.GetBreachedPasswordDetection200ResponseStagePreUserRegistrationShieldsEnum = exports.GetBreachedPasswordDetection200ResponseStagePreChangePasswordShieldsEnum = exports.GetBreachedPasswordDetection200ResponseMethodEnum = exports.GetBreachedPasswordDetection200ResponseAdminNotificationFrequencyEnum = exports.GetBreachedPasswordDetection200ResponseShieldsEnum = exports.GetBrandingPhoneProviders200ResponseProvidersInnerConfigurationAnyOf1DeliveryMethodsEnum = exports.GetBrandingPhoneProviders200ResponseProvidersInnerConfigurationAnyOfDeliveryMethodsEnum = exports.GetBrandingPhoneProviders200ResponseProvidersInnerChannelEnum = exports.GetBrandingPhoneProviders200ResponseProvidersInnerNameEnum = exports.GetAuthenticationMethods200ResponseOneOfInnerAuthenticationMethodsInnerTypeEnum = exports.GetAuthenticationMethods200ResponseOneOfInnerPreferredAuthenticationMethodEnum = exports.GetAuthenticationMethods200ResponseOneOfInnerTypeEnum = exports.GetAllRendering200ResponseOneOfInnerRenderingModeEnum = exports.GetActions200ResponseActionsInnerSupportedTriggersInnerIdAnyOf = exports.GetActions200ResponseActionsInnerIntegrationCurrentReleaseRequiredSecretsInnerTypeEnum = exports.GetActions200ResponseActionsInnerIntegrationFeatureTypeEnum = exports.GetActions200ResponseActionsInnerStatusEnum = exports.GetActionVersions200ResponseVersionsInnerStatusEnum = exports.FactorNameEnum = exports.EnrollmentStatusEnum = exports.EnabledFeaturesEnum = exports.EmailTemplateUpdateTemplateEnum = exports.EmailProviderUpdateNameEnum = exports.EmailProviderCreateNameEnum = exports.DeviceCredentialCreateTypeEnum = exports.DeviceCredentialTypeEnum = exports.CustomDomainTypeEnum = exports.CustomDomainStatusEnum = exports.CreatePhoneTemplateResponseContentTypeEnum = exports.CreatePhoneTemplateRequestContentTypeEnum = exports.CreatePhoneProviderRequestNameEnum = exports.ConnectionUpdateOptionsSetUserRootAttributesEnum = exports.ConnectionUpdateOptionsPasswordPolicyEnum = exports.ConnectionProfileOrganizationShowAsButtonEnum = exports.ConnectionProfileOrganizationAssignMembershipOnLoginEnum = exports.ConnectionCreateOptionsPasskeyOptionsChallengeUiEnum = exports.ConnectionCreateOptionsAttributesEmailSignupStatusEnum = exports.ConnectionCreateOptionsAttributesEmailVerificationMethodEnum = exports.ConnectionCreateOptionsSetUserRootAttributesEnum = exports.ConnectionCreateOptionsPasswordPolicyEnum = void 0;
exports.PatchBruteForceProtectionRequestModeEnum = exports.PatchBruteForceProtectionRequestShieldsEnum = exports.PatchBreachedPasswordDetectionRequestStagePreUserRegistrationShieldsEnum = exports.PatchBreachedPasswordDetectionRequestStagePreChangePasswordShieldsEnum = exports.PatchBreachedPasswordDetectionRequestMethodEnum = exports.PatchBreachedPasswordDetectionRequestAdminNotificationFrequencyEnum = exports.PatchBreachedPasswordDetectionRequestShieldsEnum = exports.PatchBindingsRequestBindingsInnerOneOfRefTypeEnum = exports.PatchAuthenticationMethodsByAuthenticationMethodIdRequestPreferredAuthenticationMethodEnum = exports.OrganizationDiscoveryDomainStatus = exports.JobFormatEnum = exports.HookCreateTriggerIdEnum = exports.GetTokenExchangeProfilesById200ResponseTypeEnum = exports.GetSuspiciousIpThrottling200ResponseShieldsEnum = exports.GetRendering200ResponseFiltersMatchTypeEnum = exports.GetRendering200ResponseRenderingModeEnum = exports.GetPnProviders200ResponseProviderEnum = exports.GetPhoneTemplateResponseContentTypeEnum = exports.GetPhoneProviders200ResponseProviderEnum = exports.GetNetworkAclsById200ResponseRuleAnyOfActionAnyOf3RedirectEnum = exports.GetNetworkAclsById200ResponseRuleAnyOfActionAnyOf2LogEnum = exports.GetNetworkAclsById200ResponseRuleAnyOfActionAnyOf1AllowEnum = exports.GetNetworkAclsById200ResponseRuleAnyOfActionAnyOfBlockEnum = exports.GetNetworkAclsById200ResponseRuleAnyOf1ScopeEnum = exports.GetNetworkAclsById200ResponseRuleAnyOfScopeEnum = exports.GetMessageTypes200ResponseMessageTypesEnum = exports.GetLogStreams200ResponseInnerOneOfSinkHttpContentFormatEnum = exports.GetLogStreams200ResponseInnerOneOfPiiConfigAlgorithmEnum = exports.GetLogStreams200ResponseInnerOneOfPiiConfigMethodEnum = exports.GetLogStreams200ResponseInnerOneOfPiiConfigLogFieldsEnum = exports.GetLogStreams200ResponseInnerOneOfFiltersInnerNameEnum = exports.GetLogStreams200ResponseInnerOneOfFiltersInnerTypeEnum = exports.GetLogStreams200ResponseInnerOneOf7SinkMixpanelRegionEnum = exports.GetLogStreams200ResponseInnerOneOf7TypeEnum = exports.GetLogStreams200ResponseInnerOneOf7StatusEnum = exports.GetLogStreams200ResponseInnerOneOf6TypeEnum = exports.GetLogStreams200ResponseInnerOneOf6StatusEnum = exports.GetLogStreams200ResponseInnerOneOf5TypeEnum = exports.GetLogStreams200ResponseInnerOneOf5StatusEnum = exports.GetLogStreams200ResponseInnerOneOf4TypeEnum = exports.GetLogStreams200ResponseInnerOneOf4StatusEnum = exports.GetLogStreams200ResponseInnerOneOf3SinkDatadogRegionEnum = exports.GetLogStreams200ResponseInnerOneOf3TypeEnum = exports.GetLogStreams200ResponseInnerOneOf3StatusEnum = exports.GetLogStreams200ResponseInnerOneOf2SinkAzureRegionEnum = exports.GetLogStreams200ResponseInnerOneOf2TypeEnum = exports.GetLogStreams200ResponseInnerOneOf2StatusEnum = exports.GetLogStreams200ResponseInnerOneOf1SinkAwsRegionEnum = exports.GetLogStreams200ResponseInnerOneOf1TypeEnum = exports.GetLogStreams200ResponseInnerOneOf1StatusEnum = void 0;
exports.PostFlowsVaultConnectionsRequestAnyOf15AppIdEnum = exports.PostFlowsVaultConnectionsRequestAnyOf14AppIdEnum = exports.PostFlowsVaultConnectionsRequestAnyOf13AppIdEnum = exports.PostFlowsVaultConnectionsRequestAnyOf12SetupAnyOfTypeEnum = exports.PostFlowsVaultConnectionsRequestAnyOf12AppIdEnum = exports.PostFlowsVaultConnectionsRequestAnyOf11SetupTypeEnum = exports.PostFlowsVaultConnectionsRequestAnyOf11AppIdEnum = exports.PostFlowsVaultConnectionsRequestAnyOf10AppIdEnum = exports.PostFlowsVaultConnectionsRequestAnyOf1AppIdEnum = exports.PostFlowsVaultConnectionsRequestAnyOfAppIdEnum = exports.PostEncryptionWrappingKey201ResponseAlgorithmEnum = exports.PostEncryptionRequestTypeEnum = exports.PostEmailTemplatesRequestTemplateEnum = exports.PostCustomDomainsRequestCustomClientIpHeaderEnum = exports.PostCustomDomainsRequestTlsPolicyEnum = exports.PostCustomDomainsRequestVerificationMethodEnum = exports.PostCustomDomainsRequestTypeEnum = exports.PostCustomDomains201ResponseVerificationMethodsInnerNameEnum = exports.PostCustomDomains201ResponseVerificationStatusEnum = exports.PostCustomDomains201ResponseTypeEnum = exports.PostCustomDomains201ResponseStatusEnum = exports.PostCredentialsRequestCredentialTypeEnum = exports.PostConnectionsKeysRotateResponseContentKeyUseEnum = exports.PostBrandingThemeRequestWidgetSocialButtonsLayoutEnum = exports.PostBrandingThemeRequestWidgetLogoPositionEnum = exports.PostBrandingThemeRequestWidgetHeaderTextAlignmentEnum = exports.PostBrandingThemeRequestPageBackgroundPageLayoutEnum = exports.PostBrandingThemeRequestFontsLinksStyleEnum = exports.PostBrandingThemeRequestBordersInputsStyleEnum = exports.PostBrandingThemeRequestBordersButtonsStyleEnum = exports.PostAuthenticationMethodsRequestPreferredAuthenticationMethodEnum = exports.PostAuthenticationMethodsRequestTypeEnum = exports.PostAuthenticationMethods201ResponsePreferredAuthenticationMethodEnum = exports.PostAuthenticationMethods201ResponseTypeEnum = exports.PhoneTemplateTypeEnum = exports.PatchSuspiciousIpThrottlingRequestShieldsEnum = exports.PatchRenderingRequestRenderingModeEnum = exports.PatchRendering200ResponseRenderingModeEnum = exports.PatchNetworkAclsByIdRequestRuleActionRedirectEnum = exports.PatchNetworkAclsByIdRequestRuleActionLogEnum = exports.PatchNetworkAclsByIdRequestRuleActionAllowEnum = exports.PatchNetworkAclsByIdRequestRuleActionBlockEnum = exports.PatchNetworkAclsByIdRequestRuleScopeEnum = exports.PatchLogStreamsByIdRequestSinkOneOf3MixpanelRegionEnum = exports.PatchLogStreamsByIdRequestSinkOneOfDatadogRegionEnum = exports.PatchLogStreamsByIdRequestStatusEnum = exports.PatchEmailTemplatesByTemplateNameRequestTemplateEnum = exports.PatchCustomDomainsByIdRequestCustomClientIpHeaderEnum = exports.PatchCustomDomainsByIdRequestTlsPolicyEnum = exports.PatchClientGrantsByIdRequestOrganizationUsageEnum = void 0;
exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf5TypeEnum = exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf4TypeEnum = exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf3TypeEnum = exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf2TypeEnum = exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf15TypeEnum = exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf14TypeEnum = exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf13TypeEnum = exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf12TypeEnum = exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf11ConfigChargeAnyOfOneOffCurrencyEnum = exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf11ConfigChargeAnyOf1TypeEnum = exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf11ConfigChargeAnyOfTypeEnum = exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf11ConfigProviderEnum = exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf11TypeEnum = exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf10ConfigHashEnum = exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf10TypeEnum = exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf1TypeEnum = exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOfTypeEnum = exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf1AnyOf1TypeEnum = exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf1AnyOfTypeEnum = exports.PostFormsRequestNodesInnerAnyOf2TypeEnum = exports.PostFormsRequestNodesInnerAnyOf1TypeEnum = exports.PostFormsRequestNodesInnerAnyOfTypeEnum = exports.PostFormsRequestEndingResumeFlowEnum = exports.PostFlowsVaultConnectionsRequestAnyOfSetupTypeEnum = exports.PostFlowsVaultConnectionsRequestAnyOf9SetupAlgorithmEnum = exports.PostFlowsVaultConnectionsRequestAnyOf9SetupTypeEnum = exports.PostFlowsVaultConnectionsRequestAnyOf9AppIdEnum = exports.PostFlowsVaultConnectionsRequestAnyOf8AppIdEnum = exports.PostFlowsVaultConnectionsRequestAnyOf7SetupTypeEnum = exports.PostFlowsVaultConnectionsRequestAnyOf7AppIdEnum = exports.PostFlowsVaultConnectionsRequestAnyOf6AppIdEnum = exports.PostFlowsVaultConnectionsRequestAnyOf5SetupTypeEnum = exports.PostFlowsVaultConnectionsRequestAnyOf5AppIdEnum = exports.PostFlowsVaultConnectionsRequestAnyOf4SetupTypeEnum = exports.PostFlowsVaultConnectionsRequestAnyOf4AppIdEnum = exports.PostFlowsVaultConnectionsRequestAnyOf3SetupTypeEnum = exports.PostFlowsVaultConnectionsRequestAnyOf3AppIdEnum = exports.PostFlowsVaultConnectionsRequestAnyOf2SetupTypeEnum = exports.PostFlowsVaultConnectionsRequestAnyOf20AppIdEnum = exports.PostFlowsVaultConnectionsRequestAnyOf2AppIdEnum = exports.PostFlowsVaultConnectionsRequestAnyOf1SetupTypeEnum = exports.PostFlowsVaultConnectionsRequestAnyOf19AppIdEnum = exports.PostFlowsVaultConnectionsRequestAnyOf18SetupTypeEnum = exports.PostFlowsVaultConnectionsRequestAnyOf18AppIdEnum = exports.PostFlowsVaultConnectionsRequestAnyOf17AppIdEnum = exports.PostFlowsVaultConnectionsRequestAnyOf16SetupAnyOf1EnvironmentEnum = exports.PostFlowsVaultConnectionsRequestAnyOf16SetupAnyOf1TypeEnum = exports.PostFlowsVaultConnectionsRequestAnyOf16SetupAnyOfTypeEnum = exports.PostFlowsVaultConnectionsRequestAnyOf16AppIdEnum = exports.PostFlowsVaultConnectionsRequestAnyOf15SetupAnyOfTypeEnum = void 0;
exports.ResourceServerCreateTokenDialectEnum = exports.ResourceServerCreateSigningAlgEnum = exports.ResourceServerConsentPolicyEnum = exports.ResourceServerTokenDialectEnum = exports.ResourceServerSigningAlgEnum = exports.ResetPhoneTemplateResponseContentTypeEnum = exports.PutNetworkAclsByIdRequestRuleAnyOf1ScopeEnum = exports.PutNetworkAclsByIdRequestRuleAnyOfScopeEnum = exports.PutAuthenticationMethodsRequestInnerPreferredAuthenticationMethodEnum = exports.PutAuthenticationMethodsRequestInnerTypeEnum = exports.PutAuthenticationMethods200ResponseInnerAuthenticationMethodsInnerTypeEnum = exports.PutAuthenticationMethods200ResponseInnerPreferredAuthenticationMethodEnum = exports.PutAuthenticationMethods200ResponseInnerTypeEnum = exports.PromptsSettingsUpdateUniversalLoginExperienceEnum = exports.PromptsSettingsUniversalLoginExperienceEnum = exports.PostVerify200ResponseTypeEnum = exports.PostVerify200ResponseStatusEnum = exports.PostVerificationEmailRequestIdentityProviderEnum = exports.PostUsersExportsRequestFormatEnum = exports.PostTokenExchangeProfilesRequestTypeEnum = exports.PostSsoTicketRequestProvisioningConfigScopesEnum = exports.PostSsoTicketRequestDomainAliasesConfigDomainVerificationEnum = exports.PostSsoTicketRequestConnectionConfigOptionsIdpinitiatedClientProtocolEnum = exports.PostLogStreamsRequestOneOfFiltersInnerNameEnum = exports.PostLogStreamsRequestOneOfFiltersInnerTypeEnum = exports.PostLogStreamsRequestOneOf7TypeEnum = exports.PostLogStreamsRequestOneOf6TypeEnum = exports.PostLogStreamsRequestOneOf5TypeEnum = exports.PostLogStreamsRequestOneOf4TypeEnum = exports.PostLogStreamsRequestOneOf3TypeEnum = exports.PostLogStreamsRequestOneOf2SinkAzureRegionEnum = exports.PostLogStreamsRequestOneOf2TypeEnum = exports.PostLogStreamsRequestOneOf1SinkAwsRegionEnum = exports.PostLogStreamsRequestOneOf1TypeEnum = exports.PostLogStreamsRequestOneOfTypeEnum = exports.PostIdentitiesRequestProviderEnum = exports.PostFormsRequestNodesInnerAnyOfConfigNextNodeAnyOf = exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOfAnyOf6TypeEnum = exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOfAnyOf5TypeEnum = exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOfAnyOf4TypeEnum = exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOfAnyOf3TypeEnum = exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOfAnyOf2ConfigPositionEnum = exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOfAnyOf2TypeEnum = exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOfAnyOf1TypeEnum = exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOfAnyOfTypeEnum = exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf9TypeEnum = exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf8TypeEnum = exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf7ConfigCategoriesEnum = exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf7TypeEnum = exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf6TypeEnum = void 0;
exports.PutCustomTextByLanguageLanguageEnum = exports.PutCustomTextByLanguagePromptEnum = exports.PatchRenderingOperationScreenEnum = exports.PatchRenderingOperationPromptEnum = exports.GetRenderingScreenEnum = exports.GetRenderingPromptEnum = exports.GetPartialsPromptEnum = exports.GetCustomTextByLanguageLanguageEnum = exports.GetCustomTextByLanguagePromptEnum = exports.GetAllRenderingRenderingModeEnum = exports.GetHooksTriggerIdEnum = exports.PutFactorsByNameOperationNameEnum = exports.GetFormsByIdHydrateEnum = exports.GetFormsHydrateEnum = exports.GetFlowsExecutionsByExecutionIdHydrateEnum = exports.GetFlowsByIdHydrateEnum = exports.GetFlowsHydrateEnum = exports.PutEmailTemplatesByTemplateNameTemplateNameEnum = exports.PatchEmailTemplatesByTemplateNameOperationTemplateNameEnum = exports.GetEmailTemplatesByTemplateNameTemplateNameEnum = exports.GetDeviceCredentialsTypeEnum = exports.GetConnectionsStrategyEnum = exports.GetClientConnectionsStrategyEnum = exports.GetClientGrantsSubjectTypeEnum = exports.GetClientGrantsAllowAnyOrganizationEnum = exports.UserEnrollmentAuthMethodEnum = exports.UserEnrollmentStatusEnum = exports.UserAttributeProfileUserIdOidcStrategyOverrideMapping = exports.UserAttributeProfileUserIdOidcMappingEnum = exports.UpdatePhoneTemplateResponseContentTypeEnum = exports.UpdatePhoneProviderRequestNameEnum = exports.TenantSettingsUpdateFlagsChangePwdFlowV1Enum = exports.TenantSettingsUpdateDeviceFlowCharsetEnum = exports.TenantSettingsUpdateEnabledLocalesEnum = exports.TenantSettingsSessionCookieModeEnum = exports.TenantSettingsDeviceFlowCharsetEnum = exports.TenantSettingsEnabledLocalesEnum = exports.SsProfileUpdateAllowedStrategiesEnum = exports.SsProfileCreateAllowedStrategiesEnum = exports.SsProfileAllowedStrategiesEnum = exports.SelfServiceProfileSsoTicketProvisioningScopeEnum = exports.ResourceServerUpdateConsentPolicyEnum = exports.ResourceServerUpdateTokenDialectEnum = exports.ResourceServerUpdateSigningAlgEnum = exports.ResourceServerTokenEncryptionEncryptionKeyAlgEnum = exports.ResourceServerTokenEncryptionFormatEnum = exports.ResourceServerSubjectTypeAuthorizationUserPolicyEnum = exports.ResourceServerSubjectTypeAuthorizationClientPolicyEnum = exports.ResourceServerProofOfPossessionMechanismEnum = exports.ResourceServerCreateConsentPolicyEnum = void 0;
exports.GetUsersSearchEngineEnum = exports.DeleteUserIdentityByUserIdProviderEnum = exports.DeleteMultifactorByProviderProviderEnum = exports.PutSelfServiceProfileCustomTextPageEnum = exports.PutSelfServiceProfileCustomTextLanguageEnum = exports.GetSelfServiceProfileCustomTextPageEnum = exports.GetSelfServiceProfileCustomTextLanguageEnum = exports.PutPartialsPromptEnum = void 0;
/**
 *
 */
exports.AssessorsTypeEnum = {
    new_device: 'new-device',
};
/**
 *
 */
exports.AsyncApprovalNotificationsChannelsEnum = {
    guardian_push: 'guardian-push',
    email: 'email',
};
/**
 * The id of the active provider for the CAPTCHA.
 */
exports.AttackProtectionCaptchaProviderId = {
    arkose: 'arkose',
    auth_challenge: 'auth_challenge',
    friendly_captcha: 'friendly_captcha',
    hcaptcha: 'hcaptcha',
    recaptcha_v2: 'recaptcha_v2',
    recaptcha_enterprise: 'recaptcha_enterprise',
    simple_captcha: 'simple_captcha',
};
/**
 * The policy that defines how often to show CAPTCHA
 */
exports.BotDetectionChallengePolicyPasswordFlowEnum = {
    never: 'never',
    when_risky: 'when_risky',
    always: 'always',
};
/**
 * The policy that defines how often to show CAPTCHA
 */
exports.BotDetectionChallengePolicyPasswordResetFlowEnum = {
    never: 'never',
    when_risky: 'when_risky',
    always: 'always',
};
/**
 * The policy that defines how often to show CAPTCHA
 */
exports.BotDetectionChallengePolicyPasswordlessFlowEnum = {
    never: 'never',
    when_risky: 'when_risky',
    always: 'always',
};
/**
 * The level of bot detection sensitivity
 */
exports.BotDetectionLevelEnum = {
    low: 'low',
    medium: 'medium',
    high: 'high',
};
/**
 * The captcha assessment policy for userpass flow enabled for this tenant
 */
exports.CaptchaEnforcementPolicyEnum = {
    high_risk: 'high_risk',
    always_on: 'always_on',
    off: 'off',
};
/**
 * The selected captcha provider.
 */
exports.CaptchaProviderEnum = {
    auth0: 'auth0',
    auth0_v2: 'auth0_v2',
    recaptcha_v2: 'recaptcha_v2',
    recaptcha_enterprise: 'recaptcha_enterprise',
    hcaptcha: 'hcaptcha',
    friendly_captcha: 'friendly_captcha',
    arkose: 'arkose',
};
exports.CertificateStatusEnum = {
    provisioning: 'provisioning',
    provisioning_failed: 'provisioning_failed',
    provisioned: 'provisioned',
    renewing_failed: 'renewing_failed',
};
exports.CertificateCertificateAuthorityEnum = {
    letsencrypt: 'letsencrypt',
    googletrust: 'googletrust',
};
exports.ClientAppTypeEnum = {
    native: 'native',
    spa: 'spa',
    regular_web: 'regular_web',
    non_interactive: 'non_interactive',
    resource_server: 'resource_server',
    express_configuration: 'express_configuration',
    rms: 'rms',
    box: 'box',
    cloudbees: 'cloudbees',
    concur: 'concur',
    dropbox: 'dropbox',
    mscrm: 'mscrm',
    echosign: 'echosign',
    egnyte: 'egnyte',
    newrelic: 'newrelic',
    office365: 'office365',
    salesforce: 'salesforce',
    sentry: 'sentry',
    sharepoint: 'sharepoint',
    slack: 'slack',
    springcm: 'springcm',
    zendesk: 'zendesk',
    zoom: 'zoom',
    sso_integration: 'sso_integration',
    oag: 'oag',
};
exports.ClientTokenEndpointAuthMethodEnum = {
    none: 'none',
    client_secret_post: 'client_secret_post',
    client_secret_basic: 'client_secret_basic',
};
exports.ClientOrganizationUsageEnum = {
    deny: 'deny',
    allow: 'allow',
    require: 'require',
};
exports.ClientOrganizationRequireBehaviorEnum = {
    no_prompt: 'no_prompt',
    pre_login_prompt: 'pre_login_prompt',
    post_login_prompt: 'post_login_prompt',
};
exports.ClientComplianceLevelEnum = {
    none: 'none',
    fapi1_adv_pkj_par: 'fapi1_adv_pkj_par',
    fapi1_adv_mtls_par: 'fapi1_adv_mtls_par',
    null: 'null',
};
exports.ClientCreateTokenEndpointAuthMethodEnum = {
    none: 'none',
    client_secret_post: 'client_secret_post',
    client_secret_basic: 'client_secret_basic',
};
exports.ClientCreateAppTypeEnum = {
    native: 'native',
    spa: 'spa',
    regular_web: 'regular_web',
    non_interactive: 'non_interactive',
    resource_server: 'resource_server',
    rms: 'rms',
    box: 'box',
    cloudbees: 'cloudbees',
    concur: 'concur',
    dropbox: 'dropbox',
    mscrm: 'mscrm',
    echosign: 'echosign',
    egnyte: 'egnyte',
    newrelic: 'newrelic',
    office365: 'office365',
    salesforce: 'salesforce',
    sentry: 'sentry',
    sharepoint: 'sharepoint',
    slack: 'slack',
    springcm: 'springcm',
    zendesk: 'zendesk',
    zoom: 'zoom',
    sso_integration: 'sso_integration',
    oag: 'oag',
};
exports.ClientCreateOrganizationUsageEnum = {
    deny: 'deny',
    allow: 'allow',
    require: 'require',
};
exports.ClientCreateOrganizationRequireBehaviorEnum = {
    no_prompt: 'no_prompt',
    pre_login_prompt: 'pre_login_prompt',
    post_login_prompt: 'post_login_prompt',
};
exports.ClientCreateComplianceLevelEnum = {
    none: 'none',
    fapi1_adv_pkj_par: 'fapi1_adv_pkj_par',
    fapi1_adv_mtls_par: 'fapi1_adv_mtls_par',
    null: 'null',
};
exports.ClientCreateClientAuthenticationMethodsPrivateKeyJwtCredentialsInnerCredentialTypeEnum = {
    public_key: 'public_key',
};
exports.ClientCreateClientAuthenticationMethodsPrivateKeyJwtCredentialsInnerAlgEnum = {
    RS256: 'RS256',
    RS384: 'RS384',
    PS256: 'PS256',
};
exports.ClientCreateClientAuthenticationMethodsSelfSignedTlsClientAuthCredentialsInnerCredentialTypeEnum = {
    x509_cert: 'x509_cert',
};
exports.ClientCreateClientAuthenticationMethodsTlsClientAuthCredentialsInnerCredentialTypeEnum = {
    cert_subject_dn: 'cert_subject_dn',
};
exports.ClientCreateDefaultOrganizationFlowsEnum = {
    client_credentials: 'client_credentials',
};
exports.ClientCreateJwtConfigurationAlgEnum = {
    HS256: 'HS256',
    RS256: 'RS256',
    PS256: 'PS256',
};
exports.ClientDefaultOrganizationFlowsEnum = {
    client_credentials: 'client_credentials',
};
exports.ClientGrantOrganizationUsageEnum = {
    deny: 'deny',
    allow: 'allow',
    require: 'require',
};
exports.ClientGrantSubjectTypeEnum = {
    client: 'client',
    user: 'user',
};
exports.ClientGrantCreateOrganizationUsageEnum = {
    deny: 'deny',
    allow: 'allow',
    require: 'require',
};
exports.ClientGrantCreateSubjectTypeEnum = {
    client: 'client',
    user: 'user',
};
exports.ClientJwtConfigurationAlgEnum = {
    HS256: 'HS256',
    RS256: 'RS256',
    PS256: 'PS256',
};
exports.ClientOidcLogoutBackchannelLogoutInitiatorsModeEnum = {
    custom: 'custom',
    all: 'all',
};
exports.ClientOidcLogoutBackchannelLogoutInitiatorsSelectedInitiatorsEnum = {
    rp_logout: 'rp-logout',
    idp_logout: 'idp-logout',
    password_changed: 'password-changed',
    session_expired: 'session-expired',
    session_revoked: 'session-revoked',
    account_deleted: 'account-deleted',
    email_identifier_changed: 'email-identifier-changed',
    mfa_phone_unenrolled: 'mfa-phone-unenrolled',
    account_deactivated: 'account-deactivated',
};
/**
 * Method for discovering organizations during the `pre_login_prompt`. `email` allows users to find their organization by entering their email address and performing domain matching, while `organization_name` requires users to enter the organization name directly. These methods can be combined.
 */
exports.ClientOrganizationDiscoveryEnum = {
    email: 'email',
    organization_name: 'organization_name',
};
exports.ClientRefreshTokenRotationTypeEnum = {
    rotating: 'rotating',
    non_rotating: 'non-rotating',
};
exports.ClientRefreshTokenExpirationTypeEnum = {
    expiring: 'expiring',
    non_expiring: 'non-expiring',
};
exports.ClientSessionTransferAllowedAuthenticationMethodsEnum = {
    cookie: 'cookie',
    query: 'query',
};
exports.ClientSessionTransferEnforceDeviceBindingEnum = {
    ip: 'ip',
    asn: 'asn',
    none: 'none',
};
exports.ClientUpdateTokenEndpointAuthMethodEnum = {
    none: 'none',
    client_secret_post: 'client_secret_post',
    client_secret_basic: 'client_secret_basic',
    null: 'null',
};
exports.ClientUpdateAppTypeEnum = {
    native: 'native',
    spa: 'spa',
    regular_web: 'regular_web',
    non_interactive: 'non_interactive',
    resource_server: 'resource_server',
    rms: 'rms',
    box: 'box',
    cloudbees: 'cloudbees',
    concur: 'concur',
    dropbox: 'dropbox',
    mscrm: 'mscrm',
    echosign: 'echosign',
    egnyte: 'egnyte',
    newrelic: 'newrelic',
    office365: 'office365',
    salesforce: 'salesforce',
    sentry: 'sentry',
    sharepoint: 'sharepoint',
    slack: 'slack',
    springcm: 'springcm',
    zendesk: 'zendesk',
    zoom: 'zoom',
    sso_integration: 'sso_integration',
    oag: 'oag',
};
exports.ClientUpdateOrganizationUsageEnum = {
    deny: 'deny',
    allow: 'allow',
    require: 'require',
};
exports.ClientUpdateOrganizationRequireBehaviorEnum = {
    no_prompt: 'no_prompt',
    pre_login_prompt: 'pre_login_prompt',
    post_login_prompt: 'post_login_prompt',
};
exports.ClientUpdateComplianceLevelEnum = {
    none: 'none',
    fapi1_adv_pkj_par: 'fapi1_adv_pkj_par',
    fapi1_adv_mtls_par: 'fapi1_adv_mtls_par',
    null: 'null',
};
exports.ClientUpdateJwtConfigurationAlgEnum = {
    HS256: 'HS256',
    RS256: 'RS256',
    PS256: 'PS256',
};
exports.ClientUpdateOidcLogoutBackchannelLogoutInitiatorsModeEnum = {
    custom: 'custom',
    all: 'all',
};
exports.ClientUpdateOidcLogoutBackchannelLogoutInitiatorsSelectedInitiatorsEnum = {
    rp_logout: 'rp-logout',
    idp_logout: 'idp-logout',
    password_changed: 'password-changed',
    session_expired: 'session-expired',
    session_revoked: 'session-revoked',
    account_deleted: 'account-deleted',
    email_identifier_changed: 'email-identifier-changed',
    mfa_phone_unenrolled: 'mfa-phone-unenrolled',
    account_deactivated: 'account-deactivated',
};
/**
 * The access type for the connected account.
 */
exports.ConnectedAccountAccessTypeEnum = {
    offline: 'offline',
};
exports.ConnectionCreateStrategyEnum = {
    ad: 'ad',
    adfs: 'adfs',
    amazon: 'amazon',
    apple: 'apple',
    dropbox: 'dropbox',
    bitbucket: 'bitbucket',
    aol: 'aol',
    auth0_oidc: 'auth0-oidc',
    auth0: 'auth0',
    baidu: 'baidu',
    bitly: 'bitly',
    box: 'box',
    custom: 'custom',
    daccount: 'daccount',
    dwolla: 'dwolla',
    email: 'email',
    evernote_sandbox: 'evernote-sandbox',
    evernote: 'evernote',
    exact: 'exact',
    facebook: 'facebook',
    fitbit: 'fitbit',
    flickr: 'flickr',
    github: 'github',
    google_apps: 'google-apps',
    google_oauth2: 'google-oauth2',
    instagram: 'instagram',
    ip: 'ip',
    line: 'line',
    linkedin: 'linkedin',
    miicard: 'miicard',
    oauth1: 'oauth1',
    oauth2: 'oauth2',
    office365: 'office365',
    oidc: 'oidc',
    okta: 'okta',
    paypal: 'paypal',
    paypal_sandbox: 'paypal-sandbox',
    pingfederate: 'pingfederate',
    planningcenter: 'planningcenter',
    renren: 'renren',
    salesforce_community: 'salesforce-community',
    salesforce_sandbox: 'salesforce-sandbox',
    salesforce: 'salesforce',
    samlp: 'samlp',
    sharepoint: 'sharepoint',
    shopify: 'shopify',
    sms: 'sms',
    soundcloud: 'soundcloud',
    thecity_sandbox: 'thecity-sandbox',
    thecity: 'thecity',
    thirtysevensignals: 'thirtysevensignals',
    twitter: 'twitter',
    untappd: 'untappd',
    vkontakte: 'vkontakte',
    waad: 'waad',
    weibo: 'weibo',
    windowslive: 'windowslive',
    wordpress: 'wordpress',
    yahoo: 'yahoo',
    yammer: 'yammer',
    yandex: 'yandex',
};
exports.ConnectionCreateOptionsPasswordPolicyEnum = {
    none: 'none',
    low: 'low',
    fair: 'fair',
    good: 'good',
    excellent: 'excellent',
    null: 'null',
};
exports.ConnectionCreateOptionsSetUserRootAttributesEnum = {
    each_login: 'on_each_login',
    first_login: 'on_first_login',
};
exports.ConnectionCreateOptionsAttributesEmailVerificationMethodEnum = {
    link: 'link',
    otp: 'otp',
};
exports.ConnectionCreateOptionsAttributesEmailSignupStatusEnum = {
    required: 'required',
    optional: 'optional',
    inactive: 'inactive',
};
exports.ConnectionCreateOptionsPasskeyOptionsChallengeUiEnum = {
    both: 'both',
    autofill: 'autofill',
    button: 'button',
};
/**
 * Indicates if membership should be assigned on login.
 */
exports.ConnectionProfileOrganizationAssignMembershipOnLoginEnum = {
    none: 'none',
    optional: 'optional',
    required: 'required',
};
/**
 * Indicates if the organization should be shown as a button.
 */
exports.ConnectionProfileOrganizationShowAsButtonEnum = {
    none: 'none',
    optional: 'optional',
    required: 'required',
};
exports.ConnectionUpdateOptionsPasswordPolicyEnum = {
    none: 'none',
    low: 'low',
    fair: 'fair',
    good: 'good',
    excellent: 'excellent',
    null: 'null',
};
exports.ConnectionUpdateOptionsSetUserRootAttributesEnum = {
    each_login: 'on_each_login',
    first_login: 'on_first_login',
};
exports.CreatePhoneProviderRequestNameEnum = {
    twilio: 'twilio',
    custom: 'custom',
};
exports.CreatePhoneTemplateRequestContentTypeEnum = {
    otp_verify: 'otp_verify',
    otp_enroll: 'otp_enroll',
    change_password: 'change_password',
    blocked_account: 'blocked_account',
    password_breach: 'password_breach',
};
exports.CreatePhoneTemplateResponseContentTypeEnum = {
    otp_verify: 'otp_verify',
    otp_enroll: 'otp_enroll',
    change_password: 'change_password',
    blocked_account: 'blocked_account',
    password_breach: 'password_breach',
};
exports.CustomDomainStatusEnum = {
    disabled: 'disabled',
    pending: 'pending',
    pending_verification: 'pending_verification',
    ready: 'ready',
};
exports.CustomDomainTypeEnum = {
    auth0_managed_certs: 'auth0_managed_certs',
    self_managed_certs: 'self_managed_certs',
};
exports.DeviceCredentialTypeEnum = {
    public_key: 'public_key',
    refresh_token: 'refresh_token',
    rotating_refresh_token: 'rotating_refresh_token',
};
exports.DeviceCredentialCreateTypeEnum = {
    public_key: 'public_key',
};
exports.EmailProviderCreateNameEnum = {
    mailgun: 'mailgun',
    mandrill: 'mandrill',
    sendgrid: 'sendgrid',
    ses: 'ses',
    sparkpost: 'sparkpost',
    smtp: 'smtp',
    azure_cs: 'azure_cs',
    ms365: 'ms365',
    custom: 'custom',
};
exports.EmailProviderUpdateNameEnum = {
    mailgun: 'mailgun',
    mandrill: 'mandrill',
    sendgrid: 'sendgrid',
    ses: 'ses',
    sparkpost: 'sparkpost',
    smtp: 'smtp',
    azure_cs: 'azure_cs',
    ms365: 'ms365',
    custom: 'custom',
};
exports.EmailTemplateUpdateTemplateEnum = {
    verify_email: 'verify_email',
    verify_email_by_code: 'verify_email_by_code',
    reset_email: 'reset_email',
    reset_email_by_code: 'reset_email_by_code',
    welcome_email: 'welcome_email',
    blocked_account: 'blocked_account',
    stolen_credentials: 'stolen_credentials',
    enrollment_email: 'enrollment_email',
    mfa_oob_code: 'mfa_oob_code',
    user_invitation: 'user_invitation',
    change_password: 'change_password',
    password_reset: 'password_reset',
    async_approval: 'async_approval',
};
/**
 * Enum for enabled features.
 */
exports.EnabledFeaturesEnum = {
    scim: 'scim',
    universal_logout: 'universal_logout',
};
exports.EnrollmentStatusEnum = {
    pending: 'pending',
    confirmed: 'confirmed',
};
exports.FactorNameEnum = {
    push_notification: 'push-notification',
    sms: 'sms',
    email: 'email',
    duo: 'duo',
    otp: 'otp',
    webauthn_roaming: 'webauthn-roaming',
    webauthn_platform: 'webauthn-platform',
    recovery_code: 'recovery-code',
};
exports.GetActionVersions200ResponseVersionsInnerStatusEnum = {
    pending: 'pending',
    building: 'building',
    packaged: 'packaged',
    built: 'built',
    retrying: 'retrying',
    failed: 'failed',
};
exports.GetActions200ResponseActionsInnerStatusEnum = {
    pending: 'pending',
    building: 'building',
    packaged: 'packaged',
    built: 'built',
    retrying: 'retrying',
    failed: 'failed',
};
exports.GetActions200ResponseActionsInnerIntegrationFeatureTypeEnum = {
    unspecified: 'unspecified',
    action: 'action',
    social_connection: 'social_connection',
    log_stream: 'log_stream',
    sso_integration: 'sso_integration',
    sms_provider: 'sms_provider',
};
exports.GetActions200ResponseActionsInnerIntegrationCurrentReleaseRequiredSecretsInnerTypeEnum = {
    UNSPECIFIED: 'UNSPECIFIED',
    STRING: 'STRING',
};
/**
 *
 */
exports.GetActions200ResponseActionsInnerSupportedTriggersInnerIdAnyOf = {
    post_login: 'post-login',
    credentials_exchange: 'credentials-exchange',
    pre_user_registration: 'pre-user-registration',
    post_user_registration: 'post-user-registration',
    post_change_password: 'post-change-password',
    send_phone_message: 'send-phone-message',
    iga_approval: 'iga-approval',
    iga_certification: 'iga-certification',
    iga_fulfillment_assignment: 'iga-fulfillment-assignment',
    iga_fulfillment_execution: 'iga-fulfillment-execution',
    custom_email_provider: 'custom-email-provider',
    password_reset_post_challenge: 'password-reset-post-challenge',
};
exports.GetAllRendering200ResponseOneOfInnerRenderingModeEnum = {
    advanced: 'advanced',
    standard: 'standard',
};
exports.GetAuthenticationMethods200ResponseOneOfInnerTypeEnum = {
    recovery_code: 'recovery-code',
    totp: 'totp',
    push: 'push',
    phone: 'phone',
    email: 'email',
    email_verification: 'email-verification',
    webauthn_roaming: 'webauthn-roaming',
    webauthn_platform: 'webauthn-platform',
    guardian: 'guardian',
    passkey: 'passkey',
};
exports.GetAuthenticationMethods200ResponseOneOfInnerPreferredAuthenticationMethodEnum = {
    sms: 'sms',
    voice: 'voice',
};
exports.GetAuthenticationMethods200ResponseOneOfInnerAuthenticationMethodsInnerTypeEnum = {
    totp: 'totp',
    push: 'push',
    sms: 'sms',
    voice: 'voice',
};
exports.GetBrandingPhoneProviders200ResponseProvidersInnerNameEnum = {
    twilio: 'twilio',
    custom: 'custom',
};
exports.GetBrandingPhoneProviders200ResponseProvidersInnerChannelEnum = {
    phone: 'phone',
};
exports.GetBrandingPhoneProviders200ResponseProvidersInnerConfigurationAnyOfDeliveryMethodsEnum = {
    text: 'text',
    voice: 'voice',
};
exports.GetBrandingPhoneProviders200ResponseProvidersInnerConfigurationAnyOf1DeliveryMethodsEnum = {
    text: 'text',
    voice: 'voice',
};
exports.GetBreachedPasswordDetection200ResponseShieldsEnum = {
    block: 'block',
    user_notification: 'user_notification',
    admin_notification: 'admin_notification',
};
exports.GetBreachedPasswordDetection200ResponseAdminNotificationFrequencyEnum = {
    immediately: 'immediately',
    daily: 'daily',
    weekly: 'weekly',
    monthly: 'monthly',
};
exports.GetBreachedPasswordDetection200ResponseMethodEnum = {
    standard: 'standard',
    enhanced: 'enhanced',
};
exports.GetBreachedPasswordDetection200ResponseStagePreChangePasswordShieldsEnum = {
    block: 'block',
    admin_notification: 'admin_notification',
};
exports.GetBreachedPasswordDetection200ResponseStagePreUserRegistrationShieldsEnum = {
    block: 'block',
    admin_notification: 'admin_notification',
};
exports.GetBruteForceProtection200ResponseShieldsEnum = {
    block: 'block',
    user_notification: 'user_notification',
};
exports.GetBruteForceProtection200ResponseModeEnum = {
    identifier_and_ip: 'count_per_identifier_and_ip',
    identifier: 'count_per_identifier',
};
exports.GetConnectionsKeysResponseContentInnerKeyUseEnum = {
    encryption: 'encryption',
    signing: 'signing',
};
exports.GetCredentials200ResponseInnerAlgEnum = {
    RS256: 'RS256',
    RS384: 'RS384',
    PS256: 'PS256',
};
exports.GetEmailTemplatesByTemplateName200ResponseTemplateEnum = {
    verify_email: 'verify_email',
    verify_email_by_code: 'verify_email_by_code',
    reset_email: 'reset_email',
    reset_email_by_code: 'reset_email_by_code',
    welcome_email: 'welcome_email',
    blocked_account: 'blocked_account',
    stolen_credentials: 'stolen_credentials',
    enrollment_email: 'enrollment_email',
    mfa_oob_code: 'mfa_oob_code',
    user_invitation: 'user_invitation',
    change_password: 'change_password',
    password_reset: 'password_reset',
    async_approval: 'async_approval',
};
exports.GetEncryptionKeys200ResponseOneOfInnerTypeEnum = {
    customer_provided_root_key: 'customer-provided-root-key',
    environment_root_key: 'environment-root-key',
    tenant_master_key: 'tenant-master-key',
    tenant_encryption_key: 'tenant-encryption-key',
};
exports.GetEncryptionKeys200ResponseOneOfInnerStateEnum = {
    pre_activation: 'pre-activation',
    active: 'active',
    deactivated: 'deactivated',
    destroyed: 'destroyed',
};
exports.GetExecution200ResponseStatusEnum = {
    unspecified: 'unspecified',
    pending: 'pending',
    final: 'final',
    partial: 'partial',
    canceled: 'canceled',
    suspended: 'suspended',
};
exports.GetLogStreams200ResponseInnerOneOfStatusEnum = {
    active: 'active',
    paused: 'paused',
    suspended: 'suspended',
};
exports.GetLogStreams200ResponseInnerOneOfTypeEnum = {
    http: 'http',
};
exports.GetLogStreams200ResponseInnerOneOf1StatusEnum = {
    active: 'active',
    paused: 'paused',
    suspended: 'suspended',
};
exports.GetLogStreams200ResponseInnerOneOf1TypeEnum = {
    eventbridge: 'eventbridge',
};
exports.GetLogStreams200ResponseInnerOneOf1SinkAwsRegionEnum = {
    ap_east_1: 'ap-east-1',
    ap_northeast_1: 'ap-northeast-1',
    ap_northeast_2: 'ap-northeast-2',
    ap_northeast_3: 'ap-northeast-3',
    ap_south_1: 'ap-south-1',
    ap_southeast_1: 'ap-southeast-1',
    ap_southeast_2: 'ap-southeast-2',
    ca_central_1: 'ca-central-1',
    cn_north_1: 'cn-north-1',
    cn_northwest_1: 'cn-northwest-1',
    eu_central_1: 'eu-central-1',
    eu_north_1: 'eu-north-1',
    eu_west_1: 'eu-west-1',
    eu_west_2: 'eu-west-2',
    eu_west_3: 'eu-west-3',
    me_south_1: 'me-south-1',
    sa_east_1: 'sa-east-1',
    us_gov_east_1: 'us-gov-east-1',
    us_gov_west_1: 'us-gov-west-1',
    us_east_1: 'us-east-1',
    us_east_2: 'us-east-2',
    us_west_1: 'us-west-1',
    us_west_2: 'us-west-2',
};
exports.GetLogStreams200ResponseInnerOneOf2StatusEnum = {
    active: 'active',
    paused: 'paused',
    suspended: 'suspended',
};
exports.GetLogStreams200ResponseInnerOneOf2TypeEnum = {
    eventgrid: 'eventgrid',
};
exports.GetLogStreams200ResponseInnerOneOf2SinkAzureRegionEnum = {
    australiacentral: 'australiacentral',
    australiaeast: 'australiaeast',
    australiasoutheast: 'australiasoutheast',
    brazilsouth: 'brazilsouth',
    canadacentral: 'canadacentral',
    canadaeast: 'canadaeast',
    centralindia: 'centralindia',
    centralus: 'centralus',
    eastasia: 'eastasia',
    eastus: 'eastus',
    eastus2: 'eastus2',
    francecentral: 'francecentral',
    germanywestcentral: 'germanywestcentral',
    japaneast: 'japaneast',
    japanwest: 'japanwest',
    koreacentral: 'koreacentral',
    koreasouth: 'koreasouth',
    northcentralus: 'northcentralus',
    northeurope: 'northeurope',
    norwayeast: 'norwayeast',
    southafricanorth: 'southafricanorth',
    southcentralus: 'southcentralus',
    southeastasia: 'southeastasia',
    southindia: 'southindia',
    switzerlandnorth: 'switzerlandnorth',
    uaenorth: 'uaenorth',
    uksouth: 'uksouth',
    ukwest: 'ukwest',
    westcentralus: 'westcentralus',
    westeurope: 'westeurope',
    westindia: 'westindia',
    westus: 'westus',
    westus2: 'westus2',
};
exports.GetLogStreams200ResponseInnerOneOf3StatusEnum = {
    active: 'active',
    paused: 'paused',
    suspended: 'suspended',
};
exports.GetLogStreams200ResponseInnerOneOf3TypeEnum = {
    datadog: 'datadog',
};
exports.GetLogStreams200ResponseInnerOneOf3SinkDatadogRegionEnum = {
    us: 'us',
    eu: 'eu',
    us3: 'us3',
    us5: 'us5',
};
exports.GetLogStreams200ResponseInnerOneOf4StatusEnum = {
    active: 'active',
    paused: 'paused',
    suspended: 'suspended',
};
exports.GetLogStreams200ResponseInnerOneOf4TypeEnum = {
    splunk: 'splunk',
};
exports.GetLogStreams200ResponseInnerOneOf5StatusEnum = {
    active: 'active',
    paused: 'paused',
    suspended: 'suspended',
};
exports.GetLogStreams200ResponseInnerOneOf5TypeEnum = {
    sumo: 'sumo',
};
exports.GetLogStreams200ResponseInnerOneOf6StatusEnum = {
    active: 'active',
    paused: 'paused',
    suspended: 'suspended',
};
exports.GetLogStreams200ResponseInnerOneOf6TypeEnum = {
    segment: 'segment',
};
exports.GetLogStreams200ResponseInnerOneOf7StatusEnum = {
    active: 'active',
    paused: 'paused',
    suspended: 'suspended',
};
exports.GetLogStreams200ResponseInnerOneOf7TypeEnum = {
    mixpanel: 'mixpanel',
};
exports.GetLogStreams200ResponseInnerOneOf7SinkMixpanelRegionEnum = {
    us: 'us',
    eu: 'eu',
};
exports.GetLogStreams200ResponseInnerOneOfFiltersInnerTypeEnum = {
    category: 'category',
};
exports.GetLogStreams200ResponseInnerOneOfFiltersInnerNameEnum = {
    auth_ancillary_fail: 'auth.ancillary.fail',
    auth_ancillary_success: 'auth.ancillary.success',
    auth_login_fail: 'auth.login.fail',
    auth_login_notification: 'auth.login.notification',
    auth_login_success: 'auth.login.success',
    auth_logout_fail: 'auth.logout.fail',
    auth_logout_success: 'auth.logout.success',
    auth_signup_fail: 'auth.signup.fail',
    auth_signup_success: 'auth.signup.success',
    auth_silent_auth_fail: 'auth.silent_auth.fail',
    auth_silent_auth_success: 'auth.silent_auth.success',
    auth_token_exchange_fail: 'auth.token_exchange.fail',
    auth_token_exchange_success: 'auth.token_exchange.success',
    management_fail: 'management.fail',
    management_success: 'management.success',
    system_notification: 'system.notification',
    user_fail: 'user.fail',
    user_notification: 'user.notification',
    user_success: 'user.success',
    other: 'other',
};
exports.GetLogStreams200ResponseInnerOneOfPiiConfigLogFieldsEnum = {
    first_name: 'first_name',
    last_name: 'last_name',
    username: 'username',
    email: 'email',
    phone: 'phone',
    address: 'address',
};
exports.GetLogStreams200ResponseInnerOneOfPiiConfigMethodEnum = {
    mask: 'mask',
    hash: 'hash',
};
exports.GetLogStreams200ResponseInnerOneOfPiiConfigAlgorithmEnum = {
    xxhash: 'xxhash',
};
exports.GetLogStreams200ResponseInnerOneOfSinkHttpContentFormatEnum = {
    JSONARRAY: 'JSONARRAY',
    JSONLINES: 'JSONLINES',
    JSONOBJECT: 'JSONOBJECT',
};
exports.GetMessageTypes200ResponseMessageTypesEnum = {
    sms: 'sms',
    voice: 'voice',
};
exports.GetNetworkAclsById200ResponseRuleAnyOfScopeEnum = {
    management: 'management',
    authentication: 'authentication',
    tenant: 'tenant',
};
exports.GetNetworkAclsById200ResponseRuleAnyOf1ScopeEnum = {
    management: 'management',
    authentication: 'authentication',
    tenant: 'tenant',
};
exports.GetNetworkAclsById200ResponseRuleAnyOfActionAnyOfBlockEnum = {
    true: true,
};
exports.GetNetworkAclsById200ResponseRuleAnyOfActionAnyOf1AllowEnum = {
    true: true,
};
exports.GetNetworkAclsById200ResponseRuleAnyOfActionAnyOf2LogEnum = {
    true: true,
};
exports.GetNetworkAclsById200ResponseRuleAnyOfActionAnyOf3RedirectEnum = {
    true: true,
};
exports.GetPhoneProviders200ResponseProviderEnum = {
    auth0: 'auth0',
    twilio: 'twilio',
    phone_message_hook: 'phone-message-hook',
};
exports.GetPhoneTemplateResponseContentTypeEnum = {
    otp_verify: 'otp_verify',
    otp_enroll: 'otp_enroll',
    change_password: 'change_password',
    blocked_account: 'blocked_account',
    password_breach: 'password_breach',
};
exports.GetPnProviders200ResponseProviderEnum = {
    guardian: 'guardian',
    sns: 'sns',
    direct: 'direct',
};
exports.GetRendering200ResponseRenderingModeEnum = {
    advanced: 'advanced',
    standard: 'standard',
};
exports.GetRendering200ResponseFiltersMatchTypeEnum = {
    includes_any: 'includes_any',
    excludes_any: 'excludes_any',
};
exports.GetSuspiciousIpThrottling200ResponseShieldsEnum = {
    block: 'block',
    admin_notification: 'admin_notification',
};
exports.GetTokenExchangeProfilesById200ResponseTypeEnum = {
    custom_authentication: 'custom_authentication',
};
exports.HookCreateTriggerIdEnum = {
    credentials_exchange: 'credentials-exchange',
    pre_user_registration: 'pre-user-registration',
    post_user_registration: 'post-user-registration',
    post_change_password: 'post-change-password',
    send_phone_message: 'send-phone-message',
};
exports.JobFormatEnum = {
    json: 'json',
    csv: 'csv',
};
/**
 * The verification status of the discovery domain.
 */
exports.OrganizationDiscoveryDomainStatus = {
    pending: 'pending',
    verified: 'verified',
};
exports.PatchAuthenticationMethodsByAuthenticationMethodIdRequestPreferredAuthenticationMethodEnum = {
    voice: 'voice',
    sms: 'sms',
};
exports.PatchBindingsRequestBindingsInnerOneOfRefTypeEnum = {
    binding_id: 'binding_id',
    action_id: 'action_id',
    action_name: 'action_name',
};
exports.PatchBreachedPasswordDetectionRequestShieldsEnum = {
    block: 'block',
    user_notification: 'user_notification',
    admin_notification: 'admin_notification',
};
exports.PatchBreachedPasswordDetectionRequestAdminNotificationFrequencyEnum = {
    immediately: 'immediately',
    daily: 'daily',
    weekly: 'weekly',
    monthly: 'monthly',
};
exports.PatchBreachedPasswordDetectionRequestMethodEnum = {
    standard: 'standard',
    enhanced: 'enhanced',
};
exports.PatchBreachedPasswordDetectionRequestStagePreChangePasswordShieldsEnum = {
    block: 'block',
    admin_notification: 'admin_notification',
};
exports.PatchBreachedPasswordDetectionRequestStagePreUserRegistrationShieldsEnum = {
    block: 'block',
    admin_notification: 'admin_notification',
};
exports.PatchBruteForceProtectionRequestShieldsEnum = {
    block: 'block',
    user_notification: 'user_notification',
};
exports.PatchBruteForceProtectionRequestModeEnum = {
    identifier_and_ip: 'count_per_identifier_and_ip',
    identifier: 'count_per_identifier',
};
exports.PatchClientGrantsByIdRequestOrganizationUsageEnum = {
    deny: 'deny',
    allow: 'allow',
    require: 'require',
    null: 'null',
};
exports.PatchCustomDomainsByIdRequestTlsPolicyEnum = {
    recommended: 'recommended',
};
exports.PatchCustomDomainsByIdRequestCustomClientIpHeaderEnum = {
    true_client_ip: 'true-client-ip',
    cf_connecting_ip: 'cf-connecting-ip',
    x_forwarded_for: 'x-forwarded-for',
    x_azure_clientip: 'x-azure-clientip',
    empty: '',
};
exports.PatchEmailTemplatesByTemplateNameRequestTemplateEnum = {
    verify_email: 'verify_email',
    verify_email_by_code: 'verify_email_by_code',
    reset_email: 'reset_email',
    reset_email_by_code: 'reset_email_by_code',
    welcome_email: 'welcome_email',
    blocked_account: 'blocked_account',
    stolen_credentials: 'stolen_credentials',
    enrollment_email: 'enrollment_email',
    mfa_oob_code: 'mfa_oob_code',
    user_invitation: 'user_invitation',
    change_password: 'change_password',
    password_reset: 'password_reset',
    async_approval: 'async_approval',
};
exports.PatchLogStreamsByIdRequestStatusEnum = {
    active: 'active',
    paused: 'paused',
    suspended: 'suspended',
};
exports.PatchLogStreamsByIdRequestSinkOneOfDatadogRegionEnum = {
    us: 'us',
    eu: 'eu',
    us3: 'us3',
    us5: 'us5',
};
exports.PatchLogStreamsByIdRequestSinkOneOf3MixpanelRegionEnum = {
    us: 'us',
    eu: 'eu',
};
exports.PatchNetworkAclsByIdRequestRuleScopeEnum = {
    management: 'management',
    authentication: 'authentication',
    tenant: 'tenant',
};
exports.PatchNetworkAclsByIdRequestRuleActionBlockEnum = {
    true: true,
};
exports.PatchNetworkAclsByIdRequestRuleActionAllowEnum = {
    true: true,
};
exports.PatchNetworkAclsByIdRequestRuleActionLogEnum = {
    true: true,
};
exports.PatchNetworkAclsByIdRequestRuleActionRedirectEnum = {
    true: true,
};
exports.PatchRendering200ResponseRenderingModeEnum = {
    advanced: 'advanced',
    standard: 'standard',
};
exports.PatchRenderingRequestRenderingModeEnum = {
    advanced: 'advanced',
    standard: 'standard',
};
exports.PatchSuspiciousIpThrottlingRequestShieldsEnum = {
    block: 'block',
    admin_notification: 'admin_notification',
};
exports.PhoneTemplateTypeEnum = {
    otp_verify: 'otp_verify',
    otp_enroll: 'otp_enroll',
    change_password: 'change_password',
    blocked_account: 'blocked_account',
    password_breach: 'password_breach',
};
exports.PostAuthenticationMethods201ResponseTypeEnum = {
    phone: 'phone',
    email: 'email',
    totp: 'totp',
    webauthn_roaming: 'webauthn-roaming',
};
exports.PostAuthenticationMethods201ResponsePreferredAuthenticationMethodEnum = {
    voice: 'voice',
    sms: 'sms',
};
exports.PostAuthenticationMethodsRequestTypeEnum = {
    phone: 'phone',
    email: 'email',
    totp: 'totp',
    webauthn_roaming: 'webauthn-roaming',
};
exports.PostAuthenticationMethodsRequestPreferredAuthenticationMethodEnum = {
    voice: 'voice',
    sms: 'sms',
};
exports.PostBrandingThemeRequestBordersButtonsStyleEnum = {
    pill: 'pill',
    rounded: 'rounded',
    sharp: 'sharp',
};
exports.PostBrandingThemeRequestBordersInputsStyleEnum = {
    pill: 'pill',
    rounded: 'rounded',
    sharp: 'sharp',
};
exports.PostBrandingThemeRequestFontsLinksStyleEnum = {
    normal: 'normal',
    underlined: 'underlined',
};
exports.PostBrandingThemeRequestPageBackgroundPageLayoutEnum = {
    center: 'center',
    left: 'left',
    right: 'right',
};
exports.PostBrandingThemeRequestWidgetHeaderTextAlignmentEnum = {
    center: 'center',
    left: 'left',
    right: 'right',
};
exports.PostBrandingThemeRequestWidgetLogoPositionEnum = {
    center: 'center',
    left: 'left',
    none: 'none',
    right: 'right',
};
exports.PostBrandingThemeRequestWidgetSocialButtonsLayoutEnum = {
    bottom: 'bottom',
    top: 'top',
};
exports.PostConnectionsKeysRotateResponseContentKeyUseEnum = {
    encryption: 'encryption',
    signing: 'signing',
};
exports.PostCredentialsRequestCredentialTypeEnum = {
    public_key: 'public_key',
    cert_subject_dn: 'cert_subject_dn',
    x509_cert: 'x509_cert',
};
exports.PostCustomDomains201ResponseStatusEnum = {
    disabled: 'disabled',
    pending: 'pending',
    pending_verification: 'pending_verification',
    ready: 'ready',
};
exports.PostCustomDomains201ResponseTypeEnum = {
    auth0_managed_certs: 'auth0_managed_certs',
    self_managed_certs: 'self_managed_certs',
};
exports.PostCustomDomains201ResponseVerificationStatusEnum = {
    verified: 'verified',
    pending: 'pending',
    failed: 'failed',
};
exports.PostCustomDomains201ResponseVerificationMethodsInnerNameEnum = {
    cname: 'cname',
    txt: 'txt',
};
exports.PostCustomDomainsRequestTypeEnum = {
    auth0_managed_certs: 'auth0_managed_certs',
    self_managed_certs: 'self_managed_certs',
};
exports.PostCustomDomainsRequestVerificationMethodEnum = {
    txt: 'txt',
};
exports.PostCustomDomainsRequestTlsPolicyEnum = {
    recommended: 'recommended',
};
exports.PostCustomDomainsRequestCustomClientIpHeaderEnum = {
    true_client_ip: 'true-client-ip',
    cf_connecting_ip: 'cf-connecting-ip',
    x_forwarded_for: 'x-forwarded-for',
    x_azure_clientip: 'x-azure-clientip',
    null: 'null',
};
exports.PostEmailTemplatesRequestTemplateEnum = {
    verify_email: 'verify_email',
    verify_email_by_code: 'verify_email_by_code',
    reset_email: 'reset_email',
    reset_email_by_code: 'reset_email_by_code',
    welcome_email: 'welcome_email',
    blocked_account: 'blocked_account',
    stolen_credentials: 'stolen_credentials',
    enrollment_email: 'enrollment_email',
    mfa_oob_code: 'mfa_oob_code',
    user_invitation: 'user_invitation',
    change_password: 'change_password',
    password_reset: 'password_reset',
    async_approval: 'async_approval',
};
exports.PostEncryptionRequestTypeEnum = {
    customer_provided_root_key: 'customer-provided-root-key',
    tenant_encryption_key: 'tenant-encryption-key',
};
exports.PostEncryptionWrappingKey201ResponseAlgorithmEnum = {
    CKM_RSA_AES_KEY_WRAP: 'CKM_RSA_AES_KEY_WRAP',
};
exports.PostFlowsVaultConnectionsRequestAnyOfAppIdEnum = {
    ACTIVECAMPAIGN: 'ACTIVECAMPAIGN',
};
exports.PostFlowsVaultConnectionsRequestAnyOf1AppIdEnum = {
    AIRTABLE: 'AIRTABLE',
};
exports.PostFlowsVaultConnectionsRequestAnyOf10AppIdEnum = {
    MAILCHIMP: 'MAILCHIMP',
};
exports.PostFlowsVaultConnectionsRequestAnyOf11AppIdEnum = {
    MAILJET: 'MAILJET',
};
exports.PostFlowsVaultConnectionsRequestAnyOf11SetupTypeEnum = {
    API_KEY: 'API_KEY',
};
exports.PostFlowsVaultConnectionsRequestAnyOf12AppIdEnum = {
    PIPEDRIVE: 'PIPEDRIVE',
};
exports.PostFlowsVaultConnectionsRequestAnyOf12SetupAnyOfTypeEnum = {
    TOKEN: 'TOKEN',
};
exports.PostFlowsVaultConnectionsRequestAnyOf13AppIdEnum = {
    SALESFORCE: 'SALESFORCE',
};
exports.PostFlowsVaultConnectionsRequestAnyOf14AppIdEnum = {
    SENDGRID: 'SENDGRID',
};
exports.PostFlowsVaultConnectionsRequestAnyOf15AppIdEnum = {
    SLACK: 'SLACK',
};
exports.PostFlowsVaultConnectionsRequestAnyOf15SetupAnyOfTypeEnum = {
    WEBHOOK: 'WEBHOOK',
};
exports.PostFlowsVaultConnectionsRequestAnyOf16AppIdEnum = {
    STRIPE: 'STRIPE',
};
exports.PostFlowsVaultConnectionsRequestAnyOf16SetupAnyOfTypeEnum = {
    KEY_PAIR: 'KEY_PAIR',
};
exports.PostFlowsVaultConnectionsRequestAnyOf16SetupAnyOf1TypeEnum = {
    OAUTH_CODE: 'OAUTH_CODE',
};
exports.PostFlowsVaultConnectionsRequestAnyOf16SetupAnyOf1EnvironmentEnum = {
    live: 'live',
    test: 'test',
};
exports.PostFlowsVaultConnectionsRequestAnyOf17AppIdEnum = {
    TELEGRAM: 'TELEGRAM',
};
exports.PostFlowsVaultConnectionsRequestAnyOf18AppIdEnum = {
    TWILIO: 'TWILIO',
};
exports.PostFlowsVaultConnectionsRequestAnyOf18SetupTypeEnum = {
    API_KEY: 'API_KEY',
};
exports.PostFlowsVaultConnectionsRequestAnyOf19AppIdEnum = {
    WHATSAPP: 'WHATSAPP',
};
exports.PostFlowsVaultConnectionsRequestAnyOf1SetupTypeEnum = {
    API_KEY: 'API_KEY',
};
exports.PostFlowsVaultConnectionsRequestAnyOf2AppIdEnum = {
    AUTH0: 'AUTH0',
};
exports.PostFlowsVaultConnectionsRequestAnyOf20AppIdEnum = {
    ZAPIER: 'ZAPIER',
};
exports.PostFlowsVaultConnectionsRequestAnyOf2SetupTypeEnum = {
    OAUTH_APP: 'OAUTH_APP',
};
exports.PostFlowsVaultConnectionsRequestAnyOf3AppIdEnum = {
    BIGQUERY: 'BIGQUERY',
};
exports.PostFlowsVaultConnectionsRequestAnyOf3SetupTypeEnum = {
    OAUTH_JWT: 'OAUTH_JWT',
};
exports.PostFlowsVaultConnectionsRequestAnyOf4AppIdEnum = {
    CLEARBIT: 'CLEARBIT',
};
exports.PostFlowsVaultConnectionsRequestAnyOf4SetupTypeEnum = {
    API_KEY: 'API_KEY',
};
exports.PostFlowsVaultConnectionsRequestAnyOf5AppIdEnum = {
    DOCUSIGN: 'DOCUSIGN',
};
exports.PostFlowsVaultConnectionsRequestAnyOf5SetupTypeEnum = {
    OAUTH_CODE: 'OAUTH_CODE',
};
exports.PostFlowsVaultConnectionsRequestAnyOf6AppIdEnum = {
    GOOGLE_SHEETS: 'GOOGLE_SHEETS',
};
exports.PostFlowsVaultConnectionsRequestAnyOf7AppIdEnum = {
    HTTP: 'HTTP',
};
exports.PostFlowsVaultConnectionsRequestAnyOf7SetupTypeEnum = {
    BEARER: 'BEARER',
};
exports.PostFlowsVaultConnectionsRequestAnyOf8AppIdEnum = {
    HUBSPOT: 'HUBSPOT',
};
exports.PostFlowsVaultConnectionsRequestAnyOf9AppIdEnum = {
    JWT: 'JWT',
};
exports.PostFlowsVaultConnectionsRequestAnyOf9SetupTypeEnum = {
    JWT: 'JWT',
};
exports.PostFlowsVaultConnectionsRequestAnyOf9SetupAlgorithmEnum = {
    HS256: 'HS256',
    HS384: 'HS384',
    HS512: 'HS512',
    RS256: 'RS256',
    RS384: 'RS384',
    RS512: 'RS512',
    ES256: 'ES256',
    ES384: 'ES384',
    ES512: 'ES512',
    PS256: 'PS256',
    PS384: 'PS384',
    PS512: 'PS512',
};
exports.PostFlowsVaultConnectionsRequestAnyOfSetupTypeEnum = {
    API_KEY: 'API_KEY',
};
exports.PostFormsRequestEndingResumeFlowEnum = {
    true: true,
};
exports.PostFormsRequestNodesInnerAnyOfTypeEnum = {
    FLOW: 'FLOW',
};
exports.PostFormsRequestNodesInnerAnyOf1TypeEnum = {
    ROUTER: 'ROUTER',
};
exports.PostFormsRequestNodesInnerAnyOf2TypeEnum = {
    STEP: 'STEP',
};
exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf1AnyOfTypeEnum = {
    GMAPS_ADDRESS: 'GMAPS_ADDRESS',
};
exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf1AnyOf1TypeEnum = {
    RECAPTCHA: 'RECAPTCHA',
};
exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOfTypeEnum = {
    BOOLEAN: 'BOOLEAN',
};
exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf1TypeEnum = {
    CARDS: 'CARDS',
};
exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf10TypeEnum = {
    PASSWORD: 'PASSWORD',
};
exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf10ConfigHashEnum = {
    NONE: 'NONE',
    MD5: 'MD5',
    SHA1: 'SHA1',
    SHA256: 'SHA256',
    SHA512: 'SHA512',
};
exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf11TypeEnum = {
    PAYMENT: 'PAYMENT',
};
exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf11ConfigProviderEnum = {
    STRIPE: 'STRIPE',
};
exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf11ConfigChargeAnyOfTypeEnum = {
    ONE_OFF: 'ONE_OFF',
};
exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf11ConfigChargeAnyOf1TypeEnum = {
    SUBSCRIPTION: 'SUBSCRIPTION',
};
exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf11ConfigChargeAnyOfOneOffCurrencyEnum = {
    AUD: 'AUD',
    CAD: 'CAD',
    CHF: 'CHF',
    EUR: 'EUR',
    GBP: 'GBP',
    INR: 'INR',
    MXN: 'MXN',
    SEK: 'SEK',
    USD: 'USD',
};
exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf12TypeEnum = {
    SOCIAL: 'SOCIAL',
};
exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf13TypeEnum = {
    TEL: 'TEL',
};
exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf14TypeEnum = {
    TEXT: 'TEXT',
};
exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf15TypeEnum = {
    URL: 'URL',
};
exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf2TypeEnum = {
    CHOICE: 'CHOICE',
};
exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf3TypeEnum = {
    CUSTOM: 'CUSTOM',
};
exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf4TypeEnum = {
    DATE: 'DATE',
};
exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf5TypeEnum = {
    DROPDOWN: 'DROPDOWN',
};
exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf6TypeEnum = {
    EMAIL: 'EMAIL',
};
exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf7TypeEnum = {
    FILE: 'FILE',
};
exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf7ConfigCategoriesEnum = {
    AUDIO: 'AUDIO',
    VIDEO: 'VIDEO',
    IMAGE: 'IMAGE',
    DOCUMENT: 'DOCUMENT',
    ARCHIVE: 'ARCHIVE',
};
exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf8TypeEnum = {
    LEGAL: 'LEGAL',
};
exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf9TypeEnum = {
    NUMBER: 'NUMBER',
};
exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOfAnyOfTypeEnum = {
    DIVIDER: 'DIVIDER',
};
exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOfAnyOf1TypeEnum = {
    HTML: 'HTML',
};
exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOfAnyOf2TypeEnum = {
    IMAGE: 'IMAGE',
};
exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOfAnyOf2ConfigPositionEnum = {
    LEFT: 'LEFT',
    CENTER: 'CENTER',
    RIGHT: 'RIGHT',
};
exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOfAnyOf3TypeEnum = {
    JUMP_BUTTON: 'JUMP_BUTTON',
};
exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOfAnyOf4TypeEnum = {
    NEXT_BUTTON: 'NEXT_BUTTON',
};
exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOfAnyOf5TypeEnum = {
    PREVIOUS_BUTTON: 'PREVIOUS_BUTTON',
};
exports.PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOfAnyOf6TypeEnum = {
    RICH_TEXT: 'RICH_TEXT',
};
/**
 *
 */
exports.PostFormsRequestNodesInnerAnyOfConfigNextNodeAnyOf = {
    ending: '$ending',
};
exports.PostIdentitiesRequestProviderEnum = {
    ad: 'ad',
    adfs: 'adfs',
    amazon: 'amazon',
    apple: 'apple',
    dropbox: 'dropbox',
    bitbucket: 'bitbucket',
    aol: 'aol',
    auth0_oidc: 'auth0-oidc',
    auth0: 'auth0',
    baidu: 'baidu',
    bitly: 'bitly',
    box: 'box',
    custom: 'custom',
    daccount: 'daccount',
    dwolla: 'dwolla',
    email: 'email',
    evernote_sandbox: 'evernote-sandbox',
    evernote: 'evernote',
    exact: 'exact',
    facebook: 'facebook',
    fitbit: 'fitbit',
    flickr: 'flickr',
    github: 'github',
    google_apps: 'google-apps',
    google_oauth2: 'google-oauth2',
    instagram: 'instagram',
    ip: 'ip',
    line: 'line',
    linkedin: 'linkedin',
    miicard: 'miicard',
    oauth1: 'oauth1',
    oauth2: 'oauth2',
    office365: 'office365',
    oidc: 'oidc',
    okta: 'okta',
    paypal: 'paypal',
    paypal_sandbox: 'paypal-sandbox',
    pingfederate: 'pingfederate',
    planningcenter: 'planningcenter',
    renren: 'renren',
    salesforce_community: 'salesforce-community',
    salesforce_sandbox: 'salesforce-sandbox',
    salesforce: 'salesforce',
    samlp: 'samlp',
    sharepoint: 'sharepoint',
    shopify: 'shopify',
    sms: 'sms',
    soundcloud: 'soundcloud',
    thecity_sandbox: 'thecity-sandbox',
    thecity: 'thecity',
    thirtysevensignals: 'thirtysevensignals',
    twitter: 'twitter',
    untappd: 'untappd',
    vkontakte: 'vkontakte',
    waad: 'waad',
    weibo: 'weibo',
    windowslive: 'windowslive',
    wordpress: 'wordpress',
    yahoo: 'yahoo',
    yammer: 'yammer',
    yandex: 'yandex',
};
exports.PostLogStreamsRequestOneOfTypeEnum = {
    http: 'http',
};
exports.PostLogStreamsRequestOneOf1TypeEnum = {
    eventbridge: 'eventbridge',
};
exports.PostLogStreamsRequestOneOf1SinkAwsRegionEnum = {
    ap_east_1: 'ap-east-1',
    ap_northeast_1: 'ap-northeast-1',
    ap_northeast_2: 'ap-northeast-2',
    ap_northeast_3: 'ap-northeast-3',
    ap_south_1: 'ap-south-1',
    ap_southeast_1: 'ap-southeast-1',
    ap_southeast_2: 'ap-southeast-2',
    ca_central_1: 'ca-central-1',
    cn_north_1: 'cn-north-1',
    cn_northwest_1: 'cn-northwest-1',
    eu_central_1: 'eu-central-1',
    eu_north_1: 'eu-north-1',
    eu_west_1: 'eu-west-1',
    eu_west_2: 'eu-west-2',
    eu_west_3: 'eu-west-3',
    me_south_1: 'me-south-1',
    sa_east_1: 'sa-east-1',
    us_gov_east_1: 'us-gov-east-1',
    us_gov_west_1: 'us-gov-west-1',
    us_east_1: 'us-east-1',
    us_east_2: 'us-east-2',
    us_west_1: 'us-west-1',
    us_west_2: 'us-west-2',
};
exports.PostLogStreamsRequestOneOf2TypeEnum = {
    eventgrid: 'eventgrid',
};
exports.PostLogStreamsRequestOneOf2SinkAzureRegionEnum = {
    australiacentral: 'australiacentral',
    australiaeast: 'australiaeast',
    australiasoutheast: 'australiasoutheast',
    brazilsouth: 'brazilsouth',
    canadacentral: 'canadacentral',
    canadaeast: 'canadaeast',
    centralindia: 'centralindia',
    centralus: 'centralus',
    eastasia: 'eastasia',
    eastus: 'eastus',
    eastus2: 'eastus2',
    francecentral: 'francecentral',
    germanywestcentral: 'germanywestcentral',
    japaneast: 'japaneast',
    japanwest: 'japanwest',
    koreacentral: 'koreacentral',
    koreasouth: 'koreasouth',
    northcentralus: 'northcentralus',
    northeurope: 'northeurope',
    norwayeast: 'norwayeast',
    southafricanorth: 'southafricanorth',
    southcentralus: 'southcentralus',
    southeastasia: 'southeastasia',
    southindia: 'southindia',
    switzerlandnorth: 'switzerlandnorth',
    uaenorth: 'uaenorth',
    uksouth: 'uksouth',
    ukwest: 'ukwest',
    westcentralus: 'westcentralus',
    westeurope: 'westeurope',
    westindia: 'westindia',
    westus: 'westus',
    westus2: 'westus2',
};
exports.PostLogStreamsRequestOneOf3TypeEnum = {
    datadog: 'datadog',
};
exports.PostLogStreamsRequestOneOf4TypeEnum = {
    splunk: 'splunk',
};
exports.PostLogStreamsRequestOneOf5TypeEnum = {
    sumo: 'sumo',
};
exports.PostLogStreamsRequestOneOf6TypeEnum = {
    segment: 'segment',
};
exports.PostLogStreamsRequestOneOf7TypeEnum = {
    mixpanel: 'mixpanel',
};
exports.PostLogStreamsRequestOneOfFiltersInnerTypeEnum = {
    category: 'category',
};
exports.PostLogStreamsRequestOneOfFiltersInnerNameEnum = {
    auth_ancillary_fail: 'auth.ancillary.fail',
    auth_ancillary_success: 'auth.ancillary.success',
    auth_login_fail: 'auth.login.fail',
    auth_login_notification: 'auth.login.notification',
    auth_login_success: 'auth.login.success',
    auth_logout_fail: 'auth.logout.fail',
    auth_logout_success: 'auth.logout.success',
    auth_signup_fail: 'auth.signup.fail',
    auth_signup_success: 'auth.signup.success',
    auth_silent_auth_fail: 'auth.silent_auth.fail',
    auth_silent_auth_success: 'auth.silent_auth.success',
    auth_token_exchange_fail: 'auth.token_exchange.fail',
    auth_token_exchange_success: 'auth.token_exchange.success',
    management_fail: 'management.fail',
    management_success: 'management.success',
    system_notification: 'system.notification',
    user_fail: 'user.fail',
    user_notification: 'user.notification',
    user_success: 'user.success',
    other: 'other',
};
exports.PostSsoTicketRequestConnectionConfigOptionsIdpinitiatedClientProtocolEnum = {
    samlp: 'samlp',
    wsfed: 'wsfed',
    oauth2: 'oauth2',
};
exports.PostSsoTicketRequestDomainAliasesConfigDomainVerificationEnum = {
    none: 'none',
    optional: 'optional',
    required: 'required',
};
exports.PostSsoTicketRequestProvisioningConfigScopesEnum = {
    getusers: 'get:users',
    postusers: 'post:users',
    putusers: 'put:users',
    patchusers: 'patch:users',
    deleteusers: 'delete:users',
};
exports.PostTokenExchangeProfilesRequestTypeEnum = {
    custom_authentication: 'custom_authentication',
};
exports.PostUsersExportsRequestFormatEnum = {
    json: 'json',
    csv: 'csv',
};
exports.PostVerificationEmailRequestIdentityProviderEnum = {
    ad: 'ad',
    adfs: 'adfs',
    amazon: 'amazon',
    apple: 'apple',
    dropbox: 'dropbox',
    bitbucket: 'bitbucket',
    aol: 'aol',
    auth0_oidc: 'auth0-oidc',
    auth0: 'auth0',
    baidu: 'baidu',
    bitly: 'bitly',
    box: 'box',
    custom: 'custom',
    daccount: 'daccount',
    dwolla: 'dwolla',
    email: 'email',
    evernote_sandbox: 'evernote-sandbox',
    evernote: 'evernote',
    exact: 'exact',
    facebook: 'facebook',
    fitbit: 'fitbit',
    flickr: 'flickr',
    github: 'github',
    google_apps: 'google-apps',
    google_oauth2: 'google-oauth2',
    instagram: 'instagram',
    ip: 'ip',
    line: 'line',
    linkedin: 'linkedin',
    miicard: 'miicard',
    oauth1: 'oauth1',
    oauth2: 'oauth2',
    office365: 'office365',
    oidc: 'oidc',
    okta: 'okta',
    paypal: 'paypal',
    paypal_sandbox: 'paypal-sandbox',
    pingfederate: 'pingfederate',
    planningcenter: 'planningcenter',
    renren: 'renren',
    salesforce_community: 'salesforce-community',
    salesforce_sandbox: 'salesforce-sandbox',
    salesforce: 'salesforce',
    samlp: 'samlp',
    sharepoint: 'sharepoint',
    shopify: 'shopify',
    sms: 'sms',
    soundcloud: 'soundcloud',
    thecity_sandbox: 'thecity-sandbox',
    thecity: 'thecity',
    thirtysevensignals: 'thirtysevensignals',
    twitter: 'twitter',
    untappd: 'untappd',
    vkontakte: 'vkontakte',
    waad: 'waad',
    weibo: 'weibo',
    windowslive: 'windowslive',
    wordpress: 'wordpress',
    yahoo: 'yahoo',
    yammer: 'yammer',
    yandex: 'yandex',
};
exports.PostVerify200ResponseStatusEnum = {
    disabled: 'disabled',
    pending: 'pending',
    pending_verification: 'pending_verification',
    ready: 'ready',
};
exports.PostVerify200ResponseTypeEnum = {
    auth0_managed_certs: 'auth0_managed_certs',
    self_managed_certs: 'self_managed_certs',
};
exports.PromptsSettingsUniversalLoginExperienceEnum = {
    new: 'new',
    classic: 'classic',
};
exports.PromptsSettingsUpdateUniversalLoginExperienceEnum = {
    new: 'new',
    classic: 'classic',
};
exports.PutAuthenticationMethods200ResponseInnerTypeEnum = {
    phone: 'phone',
    email: 'email',
    totp: 'totp',
    webauthn_roaming: 'webauthn-roaming',
};
exports.PutAuthenticationMethods200ResponseInnerPreferredAuthenticationMethodEnum = {
    voice: 'voice',
    sms: 'sms',
};
exports.PutAuthenticationMethods200ResponseInnerAuthenticationMethodsInnerTypeEnum = {
    totp: 'totp',
    push: 'push',
    sms: 'sms',
    voice: 'voice',
};
exports.PutAuthenticationMethodsRequestInnerTypeEnum = {
    phone: 'phone',
    email: 'email',
    totp: 'totp',
};
exports.PutAuthenticationMethodsRequestInnerPreferredAuthenticationMethodEnum = {
    voice: 'voice',
    sms: 'sms',
};
exports.PutNetworkAclsByIdRequestRuleAnyOfScopeEnum = {
    management: 'management',
    authentication: 'authentication',
    tenant: 'tenant',
};
exports.PutNetworkAclsByIdRequestRuleAnyOf1ScopeEnum = {
    management: 'management',
    authentication: 'authentication',
    tenant: 'tenant',
};
exports.ResetPhoneTemplateResponseContentTypeEnum = {
    otp_verify: 'otp_verify',
    otp_enroll: 'otp_enroll',
    change_password: 'change_password',
    blocked_account: 'blocked_account',
    password_breach: 'password_breach',
};
exports.ResourceServerSigningAlgEnum = {
    HS256: 'HS256',
    RS256: 'RS256',
    PS256: 'PS256',
};
exports.ResourceServerTokenDialectEnum = {
    access_token: 'access_token',
    access_token_authz: 'access_token_authz',
    rfc9068_profile: 'rfc9068_profile',
    rfc9068_profile_authz: 'rfc9068_profile_authz',
};
exports.ResourceServerConsentPolicyEnum = {
    transactional_authorization_with_mfa: 'transactional-authorization-with-mfa',
    null: 'null',
};
exports.ResourceServerCreateSigningAlgEnum = {
    HS256: 'HS256',
    RS256: 'RS256',
    PS256: 'PS256',
};
exports.ResourceServerCreateTokenDialectEnum = {
    access_token: 'access_token',
    access_token_authz: 'access_token_authz',
    rfc9068_profile: 'rfc9068_profile',
    rfc9068_profile_authz: 'rfc9068_profile_authz',
};
exports.ResourceServerCreateConsentPolicyEnum = {
    transactional_authorization_with_mfa: 'transactional-authorization-with-mfa',
    null: 'null',
};
exports.ResourceServerProofOfPossessionMechanismEnum = {
    mtls: 'mtls',
    dpop: 'dpop',
};
exports.ResourceServerSubjectTypeAuthorizationClientPolicyEnum = {
    deny_all: 'deny_all',
    require_client_grant: 'require_client_grant',
};
exports.ResourceServerSubjectTypeAuthorizationUserPolicyEnum = {
    allow_all: 'allow_all',
    deny_all: 'deny_all',
    require_client_grant: 'require_client_grant',
};
exports.ResourceServerTokenEncryptionFormatEnum = {
    compact_nested_jwe: 'compact-nested-jwe',
};
exports.ResourceServerTokenEncryptionEncryptionKeyAlgEnum = {
    _256: 'RSA-OAEP-256',
    _384: 'RSA-OAEP-384',
    _512: 'RSA-OAEP-512',
};
exports.ResourceServerUpdateSigningAlgEnum = {
    HS256: 'HS256',
    RS256: 'RS256',
    PS256: 'PS256',
};
exports.ResourceServerUpdateTokenDialectEnum = {
    access_token: 'access_token',
    access_token_authz: 'access_token_authz',
    rfc9068_profile: 'rfc9068_profile',
    rfc9068_profile_authz: 'rfc9068_profile_authz',
};
exports.ResourceServerUpdateConsentPolicyEnum = {
    transactional_authorization_with_mfa: 'transactional-authorization-with-mfa',
    null: 'null',
};
/**
 *
 */
exports.SelfServiceProfileSsoTicketProvisioningScopeEnum = {
    getusers: 'get:users',
    postusers: 'post:users',
    putusers: 'put:users',
    patchusers: 'patch:users',
    deleteusers: 'delete:users',
};
exports.SsProfileAllowedStrategiesEnum = {
    oidc: 'oidc',
    samlp: 'samlp',
    waad: 'waad',
    google_apps: 'google-apps',
    adfs: 'adfs',
    okta: 'okta',
    keycloak_samlp: 'keycloak-samlp',
    pingfederate: 'pingfederate',
};
exports.SsProfileCreateAllowedStrategiesEnum = {
    oidc: 'oidc',
    samlp: 'samlp',
    waad: 'waad',
    google_apps: 'google-apps',
    adfs: 'adfs',
    okta: 'okta',
    keycloak_samlp: 'keycloak-samlp',
    pingfederate: 'pingfederate',
};
exports.SsProfileUpdateAllowedStrategiesEnum = {
    oidc: 'oidc',
    samlp: 'samlp',
    waad: 'waad',
    google_apps: 'google-apps',
    adfs: 'adfs',
    okta: 'okta',
    keycloak_samlp: 'keycloak-samlp',
    pingfederate: 'pingfederate',
};
exports.TenantSettingsEnabledLocalesEnum = {
    am: 'am',
    ar: 'ar',
    ar_EG: 'ar-EG',
    ar_SA: 'ar-SA',
    az: 'az',
    bg: 'bg',
    bn: 'bn',
    bs: 'bs',
    ca_ES: 'ca-ES',
    cnr: 'cnr',
    cs: 'cs',
    cy: 'cy',
    da: 'da',
    de: 'de',
    el: 'el',
    en: 'en',
    en_CA: 'en-CA',
    es: 'es',
    es_419: 'es-419',
    es_AR: 'es-AR',
    es_MX: 'es-MX',
    et: 'et',
    eu_ES: 'eu-ES',
    fa: 'fa',
    fi: 'fi',
    fr: 'fr',
    fr_CA: 'fr-CA',
    fr_FR: 'fr-FR',
    gl_ES: 'gl-ES',
    gu: 'gu',
    he: 'he',
    hi: 'hi',
    hr: 'hr',
    hu: 'hu',
    hy: 'hy',
    id: 'id',
    is: 'is',
    it: 'it',
    ja: 'ja',
    ka: 'ka',
    kk: 'kk',
    kn: 'kn',
    ko: 'ko',
    lt: 'lt',
    lv: 'lv',
    mk: 'mk',
    ml: 'ml',
    mn: 'mn',
    mr: 'mr',
    ms: 'ms',
    my: 'my',
    nb: 'nb',
    nl: 'nl',
    nn: 'nn',
    no: 'no',
    pa: 'pa',
    pl: 'pl',
    pt: 'pt',
    pt_BR: 'pt-BR',
    pt_PT: 'pt-PT',
    ro: 'ro',
    ru: 'ru',
    sk: 'sk',
    sl: 'sl',
    so: 'so',
    sq: 'sq',
    sr: 'sr',
    sv: 'sv',
    sw: 'sw',
    ta: 'ta',
    te: 'te',
    th: 'th',
    tl: 'tl',
    tr: 'tr',
    uk: 'uk',
    ur: 'ur',
    vi: 'vi',
    zgh: 'zgh',
    zh_CN: 'zh-CN',
    zh_HK: 'zh-HK',
    zh_TW: 'zh-TW',
};
exports.TenantSettingsDeviceFlowCharsetEnum = {
    base20: 'base20',
    digits: 'digits',
};
exports.TenantSettingsSessionCookieModeEnum = {
    persistent: 'persistent',
    non_persistent: 'non-persistent',
};
exports.TenantSettingsUpdateEnabledLocalesEnum = {
    am: 'am',
    ar: 'ar',
    ar_EG: 'ar-EG',
    ar_SA: 'ar-SA',
    az: 'az',
    bg: 'bg',
    bn: 'bn',
    bs: 'bs',
    ca_ES: 'ca-ES',
    cnr: 'cnr',
    cs: 'cs',
    cy: 'cy',
    da: 'da',
    de: 'de',
    el: 'el',
    en: 'en',
    en_CA: 'en-CA',
    es: 'es',
    es_419: 'es-419',
    es_AR: 'es-AR',
    es_MX: 'es-MX',
    et: 'et',
    eu_ES: 'eu-ES',
    fa: 'fa',
    fi: 'fi',
    fr: 'fr',
    fr_CA: 'fr-CA',
    fr_FR: 'fr-FR',
    gl_ES: 'gl-ES',
    gu: 'gu',
    he: 'he',
    hi: 'hi',
    hr: 'hr',
    hu: 'hu',
    hy: 'hy',
    id: 'id',
    is: 'is',
    it: 'it',
    ja: 'ja',
    ka: 'ka',
    kk: 'kk',
    kn: 'kn',
    ko: 'ko',
    lt: 'lt',
    lv: 'lv',
    mk: 'mk',
    ml: 'ml',
    mn: 'mn',
    mr: 'mr',
    ms: 'ms',
    my: 'my',
    nb: 'nb',
    nl: 'nl',
    nn: 'nn',
    no: 'no',
    pa: 'pa',
    pl: 'pl',
    pt: 'pt',
    pt_BR: 'pt-BR',
    pt_PT: 'pt-PT',
    ro: 'ro',
    ru: 'ru',
    sk: 'sk',
    sl: 'sl',
    so: 'so',
    sq: 'sq',
    sr: 'sr',
    sv: 'sv',
    sw: 'sw',
    ta: 'ta',
    te: 'te',
    th: 'th',
    tl: 'tl',
    tr: 'tr',
    uk: 'uk',
    ur: 'ur',
    vi: 'vi',
    zgh: 'zgh',
    zh_CN: 'zh-CN',
    zh_HK: 'zh-HK',
    zh_TW: 'zh-TW',
};
exports.TenantSettingsUpdateDeviceFlowCharsetEnum = {
    base20: 'base20',
    digits: 'digits',
};
exports.TenantSettingsUpdateFlagsChangePwdFlowV1Enum = {
    false: false,
};
exports.UpdatePhoneProviderRequestNameEnum = {
    twilio: 'twilio',
    custom: 'custom',
};
exports.UpdatePhoneTemplateResponseContentTypeEnum = {
    otp_verify: 'otp_verify',
    otp_enroll: 'otp_enroll',
    change_password: 'change_password',
    blocked_account: 'blocked_account',
    password_breach: 'password_breach',
};
/**
 * OIDC mapping for user ID
 */
exports.UserAttributeProfileUserIdOidcMappingEnum = {
    sub: 'sub',
};
/**
 * OIDC mapping override for this strategy
 */
exports.UserAttributeProfileUserIdOidcStrategyOverrideMapping = {
    sub: 'sub',
    oid: 'oid',
    email: 'email',
};
exports.UserEnrollmentStatusEnum = {
    pending: 'pending',
    confirmed: 'confirmed',
};
exports.UserEnrollmentAuthMethodEnum = {
    authenticator: 'authenticator',
    guardian: 'guardian',
    sms: 'sms',
    webauthn_platform: 'webauthn-platform',
    webauthn_roaming: 'webauthn-roaming',
};
/**
 *
 */
exports.GetClientGrantsAllowAnyOrganizationEnum = {
    true: true,
};
/**
 *
 */
exports.GetClientGrantsSubjectTypeEnum = {
    client: 'client',
    user: 'user',
};
/**
 *
 */
exports.GetClientConnectionsStrategyEnum = {
    ad: 'ad',
    adfs: 'adfs',
    amazon: 'amazon',
    apple: 'apple',
    dropbox: 'dropbox',
    bitbucket: 'bitbucket',
    aol: 'aol',
    auth0_oidc: 'auth0-oidc',
    auth0: 'auth0',
    baidu: 'baidu',
    bitly: 'bitly',
    box: 'box',
    custom: 'custom',
    daccount: 'daccount',
    dwolla: 'dwolla',
    email: 'email',
    evernote_sandbox: 'evernote-sandbox',
    evernote: 'evernote',
    exact: 'exact',
    facebook: 'facebook',
    fitbit: 'fitbit',
    flickr: 'flickr',
    github: 'github',
    google_apps: 'google-apps',
    google_oauth2: 'google-oauth2',
    instagram: 'instagram',
    ip: 'ip',
    line: 'line',
    linkedin: 'linkedin',
    miicard: 'miicard',
    oauth1: 'oauth1',
    oauth2: 'oauth2',
    office365: 'office365',
    oidc: 'oidc',
    okta: 'okta',
    paypal: 'paypal',
    paypal_sandbox: 'paypal-sandbox',
    pingfederate: 'pingfederate',
    planningcenter: 'planningcenter',
    renren: 'renren',
    salesforce_community: 'salesforce-community',
    salesforce_sandbox: 'salesforce-sandbox',
    salesforce: 'salesforce',
    samlp: 'samlp',
    sharepoint: 'sharepoint',
    shopify: 'shopify',
    sms: 'sms',
    soundcloud: 'soundcloud',
    thecity_sandbox: 'thecity-sandbox',
    thecity: 'thecity',
    thirtysevensignals: 'thirtysevensignals',
    twitter: 'twitter',
    untappd: 'untappd',
    vkontakte: 'vkontakte',
    waad: 'waad',
    weibo: 'weibo',
    windowslive: 'windowslive',
    wordpress: 'wordpress',
    yahoo: 'yahoo',
    yammer: 'yammer',
    yandex: 'yandex',
};
/**
 *
 */
exports.GetConnectionsStrategyEnum = {
    ad: 'ad',
    adfs: 'adfs',
    amazon: 'amazon',
    apple: 'apple',
    dropbox: 'dropbox',
    bitbucket: 'bitbucket',
    aol: 'aol',
    auth0_oidc: 'auth0-oidc',
    auth0: 'auth0',
    baidu: 'baidu',
    bitly: 'bitly',
    box: 'box',
    custom: 'custom',
    daccount: 'daccount',
    dwolla: 'dwolla',
    email: 'email',
    evernote_sandbox: 'evernote-sandbox',
    evernote: 'evernote',
    exact: 'exact',
    facebook: 'facebook',
    fitbit: 'fitbit',
    flickr: 'flickr',
    github: 'github',
    google_apps: 'google-apps',
    google_oauth2: 'google-oauth2',
    instagram: 'instagram',
    ip: 'ip',
    line: 'line',
    linkedin: 'linkedin',
    miicard: 'miicard',
    oauth1: 'oauth1',
    oauth2: 'oauth2',
    office365: 'office365',
    oidc: 'oidc',
    okta: 'okta',
    paypal: 'paypal',
    paypal_sandbox: 'paypal-sandbox',
    pingfederate: 'pingfederate',
    planningcenter: 'planningcenter',
    renren: 'renren',
    salesforce_community: 'salesforce-community',
    salesforce_sandbox: 'salesforce-sandbox',
    salesforce: 'salesforce',
    samlp: 'samlp',
    sharepoint: 'sharepoint',
    shopify: 'shopify',
    sms: 'sms',
    soundcloud: 'soundcloud',
    thecity_sandbox: 'thecity-sandbox',
    thecity: 'thecity',
    thirtysevensignals: 'thirtysevensignals',
    twitter: 'twitter',
    untappd: 'untappd',
    vkontakte: 'vkontakte',
    waad: 'waad',
    weibo: 'weibo',
    windowslive: 'windowslive',
    wordpress: 'wordpress',
    yahoo: 'yahoo',
    yammer: 'yammer',
    yandex: 'yandex',
    auth0_adldap: 'auth0-adldap',
};
/**
 *
 */
exports.GetDeviceCredentialsTypeEnum = {
    public_key: 'public_key',
    refresh_token: 'refresh_token',
    rotating_refresh_token: 'rotating_refresh_token',
};
/**
 *
 */
exports.GetEmailTemplatesByTemplateNameTemplateNameEnum = {
    verify_email: 'verify_email',
    verify_email_by_code: 'verify_email_by_code',
    reset_email: 'reset_email',
    reset_email_by_code: 'reset_email_by_code',
    welcome_email: 'welcome_email',
    blocked_account: 'blocked_account',
    stolen_credentials: 'stolen_credentials',
    enrollment_email: 'enrollment_email',
    mfa_oob_code: 'mfa_oob_code',
    user_invitation: 'user_invitation',
    change_password: 'change_password',
    password_reset: 'password_reset',
    async_approval: 'async_approval',
};
/**
 *
 */
exports.PatchEmailTemplatesByTemplateNameOperationTemplateNameEnum = {
    verify_email: 'verify_email',
    verify_email_by_code: 'verify_email_by_code',
    reset_email: 'reset_email',
    reset_email_by_code: 'reset_email_by_code',
    welcome_email: 'welcome_email',
    blocked_account: 'blocked_account',
    stolen_credentials: 'stolen_credentials',
    enrollment_email: 'enrollment_email',
    mfa_oob_code: 'mfa_oob_code',
    user_invitation: 'user_invitation',
    change_password: 'change_password',
    password_reset: 'password_reset',
    async_approval: 'async_approval',
};
/**
 *
 */
exports.PutEmailTemplatesByTemplateNameTemplateNameEnum = {
    verify_email: 'verify_email',
    verify_email_by_code: 'verify_email_by_code',
    reset_email: 'reset_email',
    reset_email_by_code: 'reset_email_by_code',
    welcome_email: 'welcome_email',
    blocked_account: 'blocked_account',
    stolen_credentials: 'stolen_credentials',
    enrollment_email: 'enrollment_email',
    mfa_oob_code: 'mfa_oob_code',
    user_invitation: 'user_invitation',
    change_password: 'change_password',
    password_reset: 'password_reset',
    async_approval: 'async_approval',
};
/**
 *
 */
exports.GetFlowsHydrateEnum = {
    form_count: 'form_count',
};
/**
 *
 */
exports.GetFlowsByIdHydrateEnum = {
    form_count: 'form_count',
};
/**
 *
 */
exports.GetFlowsExecutionsByExecutionIdHydrateEnum = {
    debug: 'debug',
};
/**
 *
 */
exports.GetFormsHydrateEnum = {
    flow_count: 'flow_count',
    links: 'links',
};
/**
 *
 */
exports.GetFormsByIdHydrateEnum = {
    flow_count: 'flow_count',
    links: 'links',
};
/**
 *
 */
exports.PutFactorsByNameOperationNameEnum = {
    push_notification: 'push-notification',
    sms: 'sms',
    email: 'email',
    duo: 'duo',
    otp: 'otp',
    webauthn_roaming: 'webauthn-roaming',
    webauthn_platform: 'webauthn-platform',
    recovery_code: 'recovery-code',
};
/**
 *
 */
exports.GetHooksTriggerIdEnum = {
    credentials_exchange: 'credentials-exchange',
    pre_user_registration: 'pre-user-registration',
    post_user_registration: 'post-user-registration',
    post_change_password: 'post-change-password',
    send_phone_message: 'send-phone-message',
};
/**
 *
 */
exports.GetAllRenderingRenderingModeEnum = {
    advanced: 'advanced',
    standard: 'standard',
};
/**
 *
 */
exports.GetCustomTextByLanguagePromptEnum = {
    login: 'login',
    login_id: 'login-id',
    login_password: 'login-password',
    login_passwordless: 'login-passwordless',
    login_email_verification: 'login-email-verification',
    signup: 'signup',
    signup_id: 'signup-id',
    signup_password: 'signup-password',
    phone_identifier_enrollment: 'phone-identifier-enrollment',
    phone_identifier_challenge: 'phone-identifier-challenge',
    email_identifier_challenge: 'email-identifier-challenge',
    reset_password: 'reset-password',
    custom_form: 'custom-form',
    consent: 'consent',
    customized_consent: 'customized-consent',
    logout: 'logout',
    mfa_push: 'mfa-push',
    mfa_otp: 'mfa-otp',
    mfa_voice: 'mfa-voice',
    mfa_phone: 'mfa-phone',
    mfa_webauthn: 'mfa-webauthn',
    mfa_sms: 'mfa-sms',
    mfa_email: 'mfa-email',
    mfa_recovery_code: 'mfa-recovery-code',
    mfa: 'mfa',
    status: 'status',
    device_flow: 'device-flow',
    email_verification: 'email-verification',
    email_otp_challenge: 'email-otp-challenge',
    organizations: 'organizations',
    invitation: 'invitation',
    common: 'common',
    passkeys: 'passkeys',
    captcha: 'captcha',
};
/**
 *
 */
exports.GetCustomTextByLanguageLanguageEnum = {
    am: 'am',
    ar: 'ar',
    ar_EG: 'ar-EG',
    ar_SA: 'ar-SA',
    az: 'az',
    bg: 'bg',
    bn: 'bn',
    bs: 'bs',
    ca_ES: 'ca-ES',
    cnr: 'cnr',
    cs: 'cs',
    cy: 'cy',
    da: 'da',
    de: 'de',
    el: 'el',
    en: 'en',
    en_CA: 'en-CA',
    es: 'es',
    es_419: 'es-419',
    es_AR: 'es-AR',
    es_MX: 'es-MX',
    et: 'et',
    eu_ES: 'eu-ES',
    fa: 'fa',
    fi: 'fi',
    fr: 'fr',
    fr_CA: 'fr-CA',
    fr_FR: 'fr-FR',
    gl_ES: 'gl-ES',
    gu: 'gu',
    he: 'he',
    hi: 'hi',
    hr: 'hr',
    hu: 'hu',
    hy: 'hy',
    id: 'id',
    is: 'is',
    it: 'it',
    ja: 'ja',
    ka: 'ka',
    kk: 'kk',
    kn: 'kn',
    ko: 'ko',
    lt: 'lt',
    lv: 'lv',
    mk: 'mk',
    ml: 'ml',
    mn: 'mn',
    mr: 'mr',
    ms: 'ms',
    my: 'my',
    nb: 'nb',
    nl: 'nl',
    nn: 'nn',
    no: 'no',
    pa: 'pa',
    pl: 'pl',
    pt: 'pt',
    pt_BR: 'pt-BR',
    pt_PT: 'pt-PT',
    ro: 'ro',
    ru: 'ru',
    sk: 'sk',
    sl: 'sl',
    so: 'so',
    sq: 'sq',
    sr: 'sr',
    sv: 'sv',
    sw: 'sw',
    ta: 'ta',
    te: 'te',
    th: 'th',
    tl: 'tl',
    tr: 'tr',
    uk: 'uk',
    ur: 'ur',
    vi: 'vi',
    zgh: 'zgh',
    zh_CN: 'zh-CN',
    zh_HK: 'zh-HK',
    zh_TW: 'zh-TW',
};
/**
 *
 */
exports.GetPartialsPromptEnum = {
    login: 'login',
    login_id: 'login-id',
    login_password: 'login-password',
    login_passwordless: 'login-passwordless',
    signup: 'signup',
    signup_id: 'signup-id',
    signup_password: 'signup-password',
    customized_consent: 'customized-consent',
};
/**
 *
 */
exports.GetRenderingPromptEnum = {
    login: 'login',
    login_id: 'login-id',
    login_password: 'login-password',
    login_passwordless: 'login-passwordless',
    login_email_verification: 'login-email-verification',
    signup: 'signup',
    signup_id: 'signup-id',
    signup_password: 'signup-password',
    phone_identifier_enrollment: 'phone-identifier-enrollment',
    phone_identifier_challenge: 'phone-identifier-challenge',
    email_identifier_challenge: 'email-identifier-challenge',
    reset_password: 'reset-password',
    custom_form: 'custom-form',
    consent: 'consent',
    customized_consent: 'customized-consent',
    logout: 'logout',
    mfa_push: 'mfa-push',
    mfa_otp: 'mfa-otp',
    mfa_voice: 'mfa-voice',
    mfa_phone: 'mfa-phone',
    mfa_webauthn: 'mfa-webauthn',
    mfa_sms: 'mfa-sms',
    mfa_email: 'mfa-email',
    mfa_recovery_code: 'mfa-recovery-code',
    mfa: 'mfa',
    status: 'status',
    device_flow: 'device-flow',
    email_verification: 'email-verification',
    email_otp_challenge: 'email-otp-challenge',
    organizations: 'organizations',
    invitation: 'invitation',
    common: 'common',
    passkeys: 'passkeys',
    captcha: 'captcha',
};
/**
 *
 */
exports.GetRenderingScreenEnum = {
    login: 'login',
    login_id: 'login-id',
    login_password: 'login-password',
    login_passwordless_email_code: 'login-passwordless-email-code',
    login_passwordless_email_link: 'login-passwordless-email-link',
    login_passwordless_sms_otp: 'login-passwordless-sms-otp',
    login_email_verification: 'login-email-verification',
    signup: 'signup',
    signup_id: 'signup-id',
    signup_password: 'signup-password',
    phone_identifier_enrollment: 'phone-identifier-enrollment',
    phone_identifier_challenge: 'phone-identifier-challenge',
    email_identifier_challenge: 'email-identifier-challenge',
    reset_password_request: 'reset-password-request',
    reset_password_email: 'reset-password-email',
    reset_password: 'reset-password',
    reset_password_success: 'reset-password-success',
    reset_password_error: 'reset-password-error',
    reset_password_mfa_email_challenge: 'reset-password-mfa-email-challenge',
    reset_password_mfa_otp_challenge: 'reset-password-mfa-otp-challenge',
    reset_password_mfa_phone_challenge: 'reset-password-mfa-phone-challenge',
    reset_password_mfa_push_challenge_push: 'reset-password-mfa-push-challenge-push',
    reset_password_mfa_recovery_code_challenge: 'reset-password-mfa-recovery-code-challenge',
    reset_password_mfa_sms_challenge: 'reset-password-mfa-sms-challenge',
    reset_password_mfa_voice_challenge: 'reset-password-mfa-voice-challenge',
    reset_password_mfa_webauthn_platform_challenge: 'reset-password-mfa-webauthn-platform-challenge',
    reset_password_mfa_webauthn_roaming_challenge: 'reset-password-mfa-webauthn-roaming-challenge',
    custom_form: 'custom-form',
    consent: 'consent',
    customized_consent: 'customized-consent',
    logout: 'logout',
    logout_complete: 'logout-complete',
    logout_aborted: 'logout-aborted',
    mfa_push_welcome: 'mfa-push-welcome',
    mfa_push_enrollment_qr: 'mfa-push-enrollment-qr',
    mfa_push_enrollment_code: 'mfa-push-enrollment-code',
    mfa_push_success: 'mfa-push-success',
    mfa_push_challenge_push: 'mfa-push-challenge-push',
    mfa_push_list: 'mfa-push-list',
    mfa_otp_enrollment_qr: 'mfa-otp-enrollment-qr',
    mfa_otp_enrollment_code: 'mfa-otp-enrollment-code',
    mfa_otp_challenge: 'mfa-otp-challenge',
    mfa_voice_enrollment: 'mfa-voice-enrollment',
    mfa_voice_challenge: 'mfa-voice-challenge',
    mfa_phone_challenge: 'mfa-phone-challenge',
    mfa_phone_enrollment: 'mfa-phone-enrollment',
    mfa_webauthn_platform_enrollment: 'mfa-webauthn-platform-enrollment',
    mfa_webauthn_roaming_enrollment: 'mfa-webauthn-roaming-enrollment',
    mfa_webauthn_platform_challenge: 'mfa-webauthn-platform-challenge',
    mfa_webauthn_roaming_challenge: 'mfa-webauthn-roaming-challenge',
    mfa_webauthn_change_key_nickname: 'mfa-webauthn-change-key-nickname',
    mfa_webauthn_enrollment_success: 'mfa-webauthn-enrollment-success',
    mfa_webauthn_error: 'mfa-webauthn-error',
    mfa_webauthn_not_available_error: 'mfa-webauthn-not-available-error',
    mfa_country_codes: 'mfa-country-codes',
    mfa_sms_enrollment: 'mfa-sms-enrollment',
    mfa_sms_challenge: 'mfa-sms-challenge',
    mfa_sms_list: 'mfa-sms-list',
    mfa_email_challenge: 'mfa-email-challenge',
    mfa_email_list: 'mfa-email-list',
    mfa_recovery_code_enrollment: 'mfa-recovery-code-enrollment',
    mfa_recovery_code_challenge: 'mfa-recovery-code-challenge',
    mfa_detect_browser_capabilities: 'mfa-detect-browser-capabilities',
    mfa_enroll_result: 'mfa-enroll-result',
    mfa_login_options: 'mfa-login-options',
    mfa_begin_enroll_options: 'mfa-begin-enroll-options',
    status: 'status',
    device_code_activation: 'device-code-activation',
    device_code_activation_allowed: 'device-code-activation-allowed',
    device_code_activation_denied: 'device-code-activation-denied',
    device_code_confirmation: 'device-code-confirmation',
    email_verification_result: 'email-verification-result',
    email_otp_challenge: 'email-otp-challenge',
    organization_selection: 'organization-selection',
    organization_picker: 'organization-picker',
    accept_invitation: 'accept-invitation',
    redeem_ticket: 'redeem-ticket',
    passkey_enrollment: 'passkey-enrollment',
    passkey_enrollment_local: 'passkey-enrollment-local',
    interstitial_captcha: 'interstitial-captcha',
};
/**
 *
 */
exports.PatchRenderingOperationPromptEnum = {
    login: 'login',
    login_id: 'login-id',
    login_password: 'login-password',
    login_passwordless: 'login-passwordless',
    login_email_verification: 'login-email-verification',
    signup: 'signup',
    signup_id: 'signup-id',
    signup_password: 'signup-password',
    phone_identifier_enrollment: 'phone-identifier-enrollment',
    phone_identifier_challenge: 'phone-identifier-challenge',
    email_identifier_challenge: 'email-identifier-challenge',
    reset_password: 'reset-password',
    custom_form: 'custom-form',
    consent: 'consent',
    customized_consent: 'customized-consent',
    logout: 'logout',
    mfa_push: 'mfa-push',
    mfa_otp: 'mfa-otp',
    mfa_voice: 'mfa-voice',
    mfa_phone: 'mfa-phone',
    mfa_webauthn: 'mfa-webauthn',
    mfa_sms: 'mfa-sms',
    mfa_email: 'mfa-email',
    mfa_recovery_code: 'mfa-recovery-code',
    mfa: 'mfa',
    status: 'status',
    device_flow: 'device-flow',
    email_verification: 'email-verification',
    email_otp_challenge: 'email-otp-challenge',
    organizations: 'organizations',
    invitation: 'invitation',
    common: 'common',
    passkeys: 'passkeys',
    captcha: 'captcha',
};
/**
 *
 */
exports.PatchRenderingOperationScreenEnum = {
    login: 'login',
    login_id: 'login-id',
    login_password: 'login-password',
    login_passwordless_email_code: 'login-passwordless-email-code',
    login_passwordless_email_link: 'login-passwordless-email-link',
    login_passwordless_sms_otp: 'login-passwordless-sms-otp',
    login_email_verification: 'login-email-verification',
    signup: 'signup',
    signup_id: 'signup-id',
    signup_password: 'signup-password',
    phone_identifier_enrollment: 'phone-identifier-enrollment',
    phone_identifier_challenge: 'phone-identifier-challenge',
    email_identifier_challenge: 'email-identifier-challenge',
    reset_password_request: 'reset-password-request',
    reset_password_email: 'reset-password-email',
    reset_password: 'reset-password',
    reset_password_success: 'reset-password-success',
    reset_password_error: 'reset-password-error',
    reset_password_mfa_email_challenge: 'reset-password-mfa-email-challenge',
    reset_password_mfa_otp_challenge: 'reset-password-mfa-otp-challenge',
    reset_password_mfa_phone_challenge: 'reset-password-mfa-phone-challenge',
    reset_password_mfa_push_challenge_push: 'reset-password-mfa-push-challenge-push',
    reset_password_mfa_recovery_code_challenge: 'reset-password-mfa-recovery-code-challenge',
    reset_password_mfa_sms_challenge: 'reset-password-mfa-sms-challenge',
    reset_password_mfa_voice_challenge: 'reset-password-mfa-voice-challenge',
    reset_password_mfa_webauthn_platform_challenge: 'reset-password-mfa-webauthn-platform-challenge',
    reset_password_mfa_webauthn_roaming_challenge: 'reset-password-mfa-webauthn-roaming-challenge',
    custom_form: 'custom-form',
    consent: 'consent',
    customized_consent: 'customized-consent',
    logout: 'logout',
    logout_complete: 'logout-complete',
    logout_aborted: 'logout-aborted',
    mfa_push_welcome: 'mfa-push-welcome',
    mfa_push_enrollment_qr: 'mfa-push-enrollment-qr',
    mfa_push_enrollment_code: 'mfa-push-enrollment-code',
    mfa_push_success: 'mfa-push-success',
    mfa_push_challenge_push: 'mfa-push-challenge-push',
    mfa_push_list: 'mfa-push-list',
    mfa_otp_enrollment_qr: 'mfa-otp-enrollment-qr',
    mfa_otp_enrollment_code: 'mfa-otp-enrollment-code',
    mfa_otp_challenge: 'mfa-otp-challenge',
    mfa_voice_enrollment: 'mfa-voice-enrollment',
    mfa_voice_challenge: 'mfa-voice-challenge',
    mfa_phone_challenge: 'mfa-phone-challenge',
    mfa_phone_enrollment: 'mfa-phone-enrollment',
    mfa_webauthn_platform_enrollment: 'mfa-webauthn-platform-enrollment',
    mfa_webauthn_roaming_enrollment: 'mfa-webauthn-roaming-enrollment',
    mfa_webauthn_platform_challenge: 'mfa-webauthn-platform-challenge',
    mfa_webauthn_roaming_challenge: 'mfa-webauthn-roaming-challenge',
    mfa_webauthn_change_key_nickname: 'mfa-webauthn-change-key-nickname',
    mfa_webauthn_enrollment_success: 'mfa-webauthn-enrollment-success',
    mfa_webauthn_error: 'mfa-webauthn-error',
    mfa_webauthn_not_available_error: 'mfa-webauthn-not-available-error',
    mfa_country_codes: 'mfa-country-codes',
    mfa_sms_enrollment: 'mfa-sms-enrollment',
    mfa_sms_challenge: 'mfa-sms-challenge',
    mfa_sms_list: 'mfa-sms-list',
    mfa_email_challenge: 'mfa-email-challenge',
    mfa_email_list: 'mfa-email-list',
    mfa_recovery_code_enrollment: 'mfa-recovery-code-enrollment',
    mfa_recovery_code_challenge: 'mfa-recovery-code-challenge',
    mfa_detect_browser_capabilities: 'mfa-detect-browser-capabilities',
    mfa_enroll_result: 'mfa-enroll-result',
    mfa_login_options: 'mfa-login-options',
    mfa_begin_enroll_options: 'mfa-begin-enroll-options',
    status: 'status',
    device_code_activation: 'device-code-activation',
    device_code_activation_allowed: 'device-code-activation-allowed',
    device_code_activation_denied: 'device-code-activation-denied',
    device_code_confirmation: 'device-code-confirmation',
    email_verification_result: 'email-verification-result',
    email_otp_challenge: 'email-otp-challenge',
    organization_selection: 'organization-selection',
    organization_picker: 'organization-picker',
    accept_invitation: 'accept-invitation',
    redeem_ticket: 'redeem-ticket',
    passkey_enrollment: 'passkey-enrollment',
    passkey_enrollment_local: 'passkey-enrollment-local',
    interstitial_captcha: 'interstitial-captcha',
};
/**
 *
 */
exports.PutCustomTextByLanguagePromptEnum = {
    login: 'login',
    login_id: 'login-id',
    login_password: 'login-password',
    login_passwordless: 'login-passwordless',
    login_email_verification: 'login-email-verification',
    signup: 'signup',
    signup_id: 'signup-id',
    signup_password: 'signup-password',
    phone_identifier_enrollment: 'phone-identifier-enrollment',
    phone_identifier_challenge: 'phone-identifier-challenge',
    email_identifier_challenge: 'email-identifier-challenge',
    reset_password: 'reset-password',
    custom_form: 'custom-form',
    consent: 'consent',
    customized_consent: 'customized-consent',
    logout: 'logout',
    mfa_push: 'mfa-push',
    mfa_otp: 'mfa-otp',
    mfa_voice: 'mfa-voice',
    mfa_phone: 'mfa-phone',
    mfa_webauthn: 'mfa-webauthn',
    mfa_sms: 'mfa-sms',
    mfa_email: 'mfa-email',
    mfa_recovery_code: 'mfa-recovery-code',
    mfa: 'mfa',
    status: 'status',
    device_flow: 'device-flow',
    email_verification: 'email-verification',
    email_otp_challenge: 'email-otp-challenge',
    organizations: 'organizations',
    invitation: 'invitation',
    common: 'common',
    passkeys: 'passkeys',
    captcha: 'captcha',
};
/**
 *
 */
exports.PutCustomTextByLanguageLanguageEnum = {
    am: 'am',
    ar: 'ar',
    ar_EG: 'ar-EG',
    ar_SA: 'ar-SA',
    az: 'az',
    bg: 'bg',
    bn: 'bn',
    bs: 'bs',
    ca_ES: 'ca-ES',
    cnr: 'cnr',
    cs: 'cs',
    cy: 'cy',
    da: 'da',
    de: 'de',
    el: 'el',
    en: 'en',
    en_CA: 'en-CA',
    es: 'es',
    es_419: 'es-419',
    es_AR: 'es-AR',
    es_MX: 'es-MX',
    et: 'et',
    eu_ES: 'eu-ES',
    fa: 'fa',
    fi: 'fi',
    fr: 'fr',
    fr_CA: 'fr-CA',
    fr_FR: 'fr-FR',
    gl_ES: 'gl-ES',
    gu: 'gu',
    he: 'he',
    hi: 'hi',
    hr: 'hr',
    hu: 'hu',
    hy: 'hy',
    id: 'id',
    is: 'is',
    it: 'it',
    ja: 'ja',
    ka: 'ka',
    kk: 'kk',
    kn: 'kn',
    ko: 'ko',
    lt: 'lt',
    lv: 'lv',
    mk: 'mk',
    ml: 'ml',
    mn: 'mn',
    mr: 'mr',
    ms: 'ms',
    my: 'my',
    nb: 'nb',
    nl: 'nl',
    nn: 'nn',
    no: 'no',
    pa: 'pa',
    pl: 'pl',
    pt: 'pt',
    pt_BR: 'pt-BR',
    pt_PT: 'pt-PT',
    ro: 'ro',
    ru: 'ru',
    sk: 'sk',
    sl: 'sl',
    so: 'so',
    sq: 'sq',
    sr: 'sr',
    sv: 'sv',
    sw: 'sw',
    ta: 'ta',
    te: 'te',
    th: 'th',
    tl: 'tl',
    tr: 'tr',
    uk: 'uk',
    ur: 'ur',
    vi: 'vi',
    zgh: 'zgh',
    zh_CN: 'zh-CN',
    zh_HK: 'zh-HK',
    zh_TW: 'zh-TW',
};
/**
 *
 */
exports.PutPartialsPromptEnum = {
    login: 'login',
    login_id: 'login-id',
    login_password: 'login-password',
    login_passwordless: 'login-passwordless',
    signup: 'signup',
    signup_id: 'signup-id',
    signup_password: 'signup-password',
    customized_consent: 'customized-consent',
};
/**
 *
 */
exports.GetSelfServiceProfileCustomTextLanguageEnum = {
    en: 'en',
};
/**
 *
 */
exports.GetSelfServiceProfileCustomTextPageEnum = {
    get_started: 'get-started',
};
/**
 *
 */
exports.PutSelfServiceProfileCustomTextLanguageEnum = {
    en: 'en',
};
/**
 *
 */
exports.PutSelfServiceProfileCustomTextPageEnum = {
    get_started: 'get-started',
};
/**
 *
 */
exports.DeleteMultifactorByProviderProviderEnum = {
    duo: 'duo',
    google_authenticator: 'google-authenticator',
};
/**
 *
 */
exports.DeleteUserIdentityByUserIdProviderEnum = {
    ad: 'ad',
    adfs: 'adfs',
    amazon: 'amazon',
    apple: 'apple',
    dropbox: 'dropbox',
    bitbucket: 'bitbucket',
    aol: 'aol',
    auth0_oidc: 'auth0-oidc',
    auth0: 'auth0',
    baidu: 'baidu',
    bitly: 'bitly',
    box: 'box',
    custom: 'custom',
    daccount: 'daccount',
    dwolla: 'dwolla',
    email: 'email',
    evernote_sandbox: 'evernote-sandbox',
    evernote: 'evernote',
    exact: 'exact',
    facebook: 'facebook',
    fitbit: 'fitbit',
    flickr: 'flickr',
    github: 'github',
    google_apps: 'google-apps',
    google_oauth2: 'google-oauth2',
    instagram: 'instagram',
    ip: 'ip',
    line: 'line',
    linkedin: 'linkedin',
    miicard: 'miicard',
    oauth1: 'oauth1',
    oauth2: 'oauth2',
    office365: 'office365',
    oidc: 'oidc',
    okta: 'okta',
    paypal: 'paypal',
    paypal_sandbox: 'paypal-sandbox',
    pingfederate: 'pingfederate',
    planningcenter: 'planningcenter',
    renren: 'renren',
    salesforce_community: 'salesforce-community',
    salesforce_sandbox: 'salesforce-sandbox',
    salesforce: 'salesforce',
    samlp: 'samlp',
    sharepoint: 'sharepoint',
    shopify: 'shopify',
    sms: 'sms',
    soundcloud: 'soundcloud',
    thecity_sandbox: 'thecity-sandbox',
    thecity: 'thecity',
    thirtysevensignals: 'thirtysevensignals',
    twitter: 'twitter',
    untappd: 'untappd',
    vkontakte: 'vkontakte',
    waad: 'waad',
    weibo: 'weibo',
    windowslive: 'windowslive',
    wordpress: 'wordpress',
    yahoo: 'yahoo',
    yammer: 'yammer',
    yandex: 'yandex',
};
/**
 *
 */
exports.GetUsersSearchEngineEnum = {
    v1: 'v1',
    v2: 'v2',
    v3: 'v3',
};
