// auth0/dist/esm/management/__generated/models/index.js
/**
 *
 */
export const AssessorsTypeEnum = {
    new_device: 'new-device',
};
/**
 *
 */
export const AsyncApprovalNotificationsChannelsEnum = {
    guardian_push: 'guardian-push',
    email: 'email',
};
/**
 * The id of the active provider for the CAPTCHA.
 */
export const AttackProtectionCaptchaProviderId = {
    arkose: 'arkose',
    auth_challenge: 'auth_challenge',
    friendly_captcha: 'friendly_captcha',
    hcaptcha: 'hcaptcha',
    recaptcha_v2: 'recaptcha_v2',
    recaptcha_enterprise: 'recaptcha_enterprise',
    simple_captcha: 'simple_captcha',
};
/**
 * The policy that defines how often to show CAPTCHA
 */
export const BotDetectionChallengePolicyPasswordFlowEnum = {
    never: 'never',
    when_risky: 'when_risky',
    always: 'always',
};
/**
 * The policy that defines how often to show CAPTCHA
 */
export const BotDetectionChallengePolicyPasswordResetFlowEnum = {
    never: 'never',
    when_risky: 'when_risky',
    always: 'always',
};
/**
 * The policy that defines how often to show CAPTCHA
 */
export const BotDetectionChallengePolicyPasswordlessFlowEnum = {
    never: 'never',
    when_risky: 'when_risky',
    always: 'always',
};
/**
 * The level of bot detection sensitivity
 */
export const BotDetectionLevelEnum = {
    low: 'low',
    medium: 'medium',
    high: 'high',
};
/**
 * The captcha assessment policy for userpass flow enabled for this tenant
 */
export const CaptchaEnforcementPolicyEnum = {
    high_risk: 'high_risk',
    always_on: 'always_on',
    off: 'off',
};
/**
 * The selected captcha provider.
 */
export const CaptchaProviderEnum = {
    auth0: 'auth0',
    auth0_v2: 'auth0_v2',
    recaptcha_v2: 'recaptcha_v2',
    recaptcha_enterprise: 'recaptcha_enterprise',
    hcaptcha: 'hcaptcha',
    friendly_captcha: 'friendly_captcha',
    arkose: 'arkose',
};
export const CertificateStatusEnum = {
    provisioning: 'provisioning',
    provisioning_failed: 'provisioning_failed',
    provisioned: 'provisioned',
    renewing_failed: 'renewing_failed',
};
export const CertificateCertificateAuthorityEnum = {
    letsencrypt: 'letsencrypt',
    googletrust: 'googletrust',
};
export const ClientAppTypeEnum = {
    native: 'native',
    spa: 'spa',
    regular_web: 'regular_web',
    non_interactive: 'non_interactive',
    resource_server: 'resource_server',
    express_configuration: 'express_configuration',
    rms: 'rms',
    box: 'box',
    cloudbees: 'cloudbees',
    concur: 'concur',
    dropbox: 'dropbox',
    mscrm: 'mscrm',
    echosign: 'echosign',
    egnyte: 'egnyte',
    newrelic: 'newrelic',
    office365: 'office365',
    salesforce: 'salesforce',
    sentry: 'sentry',
    sharepoint: 'sharepoint',
    slack: 'slack',
    springcm: 'springcm',
    zendesk: 'zendesk',
    zoom: 'zoom',
    sso_integration: 'sso_integration',
    oag: 'oag',
};
export const ClientTokenEndpointAuthMethodEnum = {
    none: 'none',
    client_secret_post: 'client_secret_post',
    client_secret_basic: 'client_secret_basic',
};
export const ClientOrganizationUsageEnum = {
    deny: 'deny',
    allow: 'allow',
    require: 'require',
};
export const ClientOrganizationRequireBehaviorEnum = {
    no_prompt: 'no_prompt',
    pre_login_prompt: 'pre_login_prompt',
    post_login_prompt: 'post_login_prompt',
};
export const ClientComplianceLevelEnum = {
    none: 'none',
    fapi1_adv_pkj_par: 'fapi1_adv_pkj_par',
    fapi1_adv_mtls_par: 'fapi1_adv_mtls_par',
    null: 'null',
};
export const ClientCreateTokenEndpointAuthMethodEnum = {
    none: 'none',
    client_secret_post: 'client_secret_post',
    client_secret_basic: 'client_secret_basic',
};
export const ClientCreateAppTypeEnum = {
    native: 'native',
    spa: 'spa',
    regular_web: 'regular_web',
    non_interactive: 'non_interactive',
    resource_server: 'resource_server',
    rms: 'rms',
    box: 'box',
    cloudbees: 'cloudbees',
    concur: 'concur',
    dropbox: 'dropbox',
    mscrm: 'mscrm',
    echosign: 'echosign',
    egnyte: 'egnyte',
    newrelic: 'newrelic',
    office365: 'office365',
    salesforce: 'salesforce',
    sentry: 'sentry',
    sharepoint: 'sharepoint',
    slack: 'slack',
    springcm: 'springcm',
    zendesk: 'zendesk',
    zoom: 'zoom',
    sso_integration: 'sso_integration',
    oag: 'oag',
};
export const ClientCreateOrganizationUsageEnum = {
    deny: 'deny',
    allow: 'allow',
    require: 'require',
};
export const ClientCreateOrganizationRequireBehaviorEnum = {
    no_prompt: 'no_prompt',
    pre_login_prompt: 'pre_login_prompt',
    post_login_prompt: 'post_login_prompt',
};
export const ClientCreateComplianceLevelEnum = {
    none: 'none',
    fapi1_adv_pkj_par: 'fapi1_adv_pkj_par',
    fapi1_adv_mtls_par: 'fapi1_adv_mtls_par',
    null: 'null',
};
export const ClientCreateClientAuthenticationMethodsPrivateKeyJwtCredentialsInnerCredentialTypeEnum = {
    public_key: 'public_key',
};
export const ClientCreateClientAuthenticationMethodsPrivateKeyJwtCredentialsInnerAlgEnum = {
    RS256: 'RS256',
    RS384: 'RS384',
    PS256: 'PS256',
};
export const ClientCreateClientAuthenticationMethodsSelfSignedTlsClientAuthCredentialsInnerCredentialTypeEnum = {
    x509_cert: 'x509_cert',
};
export const ClientCreateClientAuthenticationMethodsTlsClientAuthCredentialsInnerCredentialTypeEnum = {
    cert_subject_dn: 'cert_subject_dn',
};
export const ClientCreateDefaultOrganizationFlowsEnum = {
    client_credentials: 'client_credentials',
};
export const ClientCreateJwtConfigurationAlgEnum = {
    HS256: 'HS256',
    RS256: 'RS256',
    PS256: 'PS256',
};
export const ClientDefaultOrganizationFlowsEnum = {
    client_credentials: 'client_credentials',
};
export const ClientGrantOrganizationUsageEnum = {
    deny: 'deny',
    allow: 'allow',
    require: 'require',
};
export const ClientGrantSubjectTypeEnum = {
    client: 'client',
    user: 'user',
};
export const ClientGrantCreateOrganizationUsageEnum = {
    deny: 'deny',
    allow: 'allow',
    require: 'require',
};
export const ClientGrantCreateSubjectTypeEnum = {
    client: 'client',
    user: 'user',
};
export const ClientJwtConfigurationAlgEnum = {
    HS256: 'HS256',
    RS256: 'RS256',
    PS256: 'PS256',
};
export const ClientOidcLogoutBackchannelLogoutInitiatorsModeEnum = {
    custom: 'custom',
    all: 'all',
};
export const ClientOidcLogoutBackchannelLogoutInitiatorsSelectedInitiatorsEnum = {
    rp_logout: 'rp-logout',
    idp_logout: 'idp-logout',
    password_changed: 'password-changed',
    session_expired: 'session-expired',
    session_revoked: 'session-revoked',
    account_deleted: 'account-deleted',
    email_identifier_changed: 'email-identifier-changed',
    mfa_phone_unenrolled: 'mfa-phone-unenrolled',
    account_deactivated: 'account-deactivated',
};
/**
 * Method for discovering organizations during the `pre_login_prompt`. `email` allows users to find their organization by entering their email address and performing domain matching, while `organization_name` requires users to enter the organization name directly. These methods can be combined.
 */
export const ClientOrganizationDiscoveryEnum = {
    email: 'email',
    organization_name: 'organization_name',
};
export const ClientRefreshTokenRotationTypeEnum = {
    rotating: 'rotating',
    non_rotating: 'non-rotating',
};
export const ClientRefreshTokenExpirationTypeEnum = {
    expiring: 'expiring',
    non_expiring: 'non-expiring',
};
export const ClientSessionTransferAllowedAuthenticationMethodsEnum = {
    cookie: 'cookie',
    query: 'query',
};
export const ClientSessionTransferEnforceDeviceBindingEnum = {
    ip: 'ip',
    asn: 'asn',
    none: 'none',
};
export const ClientUpdateTokenEndpointAuthMethodEnum = {
    none: 'none',
    client_secret_post: 'client_secret_post',
    client_secret_basic: 'client_secret_basic',
    null: 'null',
};
export const ClientUpdateAppTypeEnum = {
    native: 'native',
    spa: 'spa',
    regular_web: 'regular_web',
    non_interactive: 'non_interactive',
    resource_server: 'resource_server',
    rms: 'rms',
    box: 'box',
    cloudbees: 'cloudbees',
    concur: 'concur',
    dropbox: 'dropbox',
    mscrm: 'mscrm',
    echosign: 'echosign',
    egnyte: 'egnyte',
    newrelic: 'newrelic',
    office365: 'office365',
    salesforce: 'salesforce',
    sentry: 'sentry',
    sharepoint: 'sharepoint',
    slack: 'slack',
    springcm: 'springcm',
    zendesk: 'zendesk',
    zoom: 'zoom',
    sso_integration: 'sso_integration',
    oag: 'oag',
};
export const ClientUpdateOrganizationUsageEnum = {
    deny: 'deny',
    allow: 'allow',
    require: 'require',
};
export const ClientUpdateOrganizationRequireBehaviorEnum = {
    no_prompt: 'no_prompt',
    pre_login_prompt: 'pre_login_prompt',
    post_login_prompt: 'post_login_prompt',
};
export const ClientUpdateComplianceLevelEnum = {
    none: 'none',
    fapi1_adv_pkj_par: 'fapi1_adv_pkj_par',
    fapi1_adv_mtls_par: 'fapi1_adv_mtls_par',
    null: 'null',
};
export const ClientUpdateJwtConfigurationAlgEnum = {
    HS256: 'HS256',
    RS256: 'RS256',
    PS256: 'PS256',
};
export const ClientUpdateOidcLogoutBackchannelLogoutInitiatorsModeEnum = {
    custom: 'custom',
    all: 'all',
};
export const ClientUpdateOidcLogoutBackchannelLogoutInitiatorsSelectedInitiatorsEnum = {
    rp_logout: 'rp-logout',
    idp_logout: 'idp-logout',
    password_changed: 'password-changed',
    session_expired: 'session-expired',
    session_revoked: 'session-revoked',
    account_deleted: 'account-deleted',
    email_identifier_changed: 'email-identifier-changed',
    mfa_phone_unenrolled: 'mfa-phone-unenrolled',
    account_deactivated: 'account-deactivated',
};
/**
 * The access type for the connected account.
 */
export const ConnectedAccountAccessTypeEnum = {
    offline: 'offline',
};
export const ConnectionCreateStrategyEnum = {
    ad: 'ad',
    adfs: 'adfs',
    amazon: 'amazon',
    apple: 'apple',
    dropbox: 'dropbox',
    bitbucket: 'bitbucket',
    aol: 'aol',
    auth0_oidc: 'auth0-oidc',
    auth0: 'auth0',
    baidu: 'baidu',
    bitly: 'bitly',
    box: 'box',
    custom: 'custom',
    daccount: 'daccount',
    dwolla: 'dwolla',
    email: 'email',
    evernote_sandbox: 'evernote-sandbox',
    evernote: 'evernote',
    exact: 'exact',
    facebook: 'facebook',
    fitbit: 'fitbit',
    flickr: 'flickr',
    github: 'github',
    google_apps: 'google-apps',
    google_oauth2: 'google-oauth2',
    instagram: 'instagram',
    ip: 'ip',
    line: 'line',
    linkedin: 'linkedin',
    miicard: 'miicard',
    oauth1: 'oauth1',
    oauth2: 'oauth2',
    office365: 'office365',
    oidc: 'oidc',
    okta: 'okta',
    paypal: 'paypal',
    paypal_sandbox: 'paypal-sandbox',
    pingfederate: 'pingfederate',
    planningcenter: 'planningcenter',
    renren: 'renren',
    salesforce_community: 'salesforce-community',
    salesforce_sandbox: 'salesforce-sandbox',
    salesforce: 'salesforce',
    samlp: 'samlp',
    sharepoint: 'sharepoint',
    shopify: 'shopify',
    sms: 'sms',
    soundcloud: 'soundcloud',
    thecity_sandbox: 'thecity-sandbox',
    thecity: 'thecity',
    thirtysevensignals: 'thirtysevensignals',
    twitter: 'twitter',
    untappd: 'untappd',
    vkontakte: 'vkontakte',
    waad: 'waad',
    weibo: 'weibo',
    windowslive: 'windowslive',
    wordpress: 'wordpress',
    yahoo: 'yahoo',
    yammer: 'yammer',
    yandex: 'yandex',
};
export const ConnectionCreateOptionsPasswordPolicyEnum = {
    none: 'none',
    low: 'low',
    fair: 'fair',
    good: 'good',
    excellent: 'excellent',
    null: 'null',
};
export const ConnectionCreateOptionsSetUserRootAttributesEnum = {
    each_login: 'on_each_login',
    first_login: 'on_first_login',
};
export const ConnectionCreateOptionsAttributesEmailVerificationMethodEnum = {
    link: 'link',
    otp: 'otp',
};
export const ConnectionCreateOptionsAttributesEmailSignupStatusEnum = {
    required: 'required',
    optional: 'optional',
    inactive: 'inactive',
};
export const ConnectionCreateOptionsPasskeyOptionsChallengeUiEnum = {
    both: 'both',
    autofill: 'autofill',
    button: 'button',
};
/**
 * Indicates if membership should be assigned on login.
 */
export const ConnectionProfileOrganizationAssignMembershipOnLoginEnum = {
    none: 'none',
    optional: 'optional',
    required: 'required',
};
/**
 * Indicates if the organization should be shown as a button.
 */
export const ConnectionProfileOrganizationShowAsButtonEnum = {
    none: 'none',
    optional: 'optional',
    required: 'required',
};
export const ConnectionUpdateOptionsPasswordPolicyEnum = {
    none: 'none',
    low: 'low',
    fair: 'fair',
    good: 'good',
    excellent: 'excellent',
    null: 'null',
};
export const ConnectionUpdateOptionsSetUserRootAttributesEnum = {
    each_login: 'on_each_login',
    first_login: 'on_first_login',
};
export const CreatePhoneProviderRequestNameEnum = {
    twilio: 'twilio',
    custom: 'custom',
};
export const CreatePhoneTemplateRequestContentTypeEnum = {
    otp_verify: 'otp_verify',
    otp_enroll: 'otp_enroll',
    change_password: 'change_password',
    blocked_account: 'blocked_account',
    password_breach: 'password_breach',
};
export const CreatePhoneTemplateResponseContentTypeEnum = {
    otp_verify: 'otp_verify',
    otp_enroll: 'otp_enroll',
    change_password: 'change_password',
    blocked_account: 'blocked_account',
    password_breach: 'password_breach',
};
export const CustomDomainStatusEnum = {
    disabled: 'disabled',
    pending: 'pending',
    pending_verification: 'pending_verification',
    ready: 'ready',
};
export const CustomDomainTypeEnum = {
    auth0_managed_certs: 'auth0_managed_certs',
    self_managed_certs: 'self_managed_certs',
};
export const DeviceCredentialTypeEnum = {
    public_key: 'public_key',
    refresh_token: 'refresh_token',
    rotating_refresh_token: 'rotating_refresh_token',
};
export const DeviceCredentialCreateTypeEnum = {
    public_key: 'public_key',
};
export const EmailProviderCreateNameEnum = {
    mailgun: 'mailgun',
    mandrill: 'mandrill',
    sendgrid: 'sendgrid',
    ses: 'ses',
    sparkpost: 'sparkpost',
    smtp: 'smtp',
    azure_cs: 'azure_cs',
    ms365: 'ms365',
    custom: 'custom',
};
export const EmailProviderUpdateNameEnum = {
    mailgun: 'mailgun',
    mandrill: 'mandrill',
    sendgrid: 'sendgrid',
    ses: 'ses',
    sparkpost: 'sparkpost',
    smtp: 'smtp',
    azure_cs: 'azure_cs',
    ms365: 'ms365',
    custom: 'custom',
};
export const EmailTemplateUpdateTemplateEnum = {
    verify_email: 'verify_email',
    verify_email_by_code: 'verify_email_by_code',
    reset_email: 'reset_email',
    reset_email_by_code: 'reset_email_by_code',
    welcome_email: 'welcome_email',
    blocked_account: 'blocked_account',
    stolen_credentials: 'stolen_credentials',
    enrollment_email: 'enrollment_email',
    mfa_oob_code: 'mfa_oob_code',
    user_invitation: 'user_invitation',
    change_password: 'change_password',
    password_reset: 'password_reset',
    async_approval: 'async_approval',
};
/**
 * Enum for enabled features.
 */
export const EnabledFeaturesEnum = {
    scim: 'scim',
    universal_logout: 'universal_logout',
};
export const EnrollmentStatusEnum = {
    pending: 'pending',
    confirmed: 'confirmed',
};
export const FactorNameEnum = {
    push_notification: 'push-notification',
    sms: 'sms',
    email: 'email',
    duo: 'duo',
    otp: 'otp',
    webauthn_roaming: 'webauthn-roaming',
    webauthn_platform: 'webauthn-platform',
    recovery_code: 'recovery-code',
};
export const GetActionVersions200ResponseVersionsInnerStatusEnum = {
    pending: 'pending',
    building: 'building',
    packaged: 'packaged',
    built: 'built',
    retrying: 'retrying',
    failed: 'failed',
};
export const GetActions200ResponseActionsInnerStatusEnum = {
    pending: 'pending',
    building: 'building',
    packaged: 'packaged',
    built: 'built',
    retrying: 'retrying',
    failed: 'failed',
};
export const GetActions200ResponseActionsInnerIntegrationFeatureTypeEnum = {
    unspecified: 'unspecified',
    action: 'action',
    social_connection: 'social_connection',
    log_stream: 'log_stream',
    sso_integration: 'sso_integration',
    sms_provider: 'sms_provider',
};
export const GetActions200ResponseActionsInnerIntegrationCurrentReleaseRequiredSecretsInnerTypeEnum = {
    UNSPECIFIED: 'UNSPECIFIED',
    STRING: 'STRING',
};
/**
 *
 */
export const GetActions200ResponseActionsInnerSupportedTriggersInnerIdAnyOf = {
    post_login: 'post-login',
    credentials_exchange: 'credentials-exchange',
    pre_user_registration: 'pre-user-registration',
    post_user_registration: 'post-user-registration',
    post_change_password: 'post-change-password',
    send_phone_message: 'send-phone-message',
    iga_approval: 'iga-approval',
    iga_certification: 'iga-certification',
    iga_fulfillment_assignment: 'iga-fulfillment-assignment',
    iga_fulfillment_execution: 'iga-fulfillment-execution',
    custom_email_provider: 'custom-email-provider',
    password_reset_post_challenge: 'password-reset-post-challenge',
};
export const GetAllRendering200ResponseOneOfInnerRenderingModeEnum = {
    advanced: 'advanced',
    standard: 'standard',
};
export const GetAuthenticationMethods200ResponseOneOfInnerTypeEnum = {
    recovery_code: 'recovery-code',
    totp: 'totp',
    push: 'push',
    phone: 'phone',
    email: 'email',
    email_verification: 'email-verification',
    webauthn_roaming: 'webauthn-roaming',
    webauthn_platform: 'webauthn-platform',
    guardian: 'guardian',
    passkey: 'passkey',
};
export const GetAuthenticationMethods200ResponseOneOfInnerPreferredAuthenticationMethodEnum = {
    sms: 'sms',
    voice: 'voice',
};
export const GetAuthenticationMethods200ResponseOneOfInnerAuthenticationMethodsInnerTypeEnum = {
    totp: 'totp',
    push: 'push',
    sms: 'sms',
    voice: 'voice',
};
export const GetBrandingPhoneProviders200ResponseProvidersInnerNameEnum = {
    twilio: 'twilio',
    custom: 'custom',
};
export const GetBrandingPhoneProviders200ResponseProvidersInnerChannelEnum = {
    phone: 'phone',
};
export const GetBrandingPhoneProviders200ResponseProvidersInnerConfigurationAnyOfDeliveryMethodsEnum = {
    text: 'text',
    voice: 'voice',
};
export const GetBrandingPhoneProviders200ResponseProvidersInnerConfigurationAnyOf1DeliveryMethodsEnum = {
    text: 'text',
    voice: 'voice',
};
export const GetBreachedPasswordDetection200ResponseShieldsEnum = {
    block: 'block',
    user_notification: 'user_notification',
    admin_notification: 'admin_notification',
};
export const GetBreachedPasswordDetection200ResponseAdminNotificationFrequencyEnum = {
    immediately: 'immediately',
    daily: 'daily',
    weekly: 'weekly',
    monthly: 'monthly',
};
export const GetBreachedPasswordDetection200ResponseMethodEnum = {
    standard: 'standard',
    enhanced: 'enhanced',
};
export const GetBreachedPasswordDetection200ResponseStagePreChangePasswordShieldsEnum = {
    block: 'block',
    admin_notification: 'admin_notification',
};
export const GetBreachedPasswordDetection200ResponseStagePreUserRegistrationShieldsEnum = {
    block: 'block',
    admin_notification: 'admin_notification',
};
export const GetBruteForceProtection200ResponseShieldsEnum = {
    block: 'block',
    user_notification: 'user_notification',
};
export const GetBruteForceProtection200ResponseModeEnum = {
    identifier_and_ip: 'count_per_identifier_and_ip',
    identifier: 'count_per_identifier',
};
export const GetConnectionsKeysResponseContentInnerKeyUseEnum = {
    encryption: 'encryption',
    signing: 'signing',
};
export const GetCredentials200ResponseInnerAlgEnum = {
    RS256: 'RS256',
    RS384: 'RS384',
    PS256: 'PS256',
};
export const GetEmailTemplatesByTemplateName200ResponseTemplateEnum = {
    verify_email: 'verify_email',
    verify_email_by_code: 'verify_email_by_code',
    reset_email: 'reset_email',
    reset_email_by_code: 'reset_email_by_code',
    welcome_email: 'welcome_email',
    blocked_account: 'blocked_account',
    stolen_credentials: 'stolen_credentials',
    enrollment_email: 'enrollment_email',
    mfa_oob_code: 'mfa_oob_code',
    user_invitation: 'user_invitation',
    change_password: 'change_password',
    password_reset: 'password_reset',
    async_approval: 'async_approval',
};
export const GetEncryptionKeys200ResponseOneOfInnerTypeEnum = {
    customer_provided_root_key: 'customer-provided-root-key',
    environment_root_key: 'environment-root-key',
    tenant_master_key: 'tenant-master-key',
    tenant_encryption_key: 'tenant-encryption-key',
};
export const GetEncryptionKeys200ResponseOneOfInnerStateEnum = {
    pre_activation: 'pre-activation',
    active: 'active',
    deactivated: 'deactivated',
    destroyed: 'destroyed',
};
export const GetExecution200ResponseStatusEnum = {
    unspecified: 'unspecified',
    pending: 'pending',
    final: 'final',
    partial: 'partial',
    canceled: 'canceled',
    suspended: 'suspended',
};
export const GetLogStreams200ResponseInnerOneOfStatusEnum = {
    active: 'active',
    paused: 'paused',
    suspended: 'suspended',
};
export const GetLogStreams200ResponseInnerOneOfTypeEnum = {
    http: 'http',
};
export const GetLogStreams200ResponseInnerOneOf1StatusEnum = {
    active: 'active',
    paused: 'paused',
    suspended: 'suspended',
};
export const GetLogStreams200ResponseInnerOneOf1TypeEnum = {
    eventbridge: 'eventbridge',
};
export const GetLogStreams200ResponseInnerOneOf1SinkAwsRegionEnum = {
    ap_east_1: 'ap-east-1',
    ap_northeast_1: 'ap-northeast-1',
    ap_northeast_2: 'ap-northeast-2',
    ap_northeast_3: 'ap-northeast-3',
    ap_south_1: 'ap-south-1',
    ap_southeast_1: 'ap-southeast-1',
    ap_southeast_2: 'ap-southeast-2',
    ca_central_1: 'ca-central-1',
    cn_north_1: 'cn-north-1',
    cn_northwest_1: 'cn-northwest-1',
    eu_central_1: 'eu-central-1',
    eu_north_1: 'eu-north-1',
    eu_west_1: 'eu-west-1',
    eu_west_2: 'eu-west-2',
    eu_west_3: 'eu-west-3',
    me_south_1: 'me-south-1',
    sa_east_1: 'sa-east-1',
    us_gov_east_1: 'us-gov-east-1',
    us_gov_west_1: 'us-gov-west-1',
    us_east_1: 'us-east-1',
    us_east_2: 'us-east-2',
    us_west_1: 'us-west-1',
    us_west_2: 'us-west-2',
};
export const GetLogStreams200ResponseInnerOneOf2StatusEnum = {
    active: 'active',
    paused: 'paused',
    suspended: 'suspended',
};
export const GetLogStreams200ResponseInnerOneOf2TypeEnum = {
    eventgrid: 'eventgrid',
};
export const GetLogStreams200ResponseInnerOneOf2SinkAzureRegionEnum = {
    australiacentral: 'australiacentral',
    australiaeast: 'australiaeast',
    australiasoutheast: 'australiasoutheast',
    brazilsouth: 'brazilsouth',
    canadacentral: 'canadacentral',
    canadaeast: 'canadaeast',
    centralindia: 'centralindia',
    centralus: 'centralus',
    eastasia: 'eastasia',
    eastus: 'eastus',
    eastus2: 'eastus2',
    francecentral: 'francecentral',
    germanywestcentral: 'germanywestcentral',
    japaneast: 'japaneast',
    japanwest: 'japanwest',
    koreacentral: 'koreacentral',
    koreasouth: 'koreasouth',
    northcentralus: 'northcentralus',
    northeurope: 'northeurope',
    norwayeast: 'norwayeast',
    southafricanorth: 'southafricanorth',
    southcentralus: 'southcentralus',
    southeastasia: 'southeastasia',
    southindia: 'southindia',
    switzerlandnorth: 'switzerlandnorth',
    uaenorth: 'uaenorth',
    uksouth: 'uksouth',
    ukwest: 'ukwest',
    westcentralus: 'westcentralus',
    westeurope: 'westeurope',
    westindia: 'westindia',
    westus: 'westus',
    westus2: 'westus2',
};
export const GetLogStreams200ResponseInnerOneOf3StatusEnum = {
    active: 'active',
    paused: 'paused',
    suspended: 'suspended',
};
export const GetLogStreams200ResponseInnerOneOf3TypeEnum = {
    datadog: 'datadog',
};
export const GetLogStreams200ResponseInnerOneOf3SinkDatadogRegionEnum = {
    us: 'us',
    eu: 'eu',
    us3: 'us3',
    us5: 'us5',
};
export const GetLogStreams200ResponseInnerOneOf4StatusEnum = {
    active: 'active',
    paused: 'paused',
    suspended: 'suspended',
};
export const GetLogStreams200ResponseInnerOneOf4TypeEnum = {
    splunk: 'splunk',
};
export const GetLogStreams200ResponseInnerOneOf5StatusEnum = {
    active: 'active',
    paused: 'paused',
    suspended: 'suspended',
};
export const GetLogStreams200ResponseInnerOneOf5TypeEnum = {
    sumo: 'sumo',
};
export const GetLogStreams200ResponseInnerOneOf6StatusEnum = {
    active: 'active',
    paused: 'paused',
    suspended: 'suspended',
};
export const GetLogStreams200ResponseInnerOneOf6TypeEnum = {
    segment: 'segment',
};
export const GetLogStreams200ResponseInnerOneOf7StatusEnum = {
    active: 'active',
    paused: 'paused',
    suspended: 'suspended',
};
export const GetLogStreams200ResponseInnerOneOf7TypeEnum = {
    mixpanel: 'mixpanel',
};
export const GetLogStreams200ResponseInnerOneOf7SinkMixpanelRegionEnum = {
    us: 'us',
    eu: 'eu',
};
export const GetLogStreams200ResponseInnerOneOfFiltersInnerTypeEnum = {
    category: 'category',
};
export const GetLogStreams200ResponseInnerOneOfFiltersInnerNameEnum = {
    auth_ancillary_fail: 'auth.ancillary.fail',
    auth_ancillary_success: 'auth.ancillary.success',
    auth_login_fail: 'auth.login.fail',
    auth_login_notification: 'auth.login.notification',
    auth_login_success: 'auth.login.success',
    auth_logout_fail: 'auth.logout.fail',
    auth_logout_success: 'auth.logout.success',
    auth_signup_fail: 'auth.signup.fail',
    auth_signup_success: 'auth.signup.success',
    auth_silent_auth_fail: 'auth.silent_auth.fail',
    auth_silent_auth_success: 'auth.silent_auth.success',
    auth_token_exchange_fail: 'auth.token_exchange.fail',
    auth_token_exchange_success: 'auth.token_exchange.success',
    management_fail: 'management.fail',
    management_success: 'management.success',
    system_notification: 'system.notification',
    user_fail: 'user.fail',
    user_notification: 'user.notification',
    user_success: 'user.success',
    other: 'other',
};
export const GetLogStreams200ResponseInnerOneOfPiiConfigLogFieldsEnum = {
    first_name: 'first_name',
    last_name: 'last_name',
    username: 'username',
    email: 'email',
    phone: 'phone',
    address: 'address',
};
export const GetLogStreams200ResponseInnerOneOfPiiConfigMethodEnum = {
    mask: 'mask',
    hash: 'hash',
};
export const GetLogStreams200ResponseInnerOneOfPiiConfigAlgorithmEnum = {
    xxhash: 'xxhash',
};
export const GetLogStreams200ResponseInnerOneOfSinkHttpContentFormatEnum = {
    JSONARRAY: 'JSONARRAY',
    JSONLINES: 'JSONLINES',
    JSONOBJECT: 'JSONOBJECT',
};
export const GetMessageTypes200ResponseMessageTypesEnum = {
    sms: 'sms',
    voice: 'voice',
};
export const GetNetworkAclsById200ResponseRuleAnyOfScopeEnum = {
    management: 'management',
    authentication: 'authentication',
    tenant: 'tenant',
};
export const GetNetworkAclsById200ResponseRuleAnyOf1ScopeEnum = {
    management: 'management',
    authentication: 'authentication',
    tenant: 'tenant',
};
export const GetNetworkAclsById200ResponseRuleAnyOfActionAnyOfBlockEnum = {
    true: true,
};
export const GetNetworkAclsById200ResponseRuleAnyOfActionAnyOf1AllowEnum = {
    true: true,
};
export const GetNetworkAclsById200ResponseRuleAnyOfActionAnyOf2LogEnum = {
    true: true,
};
export const GetNetworkAclsById200ResponseRuleAnyOfActionAnyOf3RedirectEnum = {
    true: true,
};
export const GetPhoneProviders200ResponseProviderEnum = {
    auth0: 'auth0',
    twilio: 'twilio',
    phone_message_hook: 'phone-message-hook',
};
export const GetPhoneTemplateResponseContentTypeEnum = {
    otp_verify: 'otp_verify',
    otp_enroll: 'otp_enroll',
    change_password: 'change_password',
    blocked_account: 'blocked_account',
    password_breach: 'password_breach',
};
export const GetPnProviders200ResponseProviderEnum = {
    guardian: 'guardian',
    sns: 'sns',
    direct: 'direct',
};
export const GetRendering200ResponseRenderingModeEnum = {
    advanced: 'advanced',
    standard: 'standard',
};
export const GetRendering200ResponseFiltersMatchTypeEnum = {
    includes_any: 'includes_any',
    excludes_any: 'excludes_any',
};
export const GetSuspiciousIpThrottling200ResponseShieldsEnum = {
    block: 'block',
    admin_notification: 'admin_notification',
};
export const GetTokenExchangeProfilesById200ResponseTypeEnum = {
    custom_authentication: 'custom_authentication',
};
export const HookCreateTriggerIdEnum = {
    credentials_exchange: 'credentials-exchange',
    pre_user_registration: 'pre-user-registration',
    post_user_registration: 'post-user-registration',
    post_change_password: 'post-change-password',
    send_phone_message: 'send-phone-message',
};
export const JobFormatEnum = {
    json: 'json',
    csv: 'csv',
};
/**
 * The verification status of the discovery domain.
 */
export const OrganizationDiscoveryDomainStatus = {
    pending: 'pending',
    verified: 'verified',
};
export const PatchAuthenticationMethodsByAuthenticationMethodIdRequestPreferredAuthenticationMethodEnum = {
    voice: 'voice',
    sms: 'sms',
};
export const PatchBindingsRequestBindingsInnerOneOfRefTypeEnum = {
    binding_id: 'binding_id',
    action_id: 'action_id',
    action_name: 'action_name',
};
export const PatchBreachedPasswordDetectionRequestShieldsEnum = {
    block: 'block',
    user_notification: 'user_notification',
    admin_notification: 'admin_notification',
};
export const PatchBreachedPasswordDetectionRequestAdminNotificationFrequencyEnum = {
    immediately: 'immediately',
    daily: 'daily',
    weekly: 'weekly',
    monthly: 'monthly',
};
export const PatchBreachedPasswordDetectionRequestMethodEnum = {
    standard: 'standard',
    enhanced: 'enhanced',
};
export const PatchBreachedPasswordDetectionRequestStagePreChangePasswordShieldsEnum = {
    block: 'block',
    admin_notification: 'admin_notification',
};
export const PatchBreachedPasswordDetectionRequestStagePreUserRegistrationShieldsEnum = {
    block: 'block',
    admin_notification: 'admin_notification',
};
export const PatchBruteForceProtectionRequestShieldsEnum = {
    block: 'block',
    user_notification: 'user_notification',
};
export const PatchBruteForceProtectionRequestModeEnum = {
    identifier_and_ip: 'count_per_identifier_and_ip',
    identifier: 'count_per_identifier',
};
export const PatchClientGrantsByIdRequestOrganizationUsageEnum = {
    deny: 'deny',
    allow: 'allow',
    require: 'require',
    null: 'null',
};
export const PatchCustomDomainsByIdRequestTlsPolicyEnum = {
    recommended: 'recommended',
};
export const PatchCustomDomainsByIdRequestCustomClientIpHeaderEnum = {
    true_client_ip: 'true-client-ip',
    cf_connecting_ip: 'cf-connecting-ip',
    x_forwarded_for: 'x-forwarded-for',
    x_azure_clientip: 'x-azure-clientip',
    empty: '',
};
export const PatchEmailTemplatesByTemplateNameRequestTemplateEnum = {
    verify_email: 'verify_email',
    verify_email_by_code: 'verify_email_by_code',
    reset_email: 'reset_email',
    reset_email_by_code: 'reset_email_by_code',
    welcome_email: 'welcome_email',
    blocked_account: 'blocked_account',
    stolen_credentials: 'stolen_credentials',
    enrollment_email: 'enrollment_email',
    mfa_oob_code: 'mfa_oob_code',
    user_invitation: 'user_invitation',
    change_password: 'change_password',
    password_reset: 'password_reset',
    async_approval: 'async_approval',
};
export const PatchLogStreamsByIdRequestStatusEnum = {
    active: 'active',
    paused: 'paused',
    suspended: 'suspended',
};
export const PatchLogStreamsByIdRequestSinkOneOfDatadogRegionEnum = {
    us: 'us',
    eu: 'eu',
    us3: 'us3',
    us5: 'us5',
};
export const PatchLogStreamsByIdRequestSinkOneOf3MixpanelRegionEnum = {
    us: 'us',
    eu: 'eu',
};
export const PatchNetworkAclsByIdRequestRuleScopeEnum = {
    management: 'management',
    authentication: 'authentication',
    tenant: 'tenant',
};
export const PatchNetworkAclsByIdRequestRuleActionBlockEnum = {
    true: true,
};
export const PatchNetworkAclsByIdRequestRuleActionAllowEnum = {
    true: true,
};
export const PatchNetworkAclsByIdRequestRuleActionLogEnum = {
    true: true,
};
export const PatchNetworkAclsByIdRequestRuleActionRedirectEnum = {
    true: true,
};
export const PatchRendering200ResponseRenderingModeEnum = {
    advanced: 'advanced',
    standard: 'standard',
};
export const PatchRenderingRequestRenderingModeEnum = {
    advanced: 'advanced',
    standard: 'standard',
};
export const PatchSuspiciousIpThrottlingRequestShieldsEnum = {
    block: 'block',
    admin_notification: 'admin_notification',
};
export const PhoneTemplateTypeEnum = {
    otp_verify: 'otp_verify',
    otp_enroll: 'otp_enroll',
    change_password: 'change_password',
    blocked_account: 'blocked_account',
    password_breach: 'password_breach',
};
export const PostAuthenticationMethods201ResponseTypeEnum = {
    phone: 'phone',
    email: 'email',
    totp: 'totp',
    webauthn_roaming: 'webauthn-roaming',
};
export const PostAuthenticationMethods201ResponsePreferredAuthenticationMethodEnum = {
    voice: 'voice',
    sms: 'sms',
};
export const PostAuthenticationMethodsRequestTypeEnum = {
    phone: 'phone',
    email: 'email',
    totp: 'totp',
    webauthn_roaming: 'webauthn-roaming',
};
export const PostAuthenticationMethodsRequestPreferredAuthenticationMethodEnum = {
    voice: 'voice',
    sms: 'sms',
};
export const PostBrandingThemeRequestBordersButtonsStyleEnum = {
    pill: 'pill',
    rounded: 'rounded',
    sharp: 'sharp',
};
export const PostBrandingThemeRequestBordersInputsStyleEnum = {
    pill: 'pill',
    rounded: 'rounded',
    sharp: 'sharp',
};
export const PostBrandingThemeRequestFontsLinksStyleEnum = {
    normal: 'normal',
    underlined: 'underlined',
};
export const PostBrandingThemeRequestPageBackgroundPageLayoutEnum = {
    center: 'center',
    left: 'left',
    right: 'right',
};
export const PostBrandingThemeRequestWidgetHeaderTextAlignmentEnum = {
    center: 'center',
    left: 'left',
    right: 'right',
};
export const PostBrandingThemeRequestWidgetLogoPositionEnum = {
    center: 'center',
    left: 'left',
    none: 'none',
    right: 'right',
};
export const PostBrandingThemeRequestWidgetSocialButtonsLayoutEnum = {
    bottom: 'bottom',
    top: 'top',
};
export const PostConnectionsKeysRotateResponseContentKeyUseEnum = {
    encryption: 'encryption',
    signing: 'signing',
};
export const PostCredentialsRequestCredentialTypeEnum = {
    public_key: 'public_key',
    cert_subject_dn: 'cert_subject_dn',
    x509_cert: 'x509_cert',
};
export const PostCustomDomains201ResponseStatusEnum = {
    disabled: 'disabled',
    pending: 'pending',
    pending_verification: 'pending_verification',
    ready: 'ready',
};
export const PostCustomDomains201ResponseTypeEnum = {
    auth0_managed_certs: 'auth0_managed_certs',
    self_managed_certs: 'self_managed_certs',
};
export const PostCustomDomains201ResponseVerificationStatusEnum = {
    verified: 'verified',
    pending: 'pending',
    failed: 'failed',
};
export const PostCustomDomains201ResponseVerificationMethodsInnerNameEnum = {
    cname: 'cname',
    txt: 'txt',
};
export const PostCustomDomainsRequestTypeEnum = {
    auth0_managed_certs: 'auth0_managed_certs',
    self_managed_certs: 'self_managed_certs',
};
export const PostCustomDomainsRequestVerificationMethodEnum = {
    txt: 'txt',
};
export const PostCustomDomainsRequestTlsPolicyEnum = {
    recommended: 'recommended',
};
export const PostCustomDomainsRequestCustomClientIpHeaderEnum = {
    true_client_ip: 'true-client-ip',
    cf_connecting_ip: 'cf-connecting-ip',
    x_forwarded_for: 'x-forwarded-for',
    x_azure_clientip: 'x-azure-clientip',
    null: 'null',
};
export const PostEmailTemplatesRequestTemplateEnum = {
    verify_email: 'verify_email',
    verify_email_by_code: 'verify_email_by_code',
    reset_email: 'reset_email',
    reset_email_by_code: 'reset_email_by_code',
    welcome_email: 'welcome_email',
    blocked_account: 'blocked_account',
    stolen_credentials: 'stolen_credentials',
    enrollment_email: 'enrollment_email',
    mfa_oob_code: 'mfa_oob_code',
    user_invitation: 'user_invitation',
    change_password: 'change_password',
    password_reset: 'password_reset',
    async_approval: 'async_approval',
};
export const PostEncryptionRequestTypeEnum = {
    customer_provided_root_key: 'customer-provided-root-key',
    tenant_encryption_key: 'tenant-encryption-key',
};
export const PostEncryptionWrappingKey201ResponseAlgorithmEnum = {
    CKM_RSA_AES_KEY_WRAP: 'CKM_RSA_AES_KEY_WRAP',
};
export const PostFlowsVaultConnectionsRequestAnyOfAppIdEnum = {
    ACTIVECAMPAIGN: 'ACTIVECAMPAIGN',
};
export const PostFlowsVaultConnectionsRequestAnyOf1AppIdEnum = {
    AIRTABLE: 'AIRTABLE',
};
export const PostFlowsVaultConnectionsRequestAnyOf10AppIdEnum = {
    MAILCHIMP: 'MAILCHIMP',
};
export const PostFlowsVaultConnectionsRequestAnyOf11AppIdEnum = {
    MAILJET: 'MAILJET',
};
export const PostFlowsVaultConnectionsRequestAnyOf11SetupTypeEnum = {
    API_KEY: 'API_KEY',
};
export const PostFlowsVaultConnectionsRequestAnyOf12AppIdEnum = {
    PIPEDRIVE: 'PIPEDRIVE',
};
export const PostFlowsVaultConnectionsRequestAnyOf12SetupAnyOfTypeEnum = {
    TOKEN: 'TOKEN',
};
export const PostFlowsVaultConnectionsRequestAnyOf13AppIdEnum = {
    SALESFORCE: 'SALESFORCE',
};
export const PostFlowsVaultConnectionsRequestAnyOf14AppIdEnum = {
    SENDGRID: 'SENDGRID',
};
export const PostFlowsVaultConnectionsRequestAnyOf15AppIdEnum = {
    SLACK: 'SLACK',
};
export const PostFlowsVaultConnectionsRequestAnyOf15SetupAnyOfTypeEnum = {
    WEBHOOK: 'WEBHOOK',
};
export const PostFlowsVaultConnectionsRequestAnyOf16AppIdEnum = {
    STRIPE: 'STRIPE',
};
export const PostFlowsVaultConnectionsRequestAnyOf16SetupAnyOfTypeEnum = {
    KEY_PAIR: 'KEY_PAIR',
};
export const PostFlowsVaultConnectionsRequestAnyOf16SetupAnyOf1TypeEnum = {
    OAUTH_CODE: 'OAUTH_CODE',
};
export const PostFlowsVaultConnectionsRequestAnyOf16SetupAnyOf1EnvironmentEnum = {
    live: 'live',
    test: 'test',
};
export const PostFlowsVaultConnectionsRequestAnyOf17AppIdEnum = {
    TELEGRAM: 'TELEGRAM',
};
export const PostFlowsVaultConnectionsRequestAnyOf18AppIdEnum = {
    TWILIO: 'TWILIO',
};
export const PostFlowsVaultConnectionsRequestAnyOf18SetupTypeEnum = {
    API_KEY: 'API_KEY',
};
export const PostFlowsVaultConnectionsRequestAnyOf19AppIdEnum = {
    WHATSAPP: 'WHATSAPP',
};
export const PostFlowsVaultConnectionsRequestAnyOf1SetupTypeEnum = {
    API_KEY: 'API_KEY',
};
export const PostFlowsVaultConnectionsRequestAnyOf2AppIdEnum = {
    AUTH0: 'AUTH0',
};
export const PostFlowsVaultConnectionsRequestAnyOf20AppIdEnum = {
    ZAPIER: 'ZAPIER',
};
export const PostFlowsVaultConnectionsRequestAnyOf2SetupTypeEnum = {
    OAUTH_APP: 'OAUTH_APP',
};
export const PostFlowsVaultConnectionsRequestAnyOf3AppIdEnum = {
    BIGQUERY: 'BIGQUERY',
};
export const PostFlowsVaultConnectionsRequestAnyOf3SetupTypeEnum = {
    OAUTH_JWT: 'OAUTH_JWT',
};
export const PostFlowsVaultConnectionsRequestAnyOf4AppIdEnum = {
    CLEARBIT: 'CLEARBIT',
};
export const PostFlowsVaultConnectionsRequestAnyOf4SetupTypeEnum = {
    API_KEY: 'API_KEY',
};
export const PostFlowsVaultConnectionsRequestAnyOf5AppIdEnum = {
    DOCUSIGN: 'DOCUSIGN',
};
export const PostFlowsVaultConnectionsRequestAnyOf5SetupTypeEnum = {
    OAUTH_CODE: 'OAUTH_CODE',
};
export const PostFlowsVaultConnectionsRequestAnyOf6AppIdEnum = {
    GOOGLE_SHEETS: 'GOOGLE_SHEETS',
};
export const PostFlowsVaultConnectionsRequestAnyOf7AppIdEnum = {
    HTTP: 'HTTP',
};
export const PostFlowsVaultConnectionsRequestAnyOf7SetupTypeEnum = {
    BEARER: 'BEARER',
};
export const PostFlowsVaultConnectionsRequestAnyOf8AppIdEnum = {
    HUBSPOT: 'HUBSPOT',
};
export const PostFlowsVaultConnectionsRequestAnyOf9AppIdEnum = {
    JWT: 'JWT',
};
export const PostFlowsVaultConnectionsRequestAnyOf9SetupTypeEnum = {
    JWT: 'JWT',
};
export const PostFlowsVaultConnectionsRequestAnyOf9SetupAlgorithmEnum = {
    HS256: 'HS256',
    HS384: 'HS384',
    HS512: 'HS512',
    RS256: 'RS256',
    RS384: 'RS384',
    RS512: 'RS512',
    ES256: 'ES256',
    ES384: 'ES384',
    ES512: 'ES512',
    PS256: 'PS256',
    PS384: 'PS384',
    PS512: 'PS512',
};
export const PostFlowsVaultConnectionsRequestAnyOfSetupTypeEnum = {
    API_KEY: 'API_KEY',
};
export const PostFormsRequestEndingResumeFlowEnum = {
    true: true,
};
export const PostFormsRequestNodesInnerAnyOfTypeEnum = {
    FLOW: 'FLOW',
};
export const PostFormsRequestNodesInnerAnyOf1TypeEnum = {
    ROUTER: 'ROUTER',
};
export const PostFormsRequestNodesInnerAnyOf2TypeEnum = {
    STEP: 'STEP',
};
export const PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf1AnyOfTypeEnum = {
    GMAPS_ADDRESS: 'GMAPS_ADDRESS',
};
export const PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf1AnyOf1TypeEnum = {
    RECAPTCHA: 'RECAPTCHA',
};
export const PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOfTypeEnum = {
    BOOLEAN: 'BOOLEAN',
};
export const PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf1TypeEnum = {
    CARDS: 'CARDS',
};
export const PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf10TypeEnum = {
    PASSWORD: 'PASSWORD',
};
export const PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf10ConfigHashEnum = {
    NONE: 'NONE',
    MD5: 'MD5',
    SHA1: 'SHA1',
    SHA256: 'SHA256',
    SHA512: 'SHA512',
};
export const PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf11TypeEnum = {
    PAYMENT: 'PAYMENT',
};
export const PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf11ConfigProviderEnum = {
    STRIPE: 'STRIPE',
};
export const PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf11ConfigChargeAnyOfTypeEnum = {
    ONE_OFF: 'ONE_OFF',
};
export const PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf11ConfigChargeAnyOf1TypeEnum = {
    SUBSCRIPTION: 'SUBSCRIPTION',
};
export const PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf11ConfigChargeAnyOfOneOffCurrencyEnum = {
    AUD: 'AUD',
    CAD: 'CAD',
    CHF: 'CHF',
    EUR: 'EUR',
    GBP: 'GBP',
    INR: 'INR',
    MXN: 'MXN',
    SEK: 'SEK',
    USD: 'USD',
};
export const PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf12TypeEnum = {
    SOCIAL: 'SOCIAL',
};
export const PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf13TypeEnum = {
    TEL: 'TEL',
};
export const PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf14TypeEnum = {
    TEXT: 'TEXT',
};
export const PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf15TypeEnum = {
    URL: 'URL',
};
export const PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf2TypeEnum = {
    CHOICE: 'CHOICE',
};
export const PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf3TypeEnum = {
    CUSTOM: 'CUSTOM',
};
export const PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf4TypeEnum = {
    DATE: 'DATE',
};
export const PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf5TypeEnum = {
    DROPDOWN: 'DROPDOWN',
};
export const PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf6TypeEnum = {
    EMAIL: 'EMAIL',
};
export const PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf7TypeEnum = {
    FILE: 'FILE',
};
export const PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf7ConfigCategoriesEnum = {
    AUDIO: 'AUDIO',
    VIDEO: 'VIDEO',
    IMAGE: 'IMAGE',
    DOCUMENT: 'DOCUMENT',
    ARCHIVE: 'ARCHIVE',
};
export const PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf8TypeEnum = {
    LEGAL: 'LEGAL',
};
export const PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOf2AnyOf9TypeEnum = {
    NUMBER: 'NUMBER',
};
export const PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOfAnyOfTypeEnum = {
    DIVIDER: 'DIVIDER',
};
export const PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOfAnyOf1TypeEnum = {
    HTML: 'HTML',
};
export const PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOfAnyOf2TypeEnum = {
    IMAGE: 'IMAGE',
};
export const PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOfAnyOf2ConfigPositionEnum = {
    LEFT: 'LEFT',
    CENTER: 'CENTER',
    RIGHT: 'RIGHT',
};
export const PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOfAnyOf3TypeEnum = {
    JUMP_BUTTON: 'JUMP_BUTTON',
};
export const PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOfAnyOf4TypeEnum = {
    NEXT_BUTTON: 'NEXT_BUTTON',
};
export const PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOfAnyOf5TypeEnum = {
    PREVIOUS_BUTTON: 'PREVIOUS_BUTTON',
};
export const PostFormsRequestNodesInnerAnyOf2ConfigComponentsInnerAnyOfAnyOf6TypeEnum = {
    RICH_TEXT: 'RICH_TEXT',
};
/**
 *
 */
export const PostFormsRequestNodesInnerAnyOfConfigNextNodeAnyOf = {
    ending: '$ending',
};
export const PostIdentitiesRequestProviderEnum = {
    ad: 'ad',
    adfs: 'adfs',
    amazon: 'amazon',
    apple: 'apple',
    dropbox: 'dropbox',
    bitbucket: 'bitbucket',
    aol: 'aol',
    auth0_oidc: 'auth0-oidc',
    auth0: 'auth0',
    baidu: 'baidu',
    bitly: 'bitly',
    box: 'box',
    custom: 'custom',
    daccount: 'daccount',
    dwolla: 'dwolla',
    email: 'email',
    evernote_sandbox: 'evernote-sandbox',
    evernote: 'evernote',
    exact: 'exact',
    facebook: 'facebook',
    fitbit: 'fitbit',
    flickr: 'flickr',
    github: 'github',
    google_apps: 'google-apps',
    google_oauth2: 'google-oauth2',
    instagram: 'instagram',
    ip: 'ip',
    line: 'line',
    linkedin: 'linkedin',
    miicard: 'miicard',
    oauth1: 'oauth1',
    oauth2: 'oauth2',
    office365: 'office365',
    oidc: 'oidc',
    okta: 'okta',
    paypal: 'paypal',
    paypal_sandbox: 'paypal-sandbox',
    pingfederate: 'pingfederate',
    planningcenter: 'planningcenter',
    renren: 'renren',
    salesforce_community: 'salesforce-community',
    salesforce_sandbox: 'salesforce-sandbox',
    salesforce: 'salesforce',
    samlp: 'samlp',
    sharepoint: 'sharepoint',
    shopify: 'shopify',
    sms: 'sms',
    soundcloud: 'soundcloud',
    thecity_sandbox: 'thecity-sandbox',
    thecity: 'thecity',
    thirtysevensignals: 'thirtysevensignals',
    twitter: 'twitter',
    untappd: 'untappd',
    vkontakte: 'vkontakte',
    waad: 'waad',
    weibo: 'weibo',
    windowslive: 'windowslive',
    wordpress: 'wordpress',
    yahoo: 'yahoo',
    yammer: 'yammer',
    yandex: 'yandex',
};
export const PostLogStreamsRequestOneOfTypeEnum = {
    http: 'http',
};
export const PostLogStreamsRequestOneOf1TypeEnum = {
    eventbridge: 'eventbridge',
};
export const PostLogStreamsRequestOneOf1SinkAwsRegionEnum = {
    ap_east_1: 'ap-east-1',
    ap_northeast_1: 'ap-northeast-1',
    ap_northeast_2: 'ap-northeast-2',
    ap_northeast_3: 'ap-northeast-3',
    ap_south_1: 'ap-south-1',
    ap_southeast_1: 'ap-southeast-1',
    ap_southeast_2: 'ap-southeast-2',
    ca_central_1: 'ca-central-1',
    cn_north_1: 'cn-north-1',
    cn_northwest_1: 'cn-northwest-1',
    eu_central_1: 'eu-central-1',
    eu_north_1: 'eu-north-1',
    eu_west_1: 'eu-west-1',
    eu_west_2: 'eu-west-2',
    eu_west_3: 'eu-west-3',
    me_south_1: 'me-south-1',
    sa_east_1: 'sa-east-1',
    us_gov_east_1: 'us-gov-east-1',
    us_gov_west_1: 'us-gov-west-1',
    us_east_1: 'us-east-1',
    us_east_2: 'us-east-2',
    us_west_1: 'us-west-1',
    us_west_2: 'us-west-2',
};
export const PostLogStreamsRequestOneOf2TypeEnum = {
    eventgrid: 'eventgrid',
};
export const PostLogStreamsRequestOneOf2SinkAzureRegionEnum = {
    australiacentral: 'australiacentral',
    australiaeast: 'australiaeast',
    australiasoutheast: 'australiasoutheast',
    brazilsouth: 'brazilsouth',
    canadacentral: 'canadacentral',
    canadaeast: 'canadaeast',
    centralindia: 'centralindia',
    centralus: 'centralus',
    eastasia: 'eastasia',
    eastus: 'eastus',
    eastus2: 'eastus2',
    francecentral: 'francecentral',
    germanywestcentral: 'germanywestcentral',
    japaneast: 'japaneast',
    japanwest: 'japanwest',
    koreacentral: 'koreacentral',
    koreasouth: 'koreasouth',
    northcentralus: 'northcentralus',
    northeurope: 'northeurope',
    norwayeast: 'norwayeast',
    southafricanorth: 'southafricanorth',
    southcentralus: 'southcentralus',
    southeastasia: 'southeastasia',
    southindia: 'southindia',
    switzerlandnorth: 'switzerlandnorth',
    uaenorth: 'uaenorth',
    uksouth: 'uksouth',
    ukwest: 'ukwest',
    westcentralus: 'westcentralus',
    westeurope: 'westeurope',
    westindia: 'westindia',
    westus: 'westus',
    westus2: 'westus2',
};
export const PostLogStreamsRequestOneOf3TypeEnum = {
    datadog: 'datadog',
};
export const PostLogStreamsRequestOneOf4TypeEnum = {
    splunk: 'splunk',
};
export const PostLogStreamsRequestOneOf5TypeEnum = {
    sumo: 'sumo',
};
export const PostLogStreamsRequestOneOf6TypeEnum = {
    segment: 'segment',
};
export const PostLogStreamsRequestOneOf7TypeEnum = {
    mixpanel: 'mixpanel',
};
export const PostLogStreamsRequestOneOfFiltersInnerTypeEnum = {
    category: 'category',
};
export const PostLogStreamsRequestOneOfFiltersInnerNameEnum = {
    auth_ancillary_fail: 'auth.ancillary.fail',
    auth_ancillary_success: 'auth.ancillary.success',
    auth_login_fail: 'auth.login.fail',
    auth_login_notification: 'auth.login.notification',
    auth_login_success: 'auth.login.success',
    auth_logout_fail: 'auth.logout.fail',
    auth_logout_success: 'auth.logout.success',
    auth_signup_fail: 'auth.signup.fail',
    auth_signup_success: 'auth.signup.success',
    auth_silent_auth_fail: 'auth.silent_auth.fail',
    auth_silent_auth_success: 'auth.silent_auth.success',
    auth_token_exchange_fail: 'auth.token_exchange.fail',
    auth_token_exchange_success: 'auth.token_exchange.success',
    management_fail: 'management.fail',
    management_success: 'management.success',
    system_notification: 'system.notification',
    user_fail: 'user.fail',
    user_notification: 'user.notification',
    user_success: 'user.success',
    other: 'other',
};
export const PostSsoTicketRequestConnectionConfigOptionsIdpinitiatedClientProtocolEnum = {
    samlp: 'samlp',
    wsfed: 'wsfed',
    oauth2: 'oauth2',
};
export const PostSsoTicketRequestDomainAliasesConfigDomainVerificationEnum = {
    none: 'none',
    optional: 'optional',
    required: 'required',
};
export const PostSsoTicketRequestProvisioningConfigScopesEnum = {
    getusers: 'get:users',
    postusers: 'post:users',
    putusers: 'put:users',
    patchusers: 'patch:users',
    deleteusers: 'delete:users',
};
export const PostTokenExchangeProfilesRequestTypeEnum = {
    custom_authentication: 'custom_authentication',
};
export const PostUsersExportsRequestFormatEnum = {
    json: 'json',
    csv: 'csv',
};
export const PostVerificationEmailRequestIdentityProviderEnum = {
    ad: 'ad',
    adfs: 'adfs',
    amazon: 'amazon',
    apple: 'apple',
    dropbox: 'dropbox',
    bitbucket: 'bitbucket',
    aol: 'aol',
    auth0_oidc: 'auth0-oidc',
    auth0: 'auth0',
    baidu: 'baidu',
    bitly: 'bitly',
    box: 'box',
    custom: 'custom',
    daccount: 'daccount',
    dwolla: 'dwolla',
    email: 'email',
    evernote_sandbox: 'evernote-sandbox',
    evernote: 'evernote',
    exact: 'exact',
    facebook: 'facebook',
    fitbit: 'fitbit',
    flickr: 'flickr',
    github: 'github',
    google_apps: 'google-apps',
    google_oauth2: 'google-oauth2',
    instagram: 'instagram',
    ip: 'ip',
    line: 'line',
    linkedin: 'linkedin',
    miicard: 'miicard',
    oauth1: 'oauth1',
    oauth2: 'oauth2',
    office365: 'office365',
    oidc: 'oidc',
    okta: 'okta',
    paypal: 'paypal',
    paypal_sandbox: 'paypal-sandbox',
    pingfederate: 'pingfederate',
    planningcenter: 'planningcenter',
    renren: 'renren',
    salesforce_community: 'salesforce-community',
    salesforce_sandbox: 'salesforce-sandbox',
    salesforce: 'salesforce',
    samlp: 'samlp',
    sharepoint: 'sharepoint',
    shopify: 'shopify',
    sms: 'sms',
    soundcloud: 'soundcloud',
    thecity_sandbox: 'thecity-sandbox',
    thecity: 'thecity',
    thirtysevensignals: 'thirtysevensignals',
    twitter: 'twitter',
    untappd: 'untappd',
    vkontakte: 'vkontakte',
    waad: 'waad',
    weibo: 'weibo',
    windowslive: 'windowslive',
    wordpress: 'wordpress',
    yahoo: 'yahoo',
    yammer: 'yammer',
    yandex: 'yandex',
};
export const PostVerify200ResponseStatusEnum = {
    disabled: 'disabled',
    pending: 'pending',
    pending_verification: 'pending_verification',
    ready: 'ready',
};
export const PostVerify200ResponseTypeEnum = {
    auth0_managed_certs: 'auth0_managed_certs',
    self_managed_certs: 'self_managed_certs',
};
export const PromptsSettingsUniversalLoginExperienceEnum = {
    new: 'new',
    classic: 'classic',
};
export const PromptsSettingsUpdateUniversalLoginExperienceEnum = {
    new: 'new',
    classic: 'classic',
};
export const PutAuthenticationMethods200ResponseInnerTypeEnum = {
    phone: 'phone',
    email: 'email',
    totp: 'totp',
    webauthn_roaming: 'webauthn-roaming',
};
export const PutAuthenticationMethods200ResponseInnerPreferredAuthenticationMethodEnum = {
    voice: 'voice',
    sms: 'sms',
};
export const PutAuthenticationMethods200ResponseInnerAuthenticationMethodsInnerTypeEnum = {
    totp: 'totp',
    push: 'push',
    sms: 'sms',
    voice: 'voice',
};
export const PutAuthenticationMethodsRequestInnerTypeEnum = {
    phone: 'phone',
    email: 'email',
    totp: 'totp',
};
export const PutAuthenticationMethodsRequestInnerPreferredAuthenticationMethodEnum = {
    voice: 'voice',
    sms: 'sms',
};
export const PutNetworkAclsByIdRequestRuleAnyOfScopeEnum = {
    management: 'management',
    authentication: 'authentication',
    tenant: 'tenant',
};
export const PutNetworkAclsByIdRequestRuleAnyOf1ScopeEnum = {
    management: 'management',
    authentication: 'authentication',
    tenant: 'tenant',
};
export const ResetPhoneTemplateResponseContentTypeEnum = {
    otp_verify: 'otp_verify',
    otp_enroll: 'otp_enroll',
    change_password: 'change_password',
    blocked_account: 'blocked_account',
    password_breach: 'password_breach',
};
export const ResourceServerSigningAlgEnum = {
    HS256: 'HS256',
    RS256: 'RS256',
    PS256: 'PS256',
};
export const ResourceServerTokenDialectEnum = {
    access_token: 'access_token',
    access_token_authz: 'access_token_authz',
    rfc9068_profile: 'rfc9068_profile',
    rfc9068_profile_authz: 'rfc9068_profile_authz',
};
export const ResourceServerConsentPolicyEnum = {
    transactional_authorization_with_mfa: 'transactional-authorization-with-mfa',
    null: 'null',
};
export const ResourceServerCreateSigningAlgEnum = {
    HS256: 'HS256',
    RS256: 'RS256',
    PS256: 'PS256',
};
export const ResourceServerCreateTokenDialectEnum = {
    access_token: 'access_token',
    access_token_authz: 'access_token_authz',
    rfc9068_profile: 'rfc9068_profile',
    rfc9068_profile_authz: 'rfc9068_profile_authz',
};
export const ResourceServerCreateConsentPolicyEnum = {
    transactional_authorization_with_mfa: 'transactional-authorization-with-mfa',
    null: 'null',
};
export const ResourceServerProofOfPossessionMechanismEnum = {
    mtls: 'mtls',
    dpop: 'dpop',
};
export const ResourceServerSubjectTypeAuthorizationClientPolicyEnum = {
    deny_all: 'deny_all',
    require_client_grant: 'require_client_grant',
};
export const ResourceServerSubjectTypeAuthorizationUserPolicyEnum = {
    allow_all: 'allow_all',
    deny_all: 'deny_all',
    require_client_grant: 'require_client_grant',
};
export const ResourceServerTokenEncryptionFormatEnum = {
    compact_nested_jwe: 'compact-nested-jwe',
};
export const ResourceServerTokenEncryptionEncryptionKeyAlgEnum = {
    _256: 'RSA-OAEP-256',
    _384: 'RSA-OAEP-384',
    _512: 'RSA-OAEP-512',
};
export const ResourceServerUpdateSigningAlgEnum = {
    HS256: 'HS256',
    RS256: 'RS256',
    PS256: 'PS256',
};
export const ResourceServerUpdateTokenDialectEnum = {
    access_token: 'access_token',
    access_token_authz: 'access_token_authz',
    rfc9068_profile: 'rfc9068_profile',
    rfc9068_profile_authz: 'rfc9068_profile_authz',
};
export const ResourceServerUpdateConsentPolicyEnum = {
    transactional_authorization_with_mfa: 'transactional-authorization-with-mfa',
    null: 'null',
};
/**
 *
 */
export const SelfServiceProfileSsoTicketProvisioningScopeEnum = {
    getusers: 'get:users',
    postusers: 'post:users',
    putusers: 'put:users',
    patchusers: 'patch:users',
    deleteusers: 'delete:users',
};
export const SsProfileAllowedStrategiesEnum = {
    oidc: 'oidc',
    samlp: 'samlp',
    waad: 'waad',
    google_apps: 'google-apps',
    adfs: 'adfs',
    okta: 'okta',
    keycloak_samlp: 'keycloak-samlp',
    pingfederate: 'pingfederate',
};
export const SsProfileCreateAllowedStrategiesEnum = {
    oidc: 'oidc',
    samlp: 'samlp',
    waad: 'waad',
    google_apps: 'google-apps',
    adfs: 'adfs',
    okta: 'okta',
    keycloak_samlp: 'keycloak-samlp',
    pingfederate: 'pingfederate',
};
export const SsProfileUpdateAllowedStrategiesEnum = {
    oidc: 'oidc',
    samlp: 'samlp',
    waad: 'waad',
    google_apps: 'google-apps',
    adfs: 'adfs',
    okta: 'okta',
    keycloak_samlp: 'keycloak-samlp',
    pingfederate: 'pingfederate',
};
export const TenantSettingsEnabledLocalesEnum = {
    am: 'am',
    ar: 'ar',
    ar_EG: 'ar-EG',
    ar_SA: 'ar-SA',
    az: 'az',
    bg: 'bg',
    bn: 'bn',
    bs: 'bs',
    ca_ES: 'ca-ES',
    cnr: 'cnr',
    cs: 'cs',
    cy: 'cy',
    da: 'da',
    de: 'de',
    el: 'el',
    en: 'en',
    en_CA: 'en-CA',
    es: 'es',
    es_419: 'es-419',
    es_AR: 'es-AR',
    es_MX: 'es-MX',
    et: 'et',
    eu_ES: 'eu-ES',
    fa: 'fa',
    fi: 'fi',
    fr: 'fr',
    fr_CA: 'fr-CA',
    fr_FR: 'fr-FR',
    gl_ES: 'gl-ES',
    gu: 'gu',
    he: 'he',
    hi: 'hi',
    hr: 'hr',
    hu: 'hu',
    hy: 'hy',
    id: 'id',
    is: 'is',
    it: 'it',
    ja: 'ja',
    ka: 'ka',
    kk: 'kk',
    kn: 'kn',
    ko: 'ko',
    lt: 'lt',
    lv: 'lv',
    mk: 'mk',
    ml: 'ml',
    mn: 'mn',
    mr: 'mr',
    ms: 'ms',
    my: 'my',
    nb: 'nb',
    nl: 'nl',
    nn: 'nn',
    no: 'no',
    pa: 'pa',
    pl: 'pl',
    pt: 'pt',
    pt_BR: 'pt-BR',
    pt_PT: 'pt-PT',
    ro: 'ro',
    ru: 'ru',
    sk: 'sk',
    sl: 'sl',
    so: 'so',
    sq: 'sq',
    sr: 'sr',
    sv: 'sv',
    sw: 'sw',
    ta: 'ta',
    te: 'te',
    th: 'th',
    tl: 'tl',
    tr: 'tr',
    uk: 'uk',
    ur: 'ur',
    vi: 'vi',
    zgh: 'zgh',
    zh_CN: 'zh-CN',
    zh_HK: 'zh-HK',
    zh_TW: 'zh-TW',
};
export const TenantSettingsDeviceFlowCharsetEnum = {
    base20: 'base20',
    digits: 'digits',
};
export const TenantSettingsSessionCookieModeEnum = {
    persistent: 'persistent',
    non_persistent: 'non-persistent',
};
export const TenantSettingsUpdateEnabledLocalesEnum = {
    am: 'am',
    ar: 'ar',
    ar_EG: 'ar-EG',
    ar_SA: 'ar-SA',
    az: 'az',
    bg: 'bg',
    bn: 'bn',
    bs: 'bs',
    ca_ES: 'ca-ES',
    cnr: 'cnr',
    cs: 'cs',
    cy: 'cy',
    da: 'da',
    de: 'de',
    el: 'el',
    en: 'en',
    en_CA: 'en-CA',
    es: 'es',
    es_419: 'es-419',
    es_AR: 'es-AR',
    es_MX: 'es-MX',
    et: 'et',
    eu_ES: 'eu-ES',
    fa: 'fa',
    fi: 'fi',
    fr: 'fr',
    fr_CA: 'fr-CA',
    fr_FR: 'fr-FR',
    gl_ES: 'gl-ES',
    gu: 'gu',
    he: 'he',
    hi: 'hi',
    hr: 'hr',
    hu: 'hu',
    hy: 'hy',
    id: 'id',
    is: 'is',
    it: 'it',
    ja: 'ja',
    ka: 'ka',
    kk: 'kk',
    kn: 'kn',
    ko: 'ko',
    lt: 'lt',
    lv: 'lv',
    mk: 'mk',
    ml: 'ml',
    mn: 'mn',
    mr: 'mr',
    ms: 'ms',
    my: 'my',
    nb: 'nb',
    nl: 'nl',
    nn: 'nn',
    no: 'no',
    pa: 'pa',
    pl: 'pl',
    pt: 'pt',
    pt_BR: 'pt-BR',
    pt_PT: 'pt-PT',
    ro: 'ro',
    ru: 'ru',
    sk: 'sk',
    sl: 'sl',
    so: 'so',
    sq: 'sq',
    sr: 'sr',
    sv: 'sv',
    sw: 'sw',
    ta: 'ta',
    te: 'te',
    th: 'th',
    tl: 'tl',
    tr: 'tr',
    uk: 'uk',
    ur: 'ur',
    vi: 'vi',
    zgh: 'zgh',
    zh_CN: 'zh-CN',
    zh_HK: 'zh-HK',
    zh_TW: 'zh-TW',
};
export const TenantSettingsUpdateDeviceFlowCharsetEnum = {
    base20: 'base20',
    digits: 'digits',
};
export const TenantSettingsUpdateFlagsChangePwdFlowV1Enum = {
    false: false,
};
export const UpdatePhoneProviderRequestNameEnum = {
    twilio: 'twilio',
    custom: 'custom',
};
export const UpdatePhoneTemplateResponseContentTypeEnum = {
    otp_verify: 'otp_verify',
    otp_enroll: 'otp_enroll',
    change_password: 'change_password',
    blocked_account: 'blocked_account',
    password_breach: 'password_breach',
};
/**
 * OIDC mapping for user ID
 */
export const UserAttributeProfileUserIdOidcMappingEnum = {
    sub: 'sub',
};
/**
 * OIDC mapping override for this strategy
 */
export const UserAttributeProfileUserIdOidcStrategyOverrideMapping = {
    sub: 'sub',
    oid: 'oid',
    email: 'email',
};
export const UserEnrollmentStatusEnum = {
    pending: 'pending',
    confirmed: 'confirmed',
};
export const UserEnrollmentAuthMethodEnum = {
    authenticator: 'authenticator',
    guardian: 'guardian',
    sms: 'sms',
    webauthn_platform: 'webauthn-platform',
    webauthn_roaming: 'webauthn-roaming',
};
/**
 *
 */
export const GetClientGrantsAllowAnyOrganizationEnum = {
    true: true,
};
/**
 *
 */
export const GetClientGrantsSubjectTypeEnum = {
    client: 'client',
    user: 'user',
};
/**
 *
 */
export const GetClientConnectionsStrategyEnum = {
    ad: 'ad',
    adfs: 'adfs',
    amazon: 'amazon',
    apple: 'apple',
    dropbox: 'dropbox',
    bitbucket: 'bitbucket',
    aol: 'aol',
    auth0_oidc: 'auth0-oidc',
    auth0: 'auth0',
    baidu: 'baidu',
    bitly: 'bitly',
    box: 'box',
    custom: 'custom',
    daccount: 'daccount',
    dwolla: 'dwolla',
    email: 'email',
    evernote_sandbox: 'evernote-sandbox',
    evernote: 'evernote',
    exact: 'exact',
    facebook: 'facebook',
    fitbit: 'fitbit',
    flickr: 'flickr',
    github: 'github',
    google_apps: 'google-apps',
    google_oauth2: 'google-oauth2',
    instagram: 'instagram',
    ip: 'ip',
    line: 'line',
    linkedin: 'linkedin',
    miicard: 'miicard',
    oauth1: 'oauth1',
    oauth2: 'oauth2',
    office365: 'office365',
    oidc: 'oidc',
    okta: 'okta',
    paypal: 'paypal',
    paypal_sandbox: 'paypal-sandbox',
    pingfederate: 'pingfederate',
    planningcenter: 'planningcenter',
    renren: 'renren',
    salesforce_community: 'salesforce-community',
    salesforce_sandbox: 'salesforce-sandbox',
    salesforce: 'salesforce',
    samlp: 'samlp',
    sharepoint: 'sharepoint',
    shopify: 'shopify',
    sms: 'sms',
    soundcloud: 'soundcloud',
    thecity_sandbox: 'thecity-sandbox',
    thecity: 'thecity',
    thirtysevensignals: 'thirtysevensignals',
    twitter: 'twitter',
    untappd: 'untappd',
    vkontakte: 'vkontakte',
    waad: 'waad',
    weibo: 'weibo',
    windowslive: 'windowslive',
    wordpress: 'wordpress',
    yahoo: 'yahoo',
    yammer: 'yammer',
    yandex: 'yandex',
};
/**
 *
 */
export const GetConnectionsStrategyEnum = {
    ad: 'ad',
    adfs: 'adfs',
    amazon: 'amazon',
    apple: 'apple',
    dropbox: 'dropbox',
    bitbucket: 'bitbucket',
    aol: 'aol',
    auth0_oidc: 'auth0-oidc',
    auth0: 'auth0',
    baidu: 'baidu',
    bitly: 'bitly',
    box: 'box',
    custom: 'custom',
    daccount: 'daccount',
    dwolla: 'dwolla',
    email: 'email',
    evernote_sandbox: 'evernote-sandbox',
    evernote: 'evernote',
    exact: 'exact',
    facebook: 'facebook',
    fitbit: 'fitbit',
    flickr: 'flickr',
    github: 'github',
    google_apps: 'google-apps',
    google_oauth2: 'google-oauth2',
    instagram: 'instagram',
    ip: 'ip',
    line: 'line',
    linkedin: 'linkedin',
    miicard: 'miicard',
    oauth1: 'oauth1',
    oauth2: 'oauth2',
    office365: 'office365',
    oidc: 'oidc',
    okta: 'okta',
    paypal: 'paypal',
    paypal_sandbox: 'paypal-sandbox',
    pingfederate: 'pingfederate',
    planningcenter: 'planningcenter',
    renren: 'renren',
    salesforce_community: 'salesforce-community',
    salesforce_sandbox: 'salesforce-sandbox',
    salesforce: 'salesforce',
    samlp: 'samlp',
    sharepoint: 'sharepoint',
    shopify: 'shopify',
    sms: 'sms',
    soundcloud: 'soundcloud',
    thecity_sandbox: 'thecity-sandbox',
    thecity: 'thecity',
    thirtysevensignals: 'thirtysevensignals',
    twitter: 'twitter',
    untappd: 'untappd',
    vkontakte: 'vkontakte',
    waad: 'waad',
    weibo: 'weibo',
    windowslive: 'windowslive',
    wordpress: 'wordpress',
    yahoo: 'yahoo',
    yammer: 'yammer',
    yandex: 'yandex',
    auth0_adldap: 'auth0-adldap',
};
/**
 *
 */
export const GetDeviceCredentialsTypeEnum = {
    public_key: 'public_key',
    refresh_token: 'refresh_token',
    rotating_refresh_token: 'rotating_refresh_token',
};
/**
 *
 */
export const GetEmailTemplatesByTemplateNameTemplateNameEnum = {
    verify_email: 'verify_email',
    verify_email_by_code: 'verify_email_by_code',
    reset_email: 'reset_email',
    reset_email_by_code: 'reset_email_by_code',
    welcome_email: 'welcome_email',
    blocked_account: 'blocked_account',
    stolen_credentials: 'stolen_credentials',
    enrollment_email: 'enrollment_email',
    mfa_oob_code: 'mfa_oob_code',
    user_invitation: 'user_invitation',
    change_password: 'change_password',
    password_reset: 'password_reset',
    async_approval: 'async_approval',
};
/**
 *
 */
export const PatchEmailTemplatesByTemplateNameOperationTemplateNameEnum = {
    verify_email: 'verify_email',
    verify_email_by_code: 'verify_email_by_code',
    reset_email: 'reset_email',
    reset_email_by_code: 'reset_email_by_code',
    welcome_email: 'welcome_email',
    blocked_account: 'blocked_account',
    stolen_credentials: 'stolen_credentials',
    enrollment_email: 'enrollment_email',
    mfa_oob_code: 'mfa_oob_code',
    user_invitation: 'user_invitation',
    change_password: 'change_password',
    password_reset: 'password_reset',
    async_approval: 'async_approval',
};
/**
 *
 */
export const PutEmailTemplatesByTemplateNameTemplateNameEnum = {
    verify_email: 'verify_email',
    verify_email_by_code: 'verify_email_by_code',
    reset_email: 'reset_email',
    reset_email_by_code: 'reset_email_by_code',
    welcome_email: 'welcome_email',
    blocked_account: 'blocked_account',
    stolen_credentials: 'stolen_credentials',
    enrollment_email: 'enrollment_email',
    mfa_oob_code: 'mfa_oob_code',
    user_invitation: 'user_invitation',
    change_password: 'change_password',
    password_reset: 'password_reset',
    async_approval: 'async_approval',
};
/**
 *
 */
export const GetFlowsHydrateEnum = {
    form_count: 'form_count',
};
/**
 *
 */
export const GetFlowsByIdHydrateEnum = {
    form_count: 'form_count',
};
/**
 *
 */
export const GetFlowsExecutionsByExecutionIdHydrateEnum = {
    debug: 'debug',
};
/**
 *
 */
export const GetFormsHydrateEnum = {
    flow_count: 'flow_count',
    links: 'links',
};
/**
 *
 */
export const GetFormsByIdHydrateEnum = {
    flow_count: 'flow_count',
    links: 'links',
};
/**
 *
 */
export const PutFactorsByNameOperationNameEnum = {
    push_notification: 'push-notification',
    sms: 'sms',
    email: 'email',
    duo: 'duo',
    otp: 'otp',
    webauthn_roaming: 'webauthn-roaming',
    webauthn_platform: 'webauthn-platform',
    recovery_code: 'recovery-code',
};
/**
 *
 */
export const GetHooksTriggerIdEnum = {
    credentials_exchange: 'credentials-exchange',
    pre_user_registration: 'pre-user-registration',
    post_user_registration: 'post-user-registration',
    post_change_password: 'post-change-password',
    send_phone_message: 'send-phone-message',
};
/**
 *
 */
export const GetAllRenderingRenderingModeEnum = {
    advanced: 'advanced',
    standard: 'standard',
};
/**
 *
 */
export const GetCustomTextByLanguagePromptEnum = {
    login: 'login',
    login_id: 'login-id',
    login_password: 'login-password',
    login_passwordless: 'login-passwordless',
    login_email_verification: 'login-email-verification',
    signup: 'signup',
    signup_id: 'signup-id',
    signup_password: 'signup-password',
    phone_identifier_enrollment: 'phone-identifier-enrollment',
    phone_identifier_challenge: 'phone-identifier-challenge',
    email_identifier_challenge: 'email-identifier-challenge',
    reset_password: 'reset-password',
    custom_form: 'custom-form',
    consent: 'consent',
    customized_consent: 'customized-consent',
    logout: 'logout',
    mfa_push: 'mfa-push',
    mfa_otp: 'mfa-otp',
    mfa_voice: 'mfa-voice',
    mfa_phone: 'mfa-phone',
    mfa_webauthn: 'mfa-webauthn',
    mfa_sms: 'mfa-sms',
    mfa_email: 'mfa-email',
    mfa_recovery_code: 'mfa-recovery-code',
    mfa: 'mfa',
    status: 'status',
    device_flow: 'device-flow',
    email_verification: 'email-verification',
    email_otp_challenge: 'email-otp-challenge',
    organizations: 'organizations',
    invitation: 'invitation',
    common: 'common',
    passkeys: 'passkeys',
    captcha: 'captcha',
};
/**
 *
 */
export const GetCustomTextByLanguageLanguageEnum = {
    am: 'am',
    ar: 'ar',
    ar_EG: 'ar-EG',
    ar_SA: 'ar-SA',
    az: 'az',
    bg: 'bg',
    bn: 'bn',
    bs: 'bs',
    ca_ES: 'ca-ES',
    cnr: 'cnr',
    cs: 'cs',
    cy: 'cy',
    da: 'da',
    de: 'de',
    el: 'el',
    en: 'en',
    en_CA: 'en-CA',
    es: 'es',
    es_419: 'es-419',
    es_AR: 'es-AR',
    es_MX: 'es-MX',
    et: 'et',
    eu_ES: 'eu-ES',
    fa: 'fa',
    fi: 'fi',
    fr: 'fr',
    fr_CA: 'fr-CA',
    fr_FR: 'fr-FR',
    gl_ES: 'gl-ES',
    gu: 'gu',
    he: 'he',
    hi: 'hi',
    hr: 'hr',
    hu: 'hu',
    hy: 'hy',
    id: 'id',
    is: 'is',
    it: 'it',
    ja: 'ja',
    ka: 'ka',
    kk: 'kk',
    kn: 'kn',
    ko: 'ko',
    lt: 'lt',
    lv: 'lv',
    mk: 'mk',
    ml: 'ml',
    mn: 'mn',
    mr: 'mr',
    ms: 'ms',
    my: 'my',
    nb: 'nb',
    nl: 'nl',
    nn: 'nn',
    no: 'no',
    pa: 'pa',
    pl: 'pl',
    pt: 'pt',
    pt_BR: 'pt-BR',
    pt_PT: 'pt-PT',
    ro: 'ro',
    ru: 'ru',
    sk: 'sk',
    sl: 'sl',
    so: 'so',
    sq: 'sq',
    sr: 'sr',
    sv: 'sv',
    sw: 'sw',
    ta: 'ta',
    te: 'te',
    th: 'th',
    tl: 'tl',
    tr: 'tr',
    uk: 'uk',
    ur: 'ur',
    vi: 'vi',
    zgh: 'zgh',
    zh_CN: 'zh-CN',
    zh_HK: 'zh-HK',
    zh_TW: 'zh-TW',
};
/**
 *
 */
export const GetPartialsPromptEnum = {
    login: 'login',
    login_id: 'login-id',
    login_password: 'login-password',
    login_passwordless: 'login-passwordless',
    signup: 'signup',
    signup_id: 'signup-id',
    signup_password: 'signup-password',
    customized_consent: 'customized-consent',
};
/**
 *
 */
export const GetRenderingPromptEnum = {
    login: 'login',
    login_id: 'login-id',
    login_password: 'login-password',
    login_passwordless: 'login-passwordless',
    login_email_verification: 'login-email-verification',
    signup: 'signup',
    signup_id: 'signup-id',
    signup_password: 'signup-password',
    phone_identifier_enrollment: 'phone-identifier-enrollment',
    phone_identifier_challenge: 'phone-identifier-challenge',
    email_identifier_challenge: 'email-identifier-challenge',
    reset_password: 'reset-password',
    custom_form: 'custom-form',
    consent: 'consent',
    customized_consent: 'customized-consent',
    logout: 'logout',
    mfa_push: 'mfa-push',
    mfa_otp: 'mfa-otp',
    mfa_voice: 'mfa-voice',
    mfa_phone: 'mfa-phone',
    mfa_webauthn: 'mfa-webauthn',
    mfa_sms: 'mfa-sms',
    mfa_email: 'mfa-email',
    mfa_recovery_code: 'mfa-recovery-code',
    mfa: 'mfa',
    status: 'status',
    device_flow: 'device-flow',
    email_verification: 'email-verification',
    email_otp_challenge: 'email-otp-challenge',
    organizations: 'organizations',
    invitation: 'invitation',
    common: 'common',
    passkeys: 'passkeys',
    captcha: 'captcha',
};
/**
 *
 */
export const GetRenderingScreenEnum = {
    login: 'login',
    login_id: 'login-id',
    login_password: 'login-password',
    login_passwordless_email_code: 'login-passwordless-email-code',
    login_passwordless_email_link: 'login-passwordless-email-link',
    login_passwordless_sms_otp: 'login-passwordless-sms-otp',
    login_email_verification: 'login-email-verification',
    signup: 'signup',
    signup_id: 'signup-id',
    signup_password: 'signup-password',
    phone_identifier_enrollment: 'phone-identifier-enrollment',
    phone_identifier_challenge: 'phone-identifier-challenge',
    email_identifier_challenge: 'email-identifier-challenge',
    reset_password_request: 'reset-password-request',
    reset_password_email: 'reset-password-email',
    reset_password: 'reset-password',
    reset_password_success: 'reset-password-success',
    reset_password_error: 'reset-password-error',
    reset_password_mfa_email_challenge: 'reset-password-mfa-email-challenge',
    reset_password_mfa_otp_challenge: 'reset-password-mfa-otp-challenge',
    reset_password_mfa_phone_challenge: 'reset-password-mfa-phone-challenge',
    reset_password_mfa_push_challenge_push: 'reset-password-mfa-push-challenge-push',
    reset_password_mfa_recovery_code_challenge: 'reset-password-mfa-recovery-code-challenge',
    reset_password_mfa_sms_challenge: 'reset-password-mfa-sms-challenge',
    reset_password_mfa_voice_challenge: 'reset-password-mfa-voice-challenge',
    reset_password_mfa_webauthn_platform_challenge: 'reset-password-mfa-webauthn-platform-challenge',
    reset_password_mfa_webauthn_roaming_challenge: 'reset-password-mfa-webauthn-roaming-challenge',
    custom_form: 'custom-form',
    consent: 'consent',
    customized_consent: 'customized-consent',
    logout: 'logout',
    logout_complete: 'logout-complete',
    logout_aborted: 'logout-aborted',
    mfa_push_welcome: 'mfa-push-welcome',
    mfa_push_enrollment_qr: 'mfa-push-enrollment-qr',
    mfa_push_enrollment_code: 'mfa-push-enrollment-code',
    mfa_push_success: 'mfa-push-success',
    mfa_push_challenge_push: 'mfa-push-challenge-push',
    mfa_push_list: 'mfa-push-list',
    mfa_otp_enrollment_qr: 'mfa-otp-enrollment-qr',
    mfa_otp_enrollment_code: 'mfa-otp-enrollment-code',
    mfa_otp_challenge: 'mfa-otp-challenge',
    mfa_voice_enrollment: 'mfa-voice-enrollment',
    mfa_voice_challenge: 'mfa-voice-challenge',
    mfa_phone_challenge: 'mfa-phone-challenge',
    mfa_phone_enrollment: 'mfa-phone-enrollment',
    mfa_webauthn_platform_enrollment: 'mfa-webauthn-platform-enrollment',
    mfa_webauthn_roaming_enrollment: 'mfa-webauthn-roaming-enrollment',
    mfa_webauthn_platform_challenge: 'mfa-webauthn-platform-challenge',
    mfa_webauthn_roaming_challenge: 'mfa-webauthn-roaming-challenge',
    mfa_webauthn_change_key_nickname: 'mfa-webauthn-change-key-nickname',
    mfa_webauthn_enrollment_success: 'mfa-webauthn-enrollment-success',
    mfa_webauthn_error: 'mfa-webauthn-error',
    mfa_webauthn_not_available_error: 'mfa-webauthn-not-available-error',
    mfa_country_codes: 'mfa-country-codes',
    mfa_sms_enrollment: 'mfa-sms-enrollment',
    mfa_sms_challenge: 'mfa-sms-challenge',
    mfa_sms_list: 'mfa-sms-list',
    mfa_email_challenge: 'mfa-email-challenge',
    mfa_email_list: 'mfa-email-list',
    mfa_recovery_code_enrollment: 'mfa-recovery-code-enrollment',
    mfa_recovery_code_challenge: 'mfa-recovery-code-challenge',
    mfa_detect_browser_capabilities: 'mfa-detect-browser-capabilities',
    mfa_enroll_result: 'mfa-enroll-result',
    mfa_login_options: 'mfa-login-options',
    mfa_begin_enroll_options: 'mfa-begin-enroll-options',
    status: 'status',
    device_code_activation: 'device-code-activation',
    device_code_activation_allowed: 'device-code-activation-allowed',
    device_code_activation_denied: 'device-code-activation-denied',
    device_code_confirmation: 'device-code-confirmation',
    email_verification_result: 'email-verification-result',
    email_otp_challenge: 'email-otp-challenge',
    organization_selection: 'organization-selection',
    organization_picker: 'organization-picker',
    accept_invitation: 'accept-invitation',
    redeem_ticket: 'redeem-ticket',
    passkey_enrollment: 'passkey-enrollment',
    passkey_enrollment_local: 'passkey-enrollment-local',
    interstitial_captcha: 'interstitial-captcha',
};
/**
 *
 */
export const PatchRenderingOperationPromptEnum = {
    login: 'login',
    login_id: 'login-id',
    login_password: 'login-password',
    login_passwordless: 'login-passwordless',
    login_email_verification: 'login-email-verification',
    signup: 'signup',
    signup_id: 'signup-id',
    signup_password: 'signup-password',
    phone_identifier_enrollment: 'phone-identifier-enrollment',
    phone_identifier_challenge: 'phone-identifier-challenge',
    email_identifier_challenge: 'email-identifier-challenge',
    reset_password: 'reset-password',
    custom_form: 'custom-form',
    consent: 'consent',
    customized_consent: 'customized-consent',
    logout: 'logout',
    mfa_push: 'mfa-push',
    mfa_otp: 'mfa-otp',
    mfa_voice: 'mfa-voice',
    mfa_phone: 'mfa-phone',
    mfa_webauthn: 'mfa-webauthn',
    mfa_sms: 'mfa-sms',
    mfa_email: 'mfa-email',
    mfa_recovery_code: 'mfa-recovery-code',
    mfa: 'mfa',
    status: 'status',
    device_flow: 'device-flow',
    email_verification: 'email-verification',
    email_otp_challenge: 'email-otp-challenge',
    organizations: 'organizations',
    invitation: 'invitation',
    common: 'common',
    passkeys: 'passkeys',
    captcha: 'captcha',
};
/**
 *
 */
export const PatchRenderingOperationScreenEnum = {
    login: 'login',
    login_id: 'login-id',
    login_password: 'login-password',
    login_passwordless_email_code: 'login-passwordless-email-code',
    login_passwordless_email_link: 'login-passwordless-email-link',
    login_passwordless_sms_otp: 'login-passwordless-sms-otp',
    login_email_verification: 'login-email-verification',
    signup: 'signup',
    signup_id: 'signup-id',
    signup_password: 'signup-password',
    phone_identifier_enrollment: 'phone-identifier-enrollment',
    phone_identifier_challenge: 'phone-identifier-challenge',
    email_identifier_challenge: 'email-identifier-challenge',
    reset_password_request: 'reset-password-request',
    reset_password_email: 'reset-password-email',
    reset_password: 'reset-password',
    reset_password_success: 'reset-password-success',
    reset_password_error: 'reset-password-error',
    reset_password_mfa_email_challenge: 'reset-password-mfa-email-challenge',
    reset_password_mfa_otp_challenge: 'reset-password-mfa-otp-challenge',
    reset_password_mfa_phone_challenge: 'reset-password-mfa-phone-challenge',
    reset_password_mfa_push_challenge_push: 'reset-password-mfa-push-challenge-push',
    reset_password_mfa_recovery_code_challenge: 'reset-password-mfa-recovery-code-challenge',
    reset_password_mfa_sms_challenge: 'reset-password-mfa-sms-challenge',
    reset_password_mfa_voice_challenge: 'reset-password-mfa-voice-challenge',
    reset_password_mfa_webauthn_platform_challenge: 'reset-password-mfa-webauthn-platform-challenge',
    reset_password_mfa_webauthn_roaming_challenge: 'reset-password-mfa-webauthn-roaming-challenge',
    custom_form: 'custom-form',
    consent: 'consent',
    customized_consent: 'customized-consent',
    logout: 'logout',
    logout_complete: 'logout-complete',
    logout_aborted: 'logout-aborted',
    mfa_push_welcome: 'mfa-push-welcome',
    mfa_push_enrollment_qr: 'mfa-push-enrollment-qr',
    mfa_push_enrollment_code: 'mfa-push-enrollment-code',
    mfa_push_success: 'mfa-push-success',
    mfa_push_challenge_push: 'mfa-push-challenge-push',
    mfa_push_list: 'mfa-push-list',
    mfa_otp_enrollment_qr: 'mfa-otp-enrollment-qr',
    mfa_otp_enrollment_code: 'mfa-otp-enrollment-code',
    mfa_otp_challenge: 'mfa-otp-challenge',
    mfa_voice_enrollment: 'mfa-voice-enrollment',
    mfa_voice_challenge: 'mfa-voice-challenge',
    mfa_phone_challenge: 'mfa-phone-challenge',
    mfa_phone_enrollment: 'mfa-phone-enrollment',
    mfa_webauthn_platform_enrollment: 'mfa-webauthn-platform-enrollment',
    mfa_webauthn_roaming_enrollment: 'mfa-webauthn-roaming-enrollment',
    mfa_webauthn_platform_challenge: 'mfa-webauthn-platform-challenge',
    mfa_webauthn_roaming_challenge: 'mfa-webauthn-roaming-challenge',
    mfa_webauthn_change_key_nickname: 'mfa-webauthn-change-key-nickname',
    mfa_webauthn_enrollment_success: 'mfa-webauthn-enrollment-success',
    mfa_webauthn_error: 'mfa-webauthn-error',
    mfa_webauthn_not_available_error: 'mfa-webauthn-not-available-error',
    mfa_country_codes: 'mfa-country-codes',
    mfa_sms_enrollment: 'mfa-sms-enrollment',
    mfa_sms_challenge: 'mfa-sms-challenge',
    mfa_sms_list: 'mfa-sms-list',
    mfa_email_challenge: 'mfa-email-challenge',
    mfa_email_list: 'mfa-email-list',
    mfa_recovery_code_enrollment: 'mfa-recovery-code-enrollment',
    mfa_recovery_code_challenge: 'mfa-recovery-code-challenge',
    mfa_detect_browser_capabilities: 'mfa-detect-browser-capabilities',
    mfa_enroll_result: 'mfa-enroll-result',
    mfa_login_options: 'mfa-login-options',
    mfa_begin_enroll_options: 'mfa-begin-enroll-options',
    status: 'status',
    device_code_activation: 'device-code-activation',
    device_code_activation_allowed: 'device-code-activation-allowed',
    device_code_activation_denied: 'device-code-activation-denied',
    device_code_confirmation: 'device-code-confirmation',
    email_verification_result: 'email-verification-result',
    email_otp_challenge: 'email-otp-challenge',
    organization_selection: 'organization-selection',
    organization_picker: 'organization-picker',
    accept_invitation: 'accept-invitation',
    redeem_ticket: 'redeem-ticket',
    passkey_enrollment: 'passkey-enrollment',
    passkey_enrollment_local: 'passkey-enrollment-local',
    interstitial_captcha: 'interstitial-captcha',
};
/**
 *
 */
export const PutCustomTextByLanguagePromptEnum = {
    login: 'login',
    login_id: 'login-id',
    login_password: 'login-password',
    login_passwordless: 'login-passwordless',
    login_email_verification: 'login-email-verification',
    signup: 'signup',
    signup_id: 'signup-id',
    signup_password: 'signup-password',
    phone_identifier_enrollment: 'phone-identifier-enrollment',
    phone_identifier_challenge: 'phone-identifier-challenge',
    email_identifier_challenge: 'email-identifier-challenge',
    reset_password: 'reset-password',
    custom_form: 'custom-form',
    consent: 'consent',
    customized_consent: 'customized-consent',
    logout: 'logout',
    mfa_push: 'mfa-push',
    mfa_otp: 'mfa-otp',
    mfa_voice: 'mfa-voice',
    mfa_phone: 'mfa-phone',
    mfa_webauthn: 'mfa-webauthn',
    mfa_sms: 'mfa-sms',
    mfa_email: 'mfa-email',
    mfa_recovery_code: 'mfa-recovery-code',
    mfa: 'mfa',
    status: 'status',
    device_flow: 'device-flow',
    email_verification: 'email-verification',
    email_otp_challenge: 'email-otp-challenge',
    organizations: 'organizations',
    invitation: 'invitation',
    common: 'common',
    passkeys: 'passkeys',
    captcha: 'captcha',
};
/**
 *
 */
export const PutCustomTextByLanguageLanguageEnum = {
    am: 'am',
    ar: 'ar',
    ar_EG: 'ar-EG',
    ar_SA: 'ar-SA',
    az: 'az',
    bg: 'bg',
    bn: 'bn',
    bs: 'bs',
    ca_ES: 'ca-ES',
    cnr: 'cnr',
    cs: 'cs',
    cy: 'cy',
    da: 'da',
    de: 'de',
    el: 'el',
    en: 'en',
    en_CA: 'en-CA',
    es: 'es',
    es_419: 'es-419',
    es_AR: 'es-AR',
    es_MX: 'es-MX',
    et: 'et',
    eu_ES: 'eu-ES',
    fa: 'fa',
    fi: 'fi',
    fr: 'fr',
    fr_CA: 'fr-CA',
    fr_FR: 'fr-FR',
    gl_ES: 'gl-ES',
    gu: 'gu',
    he: 'he',
    hi: 'hi',
    hr: 'hr',
    hu: 'hu',
    hy: 'hy',
    id: 'id',
    is: 'is',
    it: 'it',
    ja: 'ja',
    ka: 'ka',
    kk: 'kk',
    kn: 'kn',
    ko: 'ko',
    lt: 'lt',
    lv: 'lv',
    mk: 'mk',
    ml: 'ml',
    mn: 'mn',
    mr: 'mr',
    ms: 'ms',
    my: 'my',
    nb: 'nb',
    nl: 'nl',
    nn: 'nn',
    no: 'no',
    pa: 'pa',
    pl: 'pl',
    pt: 'pt',
    pt_BR: 'pt-BR',
    pt_PT: 'pt-PT',
    ro: 'ro',
    ru: 'ru',
    sk: 'sk',
    sl: 'sl',
    so: 'so',
    sq: 'sq',
    sr: 'sr',
    sv: 'sv',
    sw: 'sw',
    ta: 'ta',
    te: 'te',
    th: 'th',
    tl: 'tl',
    tr: 'tr',
    uk: 'uk',
    ur: 'ur',
    vi: 'vi',
    zgh: 'zgh',
    zh_CN: 'zh-CN',
    zh_HK: 'zh-HK',
    zh_TW: 'zh-TW',
};
/**
 *
 */
export const PutPartialsPromptEnum = {
    login: 'login',
    login_id: 'login-id',
    login_password: 'login-password',
    login_passwordless: 'login-passwordless',
    signup: 'signup',
    signup_id: 'signup-id',
    signup_password: 'signup-password',
    customized_consent: 'customized-consent',
};
/**
 *
 */
export const GetSelfServiceProfileCustomTextLanguageEnum = {
    en: 'en',
};
/**
 *
 */
export const GetSelfServiceProfileCustomTextPageEnum = {
    get_started: 'get-started',
};
/**
 *
 */
export const PutSelfServiceProfileCustomTextLanguageEnum = {
    en: 'en',
};
/**
 *
 */
export const PutSelfServiceProfileCustomTextPageEnum = {
    get_started: 'get-started',
};
/**
 *
 */
export const DeleteMultifactorByProviderProviderEnum = {
    duo: 'duo',
    google_authenticator: 'google-authenticator',
};
/**
 *
 */
export const DeleteUserIdentityByUserIdProviderEnum = {
    ad: 'ad',
    adfs: 'adfs',
    amazon: 'amazon',
    apple: 'apple',
    dropbox: 'dropbox',
    bitbucket: 'bitbucket',
    aol: 'aol',
    auth0_oidc: 'auth0-oidc',
    auth0: 'auth0',
    baidu: 'baidu',
    bitly: 'bitly',
    box: 'box',
    custom: 'custom',
    daccount: 'daccount',
    dwolla: 'dwolla',
    email: 'email',
    evernote_sandbox: 'evernote-sandbox',
    evernote: 'evernote',
    exact: 'exact',
    facebook: 'facebook',
    fitbit: 'fitbit',
    flickr: 'flickr',
    github: 'github',
    google_apps: 'google-apps',
    google_oauth2: 'google-oauth2',
    instagram: 'instagram',
    ip: 'ip',
    line: 'line',
    linkedin: 'linkedin',
    miicard: 'miicard',
    oauth1: 'oauth1',
    oauth2: 'oauth2',
    office365: 'office365',
    oidc: 'oidc',
    okta: 'okta',
    paypal: 'paypal',
    paypal_sandbox: 'paypal-sandbox',
    pingfederate: 'pingfederate',
    planningcenter: 'planningcenter',
    renren: 'renren',
    salesforce_community: 'salesforce-community',
    salesforce_sandbox: 'salesforce-sandbox',
    salesforce: 'salesforce',
    samlp: 'samlp',
    sharepoint: 'sharepoint',
    shopify: 'shopify',
    sms: 'sms',
    soundcloud: 'soundcloud',
    thecity_sandbox: 'thecity-sandbox',
    thecity: 'thecity',
    thirtysevensignals: 'thirtysevensignals',
    twitter: 'twitter',
    untappd: 'untappd',
    vkontakte: 'vkontakte',
    waad: 'waad',
    weibo: 'weibo',
    windowslive: 'windowslive',
    wordpress: 'wordpress',
    yahoo: 'yahoo',
    yammer: 'yammer',
    yandex: 'yandex',
};
/**
 *
 */
export const GetUsersSearchEngineEnum = {
    v1: 'v1',
    v2: 'v2',
    v3: 'v3',
};
